{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import nest_asyncio\n",
    "nest_asyncio.apply()\n",
    "\n",
    "from llama_index.core import SimpleDirectoryReader\n",
    "\n",
    "from plotreader.utils.document import DirectoryHandler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Ignoring wrong pointing object 92 0 (offset 0)\n"
     ]
    }
   ],
   "source": [
    "# what does a default Document look like?\n",
    "documents = SimpleDirectoryReader(input_files=[\"/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf\"]).load_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(id_='c39edfcc-f630-48f9-a34f-710a7486739c', embedding=None, metadata={'page_label': '1', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogeneticsClaire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana Gradinaru1,*, Frances H. Arnold1,2,*1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USAAbstractWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.IntroductionChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation1. These channels are widely applied as tools in neuroscience research2; however, functional Users may view, print, copy, and download text and data-mine the content in such documents, for the purposes of academic research, subject always to the full Conditions of use:http://www.nature.com/authors/editorial_policies/license.html#terms*Corresponding: viviana@caltech.edu (V .G.) & fha@cheme.caltech.edu (F.H.A).†K.K.Y . & J.E.R. contributed equally to this work.Author ContributionsC.N.B., K.K.Y ., V .G., and F.H.A. conceptualized the project. C.N.B. coordinated all experiments and data analysis. C.N.B. and K.K.Y . built machine-learning models. C.N.B. performed construct design and cloning. C.N.B. and E.D.M. performed AA V production. E.D.M. prepared cultured neurons. C.N.B and J.E.R. conducted electrophysiology. C.N.B. and J.E.R. performed injections. J.E.R. performed fiber cannula implants and behavioral experiments. C.N.B. performed all data analysis. C.N.B. wrote the manuscript with input and editing from all authors. V .G. supervised optogenetics/electrophysiology, and F.H.A. supervised the protein engineering.Competing interestsA provisional patent application (CIT File No.: CIT-8092-P) has been filed by Caltech based on these results. C.N.B., K.K.Y ., V .G., and F.H.A. are inventors on this provisional patent.\\nHHS Public AccessAuthor manuscriptNat Methods. Author manuscript; available in PMC 2020 April 14.Published in final edited form as:Nat Methods. 2019 November ; 16(11): 1176–1184. doi:10.1038/s41592-019-0583-8.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a462206e-6e65-49ea-867f-30c402c6e8c5', embedding=None, metadata={'page_label': '2', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm!2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 µm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 ‘designer’ ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AA V), rAA V-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.Bedbrook et al.Page 2\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='799c6054-269c-475f-8ba1-5a8eb61edafb', embedding=None, metadata={'page_label': '3', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='ResultsFunctional characterization of ChR variants for machine learningIn previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2\"310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.Training Gaussian process (GP) classification and regression modelsUsing the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a ‘contact map’ that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural Bedbrook et al.Page 3\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c92a1780-b52b-439d-b379-37b0d39c5dad', embedding=None, metadata={'page_label': '4', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).Selection of designer ChRs using trained modelsTo select ‘designer’ ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model’s ability to make useful predictions about novel functional sequences, even for sequences that are very distant from Bedbrook et al.Page 4\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='51b072a6-2821-4dd4-8628-b315695a77a9', embedding=None, metadata={'page_label': '5', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.Sequence and structural determinants of ChR functional propertiesWe used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate1. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed14,27. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position Bedbrook et al.Page 5\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f2e00e2b-fa78-4396-bd2e-f997778bef57', embedding=None, metadata={'page_label': '6', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.Machine-guided search identifies ChRs with a range of useful functional propertiesWe assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models’ predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with #2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit #1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.Designer ChRs’ off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10’s red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have #9-times larger currents at the lowest intensity of light tested (10!1 mW mm!2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10!1–101 mW mm!2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are Bedbrook et al.Page 6\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='54fd1549-1a21-44d1-a1b3-3c60756030ed', embedding=None, metadata={'page_label': '7', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm!2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5\"10!2 mW mm!2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.Validation of designer ChRs for neuroscience applicationsFor further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAA V-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10!3 mW mm!2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10!2 mW mm!2 light, a four-fold improvement over ChR2(H134R)’s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)’s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitationWe investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of ‘non-invasive optogenetics’ relied on invasive intracranial virus delivery, which results in Bedbrook et al.Page 7\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='823ec766-8664-4d1b-8178-fc39c26121c8', embedding=None, metadata={'page_label': '8', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AA V capsid rAA V-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAA V-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAA V-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAA V-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.We systemically delivered rAA V-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAA V-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in Bedbrook et al.Page 8\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7879c591-be5d-47fd-ad25-efdf9f81250d', embedding=None, metadata={'page_label': '9', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAA V-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.DiscussionWe demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other Bedbrook et al.Page 9\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='499544fb-1ba7-46c1-b713-384ced2b88d5', embedding=None, metadata={'page_label': '10', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AA Vs30.We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2’s application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.Online methodsConstruct design and cloningThe design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAA V-hSyn vector backbone [Addgene plasmid #26973], a pAA V-CamKIIa vector backbone [Addgene plasmid #51087], and a pAA V-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).HEK293T cell and primary neuronal culturesThe culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer’s (Promega) recommendations. Cells were given 48 hours to express the ChRs Bedbrook et al.Page 10\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='845e004b-3a26-484e-a8ac-49b4968385c3', embedding=None, metadata={'page_label': '11', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='before photocurrent measurements. Primary hippocampal neuronal cultures were prepped from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented with glutamine and B27. Cells were transduced 3–4 days after plating with rAA V-PHP.eB packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings were performed 5–10 days after transduction.Patch-clamp electrophysiologyWhole-cell patch-clamp and cell-attached recordings were performed in transfected HEK cells, transduced cultured neurons, and acute brain slices to measure light-activated inward currents or neuronal firing. For electrophysiological recordings, cultured cells were continuously perfused with extracellular solution at room temperature (in mM: 140 NaCl, 5 KCl, 10 HEPES, 2 MgCl2, 2 CaCl2, 10 glucose; pH 7.35) while mounted on the microscope stage. For slice recordings, 32 °C artificial cerebrospinal fluid (ACSF) was continuously perfused over slices. ACSF contained 127 mM NaCl, 2.5 mM KCl, 25 mM NaHCO3, 1.25 mM NaH2PO4, 12 mM d-glucose, 0.4 mM sodium ascorbate, 2 mM CaCl2, and 1 mM MgCl2 and was bubbled continuously with 95% oxygen / 5% CO2. Firing and photocurrent measurements were performed in the presence of 3 mM kynurenic acid and 100 µM picrotoxin to block optically evoked ionotropic glutamatergic and GABAergic currents, respectively.Patch pipettes were fabricated from borosilicate capillary glass tubing (1B150–4; World Precision Instruments) using a model P-2000 laser puller (Sutter Instruments) to resistances of 3–6 MΩ. Pipettes were filled with K-gluconate intracellular solution containing the following (in mM): 134 K gluconate, 5 EGTA, 10 HEPES, 2 MgCl2, 0.5 CaCl2, 3 ATP, and 0.2 GTP. Whole-cell patch-clamp and cell-attached recordings were made using a Multiclamp 700B amplifier (Molecular Devices), a Digidata 1440 digitizer (Molecular Devices), and a PC running pClamp (version 10.4) software (Molecular Devices) to generate current injection waveforms and to record voltage and current traces.Photocurrents were recorded from cells in voltage clamp held at !60 mV . Neuronal firing was measured in current clamp mode with current injection for a !60 mV holding potential. Access resistance (Ra) and membrane resistance (Rm) were monitored throughout recording, and cells were discarded if Ra or Rm changed more than 15%. During ChR variant functional screening in HEK cells, photocurrents were only recorded from cells that passed our recording criteria: Rm > 200 MΩ and holding current >!100 pA. Our measured membrane properties of ChR expressing neurons were consistent with previous literature of opsin-expressing cells11 and are also consistent with previous reports of properties of cultured hippocampal neurons38,39 and PFC neurons in slice40,41 (Supplemental Figures 10). For cell culture experiments, the experimenter was blinded to the identity of the ChR being patched but not to the fluorescence level of the cells. For acute slice recordings, the experimenter was not blinded to the identity of the ChR.Bedbrook et al.Page 11\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='32c4aeac-d091-43eb-a071-c0298df5df98', embedding=None, metadata={'page_label': '12', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Light delivery and imagingPatch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm!2, 546 nm light at 2.8 mW mm!2, and 640 nm light at 2.2 mW mm!2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm!2.Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.Electrophysiology data analysisElectrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >!150 pA (holding at !60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).AAV production and purificationProduction of recombinant AA V-PHP.eB packaging pAA V-hSyn-X-TS-eYFP-WPRE, pAA V-CAG-DIO[X-TS-eYFP]-WPRE, and pAA V-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in Bedbrook et al.Page 12\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='91cc1829-f16f-4326-921f-7f1d7e27f5f6', embedding=None, metadata={'page_label': '13', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.AnimalsAll procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.Intravenous injections, stereotactic injections, and cannula implantationIntravenous administration of rAA V vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at !60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 µl of purified AA V vectors at 5\"1012 vg ml!1 targeting the following coordinates: anterior-posterior (AP), !1.7; media-lateral (ML), +/! 0.5; and dorsal-ventral (DV), !2.2. For stimulation of the VTA, 300 µm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 µm above the VTA bilaterally targeted to the following coordinates: AP, !3.44 mm; ML, +/!0.48 mm; DV , 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 µm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse43 to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).Gaussian process modelingBoth the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use Bedbrook et al.Page 13\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cc0275e6-0351-464f-896b-6a7a623c2af4', embedding=None, metadata={'page_label': '14', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [(‘A134’), (‘M1’)]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).GP regression—In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance /uni03C3n2, the posterior distribution of f*=fx* for Gaussian process regression is Gaussian with meanf*=k*TK+/uni03C3n2I−1y(1)and variancev*=kx*,x*−k*TK+/uni03C3n2I−1k*(2)Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i=kx*,xi. The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:Bedbrook et al.Page 14\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='69f437a6-f55c-4c14-bfd2-f1a22a887b83', embedding=None, metadata={'page_label': '15', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='logpy|X=−12yTK+/uni03C3n2I−1y−12logK+/uni03C3n2I−n2log 2/uni03C0(3)where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.GP classification—In binary classification, instead of continuous outputs y, the outputs are class labels yi ∈ {+1, !1}, and the goal is to use the training data to make probabilistic predictions /uni03C0x*=py*=+ 1x*. We use Laplace’s method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.GP kernels for modeling proteins—Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector xse that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein’s structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector xst that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.Three types of kernel functions k(si, sj) were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein’s fitness landscape. The polynomial kernel is defined as:ks,s/uni2032=/uni00A0/uni03C302+/uni03C3p2xTx/uni2032d(4)where σ0 and σp are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:ks,s/uni2032=/uni00A0/uni03C3p2exp−x−x/uni203222l(5)where l and σp are also hyperparameters and |⋅|2 is the L2 norm. Finally, the Matérn kernel with v=52 is defined as:ks,s/uni2032=/uni00A01+5x−x/uni203222l+5x−x/uni2032223l2exp−5x−x/uni203222l(6)Bedbrook et al.Page 15\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6362f646-0209-40f7-b698-5d79df236555', embedding=None, metadata={'page_label': '16', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152] 41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807] 42. Deverman BE et al. Cre-dependent selection yields AA V variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280] 44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).Bedbrook et al.Page 19\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2e756391-2fe8-448d-829a-26b0ec8c8938', embedding=None, metadata={'page_label': '17', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents. (a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels. (b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm!2, 481 nm light at 2.3 mW mm!2, 546 nm light at 2.8 mW mm!2, and 640 nm light at 2.2 mW mm!2. (c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm!2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure. (d) Peak and steady-state photocurrent strength Bedbrook et al.Page 22\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b4936087-26a6-45e5-8011-192dd7bcfdef', embedding=None, metadata={'page_label': '18', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.Bedbrook et al.Page 23\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2398081d-95d1-4816-bc93-44ba66306a0a', embedding=None, metadata={'page_label': '19', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR. (a) ChRs were cloned into an AA V vector with either the hSyn or CamKIIa promoter and packaged into rAA V-PHP.eB for expression in culture and in vivo. (b) Cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter (repeated independently six times per construct with similar results). (c) Peak and steady-state photocurrent with low-intensity (8\"10!3 mW mm!2) and moderate-intensity (0.8 mW mm!2) light in cultured neurons (ChR2, n = 16 cells; CoChR, n = 17 cells; ChRger1, n = 9 cells; ChRger2, n = 24 cells; ChRger3, n = 9 cells). (d) Spike fidelity with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1, n = 6 cells; ChRger2, n = 6 cells; ChRger3, n = 6 cells; CoChR, n = 7 cells; ChR2, n = 7 cells). (e) Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons (ChRger1, n = 9 cells; ChRger2, n = 12 cells; ChRger3, n = 7 cells; ChR2, n = 8 cells). (f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3\"10!2 mW mm!2) shows robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization. (g) Photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAA V-PHP.eB packaged hSyn-ChR constructs into the PFC (ChRger1, n = 11 cells; ChRger2, n = 11 cells; ChRger3, n = 11 cells; ChR2, n = 9 cells) or after systemic delivery of CamKIIa-ChRger2 (ChRger2, n = 6 cells; 5\"1011 vg/animal). (h) Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at Bedbrook et al.Page 24\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8ecdab39-7a3c-420f-90c8-774ed7158ba1', embedding=None, metadata={'page_label': '20', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAA V-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5\"1011 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1\"1011 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1\"1011 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1\"1011 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm!2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm!2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.Bedbrook et al.Page 25\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a62a6e78-d1f0-42a2-a78d-872e34cd69c9', embedding=None, metadata={'page_label': '21', 'file_name': 'nihms-1538039.pdf', 'file_path': '/Users/loyalshababo/dev/plotreader/real_figures/opsins/nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1799688, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-19'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 4. Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation. (a) Systemic delivery of rAA V-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3\"1011 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom). (b) Minimally-invasive, systemic delivery of rAA V-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5\"1011 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.Bedbrook et al.Page 26\\nNat Methods. Author manuscript; available in PMC 2020 April 14.\\nAuthor Manuscript Author Manuscript Author Manuscript Author Manuscript', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n')]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "documents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# let's use our llamaparse approach\n",
    "\n",
    "reader = DirectoryHandler(\n",
    "    name=\"nihms-1538039\",\n",
    "    dirpath=\"/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp\",\n",
    "    storage_dir=\"./storage/storage_tmp\",\n",
    "    desc=\"A scientific paper\",\n",
    "    use_cache = False\n",
    ")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started parsing the file under job_id 7eae79ff-80ee-4f2d-b401-4b46319b6851\n",
      "410\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n"
     ]
    }
   ],
   "source": [
    "docs = reader.load_docs()\n",
    "print(len(docs))\n",
    "print(docs[0].text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(id_='75734b40-ec05-4165-a8bf-ef9876880c04', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ac321ed6-cefa-49bb-9559-5db5be8506d7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='161cc53b-3562-4bbc-94e7-f1f59dff42c2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b942bab3-993d-454b-8e85-8f065ac83635', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0a3a7174-98eb-4f09-9bd7-5ac9005e2d1e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a1fe3e50-5c82-4327-9dd3-ef9d97152e44', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='87b618f7-bbb5-4431-8475-3bbb637246f1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='41310b10-07a7-4e21-a4d3-fa0381a4878d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='242b04e9-9200-422f-b3d4-c58f093bac27', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Abstract', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='73238c9a-b374-405d-8d3a-a2beda212be6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a091d9b9-a2c9-4fb7-9833-22b8b1a4bde8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3ab3ffa1-d116-47bd-b4ec-3d27399ef456', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='28b773b8-5715-4bd7-bc56-b238f689f3ae', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Introduction', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f640cfb1-dbbf-42d8-80cb-c25fd6900650', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4caec2fd-58ec-4887-b91d-aaa271c1b8a4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0edd981d-bfb1-4fee-9443-1a14b7f716e7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ec18af9d-ce11-41a8-abbd-85ff9781c2f8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='[The rest of the introduction is cut off in the image]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9289bc7d-8102-42ee-af36-78611cc459e4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='992e4d2b-042f-4db4-bab3-63c8d88c38b9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='079e026e-eb04-47a8-8473-0d7031d01a35', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1c178438-910b-4d3d-bab4-3a8ec33a834c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='62694194-c71c-458c-84f5-da7841d829fb', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2ca7bc1c-aa1f-4500-8c84-de560642b59a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='72a4e741-d04f-4a44-93f0-6eff7f2519f9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1a58c713-0e19-4bd4-8601-71640c2ea9c3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='69b9aaec-cebd-456a-a327-051957a5efa9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='89298b42-5205-45d4-9283-2726a3f757f3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='# Results', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2ebb1687-1b11-45d5-adbf-87d5daa9174e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f512c5fd-ed41-4c63-ade3-9a3c97f8a4e3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Functional characterization of ChR variants for machine learning', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a0faccef-3a78-46c3-bb8f-d5b32dc8a87f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='124efce1-a11b-4954-aaff-ea1c8e40e406', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2559eae3-f1c0-4f8e-a155-c484ee7fad38', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='71770c47-d179-46b0-8352-6c8775494482', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='46ce8fad-18ac-4dab-a7e2-09e2e2c0f86e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='67e7362c-0073-4ac4-a1e0-1c6a117309fd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Training Gaussian process (GP) classification and regression models', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1547caa0-4eef-437f-82d0-94de6c9e0cc3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2fa55879-e4e6-4a1b-adfe-c7a46155a97b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5cd92ec5-8896-4220-b103-878b6470408f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='56bdfcf2-24d5-4be9-ac3c-0e0a34124fa1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9afe53d2-5245-4805-81b1-d515598196c4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f4a6bc53-1cf3-4652-8a75-29a7f3019781', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d2d32234-a73c-4d90-aed8-89b13de9d6e0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1177ba71-4d3d-4d15-abfd-76de48145fb9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bf562cbd-6ceb-4a3e-9bc6-aababdfede6d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Selection of designer ChRs using trained models', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7a9cf3d9-4962-4e75-b67e-883457c3296f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='275ac4a8-4397-4235-ba40-3796cf3e861f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ce659b73-d217-4708-a4f9-6d3aa714be7e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='02dff716-1b4b-41c4-ae72-9b4bd5cdc664', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='49d8d6e8-92ef-4d3c-866f-b5cf2442cc9c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='241ca5fe-d3e7-4c37-961d-595fafa9ac87', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='357d1a91-bdba-466b-ad91-fbe168202873', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='222803d6-8935-4f29-b5d8-3a2955ac18fc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='11095a27-0e96-4071-a350-10fde8f18d5e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='15186592-12d7-472e-add0-b5065f27cd1f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6668ac20-478f-4514-a2dc-d511bbaae890', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Sequence and structural determinants of ChR functional properties', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='be052c10-e0cd-4c1a-8f11-3609e12a02d6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a660a842-03e5-4829-a782-0f23b35b61d2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='390babc9-0f01-481f-97a0-99ec6215b5fe', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aa2facad-de2e-47f1-a427-b15f32f3e672', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f2674b60-8f6f-4cbb-a813-6e0d389bf236', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Machine-guided search identifies ChRs with a range of useful functional properties', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8540f231-85ed-4f83-aaae-b9c46603c981', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='40d87471-95f2-44ec-9f0a-b28b37cc7cc5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2f0bab1c-e90f-4ff8-98ad-c18c9efdd4ba', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5cb7a5ef-7442-40a6-b358-81bd0f954e4b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b48087b1-f0e1-43d7-9359-4b221f4921b6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ce69428d-3373-4c39-949d-556ca5449096', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='42daabaf-9f64-4a04-a768-ee49735988ea', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d9182c88-ae83-472c-b8a3-31e15bee781e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b2e62129-baec-4ab4-9dfa-f694e6e138e2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d4476ad6-4efa-4624-bf6f-08d63860e05f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9ea858b8-5010-4a3e-bce5-3a786443650a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d2b130d3-4a1f-4b8e-b2bb-d0a6297d6f81', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1df86c0f-165c-4af6-91c6-8d38573aa91d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Validation of designer ChRs for neuroscience applications', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='023e2452-e449-45aa-a72f-42a781bb59de', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d7dce03b-44bb-4bb0-b97b-87a4fa7a25d2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cb95148a-894d-4545-ae63-79f602a4d6ac', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e0b60418-714b-47bf-8f0d-1b425fb5d0e2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d4b543a0-9920-43c5-95e1-6cf9a9d0a3c1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='92dd2032-b488-4032-bd82-b9c08dc3f308', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='377ba76d-bc2d-4e56-8770-d3579bcb167f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='32202850-7513-4044-a21a-ae160b3006d3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='405e5b19-587f-4dad-b530-27bb018caf2e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7aa987d1-db6c-42b0-9cec-9ce95e3e0a44', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='63fc69b9-59c6-4f65-b489-a0e8a93e0aa7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='141d952e-3819-4279-965d-fd6c815288cd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='81520d4b-2b2c-4a45-b6ef-6b170eb1881e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='56790098-100e-4e6a-bfa3-646da3b039f9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='59013224-969e-4485-bae2-1517ca686959', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5493d7dc-8e9e-4881-8166-3bc0c6945137', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4a386b73-b692-4508-abaf-78a7ceeb7d87', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='feee84e3-34db-4965-bfb3-a15c9410d5d7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='475c4e4d-4661-40f6-bad0-503a0f71148a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='99fce207-f72e-4d70-b7d1-9c79115318ee', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Discussion', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9ddcb221-6aed-40fd-b68e-d0e435ee8963', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='26bd6ecd-9735-47b9-8970-f7766742927f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='271b43cb-b5de-4ccc-8756-d4cd69d75fd0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ceffe99b-a01f-4460-bcf3-34375a7afe37', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='11f522f6-adb7-4042-b0e6-b0f609a13e82', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='954a9d12-8796-4c38-a4b4-16c2959e4b6d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='96ed6af9-6913-4b68-8bbc-e7201a91b42b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7f2ef611-beaf-4dbe-a911-d9e1836e1259', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c8ff2f83-c88f-4ea6-af2a-c3f943d4d8b2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Online methods', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e9d56a32-3c2f-47eb-902e-894cc334ac21', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ce136834-9ded-4ecf-b63f-7cf2901ad212', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='### Construct design and cloning', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3f35b648-4093-4c40-8a58-2e257e287994', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2e325fc0-fb8e-4e62-8227-84c397de2873', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d01347ca-7187-41f4-b7fb-f3e91bed9868', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='30eaf107-dece-4011-90fa-9b5c8c322413', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f4341171-9e95-4d67-aaea-4e9588575451', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='513413d9-517c-4ccd-ab95-11a4fb2c1c18', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e058fbf0-106b-4f37-9fbb-5382a0cee172', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8efd430c-7aa6-4576-893a-c0e27017258e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='### HEK293T cell and primary neuronal cultures', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f4ae0df1-976a-40c0-905d-884651da65a7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aa1a9022-91f6-4dff-98f6-547214b9547e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='491763a1-1a55-4b8a-a95c-9c119342daec', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='NO_CONTENT_HERE', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='15ca8dbc-f1df-47c3-a550-105d294704aa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Light delivery and imaging', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='299b8abf-ef86-44d4-b2e4-c8db567fbce3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='96f48e11-93f5-45aa-b963-ad2c323746a9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='69820880-9ebd-4a35-b0f3-3eda0c7683a6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5335dcbf-6ed9-475f-a0b5-783bc2807ae5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e9ae45e7-d286-4b03-a08e-f84150ff2f89', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3a8be7db-a4f8-4b59-ad7a-2be6c77ee7dc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Electrophysiology data analysis', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='90f98c66-64e6-46c8-a095-3ef42deef657', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='79b4af6a-4846-4c16-82c4-a0da17a17bd5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8a856d4d-0a87-4970-975d-5b2b1cc192d7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='883563f1-b0d9-4f72-bf22-66ce28c0354c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## AAV production and purification', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8d07e45c-d95d-41a7-a521-07d0d21ee5d2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='13b20d83-f4ab-4c2d-aabd-006fa146d274', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e8c73c21-c93c-44f8-982c-c61de76dfc73', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='31608dbb-0030-4894-8286-f6e8129b10d3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cd7aafbe-5da5-48f4-bb0b-3aec2c7618df', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Animals', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8491983e-b3f3-48c4-b62e-f1fbc2dcc689', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2ddb7dfd-6e6e-437c-99ff-ecb52d963702', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='daeeb345-f0c9-4218-b39f-819153230921', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f043a6ea-d4f4-4196-9359-0c2dab9ec93b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Intravenous injections, stereotactic injections, and cannula implantation', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='99f651e4-5bec-440a-9a02-4464c976090e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bd26aead-5b23-4b07-a940-72211aad3b1c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6aacc0f8-342f-4af8-a6e5-36cc9d255ec5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4b7fccf4-3892-4233-9490-f9142ec0557a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='32ca5799-3b31-4794-8070-bf0a24cc4310', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bf4a8f73-50ba-4ef3-8e22-817eb5b926aa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Gaussian process modeling', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1f8c652d-3831-4049-9e69-329d883cf43b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='11b0656c-0044-4f04-aa00-1ad61b73458a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6a56713b-7b8b-4a39-8c6e-ddf5d3c81f34', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fae778a5-b511-41c2-be26-a7b2036c9d2c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d23f66ac-f91f-47ca-bbbe-3b37fd1f0bff', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6efccaa3-4470-4e22-b5cd-a38fb67a9dc4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c191bd06-acac-4439-a9cb-347be4748fbd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='db28ad9a-648e-49f0-98a4-922b15f0fa38', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='85dc5b02-5e5b-44ff-9ac7-d6d00074d7bc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d9a41c3c-5876-4dee-b07b-f7152b6848aa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='40b1d03e-bc27-4000-acf4-329d4d750406', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## GP regression', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='db92a412-e0df-4993-9083-3b84feafaf45', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='264b8203-f03a-4bbc-96a3-bc4ed3d52a30', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='605b9535-59f4-4ee5-a891-7e7f7c028fa1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fa3bb279-6284-45c0-91ae-43921abde6ee', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='$$f* = k*^T(K + σn^2I)^{-1}y$$', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fa581a9e-843b-491d-ab0e-fc68312c6867', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='876da140-4643-49f4-9357-0627442cee08', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='and variance', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aaff62e9-da92-4a46-9d52-745ff09447c8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f0ce1042-8a3b-4739-b740-65f65ce4076b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fd10391b-43ee-45dd-b95a-8ecd7a86cca1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2fd89ce6-1985-4d8a-b3bb-8af446bbd709', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7b95e4cd-342f-417d-852e-9f2fea1f79ba', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9c5d66f5-3abe-4765-af3c-e4c64e9aa3d8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='614ce48a-2cd6-4ce2-a762-e5551e2ea27c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='182a670b-54a6-49b9-b0c4-712fcb9aaa3e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dae215c8-60a7-4e6e-b690-d493c621b0bf', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## GP classification', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='75196220-85f0-487c-a100-600ad5561937', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6e2c9a53-e2eb-4976-886e-c332c055d79d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a2bb8b28-833e-4dd6-bcfd-108a24450fed', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='074dff95-6b8b-4090-9194-9623ab6949ce', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## GP kernels for modeling proteins', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8331f9f3-d899-4e8c-85f8-43bd39513146', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6c0a57dc-ccc9-4ce8-a2bc-6b3e624bc504', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d49cf3a9-c083-438c-b17a-9d3398964ee2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='820df3c7-39d4-4c69-af3c-eb3aef3e86f0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='65a511e8-5296-4ff2-9a17-f5055d8ab84c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c8a1c33a-a5ad-4630-b349-386b7efcd304', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='88e67c07-a029-4903-98e7-1ed5947b9f0e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='57c130f1-993b-46d4-a25f-210d56f9b07d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='where $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9c700f58-f843-461a-a9a4-ac2aa23e472c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='22ec94ec-1c52-4d0a-abc8-e6543ff2f405', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='66dc53d8-e284-4f5d-aa16-6fdac0c6d2e5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a2d7ba30-6cf8-4898-a029-8c7ded75835c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='where l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8cc80721-9afa-477e-af45-76d4b421dfc3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='11ec0ee1-17cf-465e-8486-76db1854dff7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ca5950e4-e2ac-47f0-b0af-e691f56423af', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Where l is once again a hyperparameter.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c707e4cf-6103-464f-8343-4823ef0f7318', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='63e2ba54-30ef-4b25-9b87-41af90136195', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## L1 regression feature identification and weighting', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cdd0a482-fce8-4c5f-9ec8-a485e1896e8f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='00e92dde-1b70-4fca-94be-7fe18dfb1974', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bd84d268-c3de-4a25-b601-06f7495d9300', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2123ae20-4bc2-47aa-adbe-25840a729393', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Statistical analysis', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f18d0c5c-0d05-4fb8-ac7c-6f172f8dab33', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dac1bb6e-71f0-4b44-b581-82785e40c8ef', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f1891082-1677-42f3-8908-7b4e5e9767b8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9e740c0f-0970-49fa-ba1a-305f4289d5e4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Data availability', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='13737d7b-574d-48e9-a016-f581e3a0da5f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3f5935de-d155-462d-9268-19da664e6696', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='461245f8-0dde-4345-8fa1-34d588a54878', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6fe288a8-03d5-45ae-8dda-abe34bf1e138', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Code availability', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e47d4b1a-08e8-4ead-a6c5-6219d70fd8a5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b895e032-eeaf-4de8-9c6c-ca4abe2c6390', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aa8fe24e-61f2-47b0-82e5-bfa77d366e9b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6abbcf8d-4903-4c08-bd63-8c3f8f5c7220', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Supplementary Material', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7434ec67-9b58-4823-bf65-e39501dcd56d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='54a3b729-8628-422a-9830-9e2e8e22155f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Refer to Web version on PubMed Central for supplementary material.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6459e850-89fe-4752-918b-f0a964b3e6ba', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='87682561-1fed-4a84-9bcf-36716f04fa51', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Acknowledgements', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='10dd5191-5297-4d4a-a21e-2e068f24f1c5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3a040ea2-7536-4d46-aa6b-4aaae56a1d20', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2dc03fb6-e168-42de-a353-ce88b0c27d87', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='00008904-41df-41d3-9742-cc0eaea95c78', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='99d2ca85-fc6b-41f9-aa2c-ace7b88fe0dd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## References', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9382a076-1b15-4495-bc62-7eff65b9ee5a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f56a8d1c-f961-48ec-b906-4f94d53d9e62', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0c7b9341-6d5b-4c63-89c4-69585663a87c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7ac378e0-9383-43ef-8582-550bd42e0a7d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0ce1ea1d-78be-483a-81a7-c23c9cc5520c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c011cf94-a815-4cd7-8f23-97323985fef5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text=\"3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dc695ac6-4efc-44db-959b-b47ac628096a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8316efe7-ab3c-495c-bc4b-062dd553631b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6fb88953-457b-455e-9fe4-418a70686b49', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='121e1a95-c1bc-4bd8-baa2-04fbd11295e8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='41424642-864f-4ea2-bab6-8cf6886ca8ce', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d5304680-a56f-4a50-9394-13327b0b68c2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a52a8975-e7b2-4241-b3e9-cfb5702d28e2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fcd77035-1d8a-4089-bb9a-5a1dc9ca014d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3d1db074-4a22-413f-8470-5cc3eb98da91', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aa8ba354-acdc-483e-b638-0c7b15446d40', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9081ca50-47bb-415b-a0de-17eb5ec9442c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='51987ce9-1701-4b11-947a-fe8e98efb0c4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='734f92e6-4230-49e9-a876-052508279f05', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ddda2d92-21f1-4ce8-8e11-b5bb5fac8e0a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6cec48b6-b7ce-4428-8a26-d6c0261968c3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='57e93a34-7978-43d3-944d-40eae297ce0b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ff847f67-2fb1-41bd-b7ce-165712c641a9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='719dffb0-1d37-4e7a-8c6c-e7c363602887', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='28c38d0f-31cc-43d1-943a-7adcd6a46f48', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d4f79401-127c-4199-a43b-49ad2593ba93', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ac77aa2e-d48f-4317-8c8e-06bbbc8c69a3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c3f6b7bf-bbec-482e-873e-f2e963d402dc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='66e263fa-bcf4-48ff-b549-d07c55f504af', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e941d770-2ddb-45bc-88fa-aa4c2ce8be1a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aa41785d-620b-41b7-92df-7126fc6b78f9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7e2bf6b4-5518-410a-b59e-f46c98aeb53b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dc0f14b2-9066-45a1-82f1-b2d125f2f21d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3f14f6fb-28f2-4ef8-ae7a-d76eecfeb7d2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3a720a56-13ae-4714-b8f7-a518586d5407', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='88462280-e7a7-410a-838c-f69b047894f5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f04efcf8-d824-48f2-a49b-026a306b733b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='33499264-2aa8-4afa-a6c6-b1413c62ed2b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='08b7d50e-9f31-48fd-b60b-49f308408e23', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='65e8ab79-c715-482b-a2a1-44bf836da205', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2f5265ef-a7e5-4e4f-a184-e79da9627fb2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='07d9a07a-979f-4542-b4e2-afe1b8fe2da1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4dd2fda7-f85a-49a3-9bb6-bcf6bcb1199d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4edea069-40d8-49a0-990b-4850f3d099ba', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='98539fda-56db-4de7-964a-f8e6bf02b3a9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b241e359-1d6b-462c-889d-f462fa7f3421', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='809edf68-b253-4050-a04b-322beda39b64', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='406ec6af-5044-4a6a-b90c-31ad17b1c65a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a98ce973-ffac-431a-8485-582062ebf70c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6e30bddf-4665-4837-a0c6-5a9675729364', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4875271d-068c-4c12-aa72-a4cec6946292', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0be9737c-42f5-40c7-bb1a-88fd72beaa26', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e6f3c40c-88a8-4bc5-95ab-a123d0e95aa0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f1b13e31-14c2-4ab5-8c27-e488342d5dfc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='156b92f8-1424-4962-b09c-5bf9e3c925b9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8a622ce0-44e8-483b-852f-3ab520758fb0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a7e66d02-23c7-4f02-b6aa-4b64eed39b5c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='508aad55-621d-4032-b5f8-416fc15accc5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fa4d58f9-7458-4445-ad61-03b5ba2a15d7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cd2dc087-b7fe-4a3f-95ba-8f0a418b6566', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1d95103d-4a5c-46dc-8e6f-275471fcea0a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b31d93f7-6c3c-41f0-ba33-ba4518f1a02e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9317955b-b40f-4595-ab49-7fff74ef922d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1645f0eb-7047-4bd9-a0b0-32f048ca7cf0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c14c1826-adc1-4cb5-8746-9f989d002cfd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9ee61236-d00a-47ab-9e73-05f8e68dc6c9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3709abf8-bd01-4dd7-8909-c647a6a57419', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8e669081-233d-4da5-857b-1e006bf5a960', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1b551488-2004-4c3d-b389-9fa1ada36e23', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='575dc10d-1b22-476b-987c-1ec44072cfaa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f8bb83ec-f199-4f43-a569-120bb9071d93', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='116756d2-4eb9-4576-9ae1-2592a0d7dffa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ff0837f0-6d24-40f1-87fd-5e58533480c8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='392e671c-39b4-44a4-879a-a233b86f7056', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9d8b7fdc-d574-45ef-872f-45dc4e897384', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='27c0f535-6306-48b2-9622-0fe467a0e0ec', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='faffde58-2478-4d3d-954b-7479932a1321', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6796c806-a583-46b1-812d-dd0f60306ca1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8e79290a-1cfe-4b98-85d4-685acc76afc4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f37d6ae7-5c43-4e23-a1f9-f9eb402c06da', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='cdc067b0-da16-42ea-9b51-894d05789918', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6c3d5216-679a-4c02-95ab-a6170684931e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0f46837d-b92e-4e20-abb7-defc92b47955', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7ad7dd69-fe57-4d34-9490-88346276e5ed', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='398820bb-f3cf-4332-ac9b-adb451302c92', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dd02614a-144d-4fbc-80d4-42fb45dd0186', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c091f3db-a9c8-4f95-b42c-ba18ec9b842d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2971abc4-f79f-44d0-898c-2e2e509cca80', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c45fb49a-864d-41ef-97e1-b1db946c5e15', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c37265ff-2367-4ffc-9e2e-4b6b2e9d6fc2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='106b132f-067f-4d0a-a4dc-ad07fa0bf8f1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b9ce3848-542e-4a5f-baf4-de13cf8b2939', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5136f319-cc76-430e-a72f-54ad3f74fa18', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 1', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f145bb26-1d51-4ef9-82ea-9f9959e18b6c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d586bb29-459c-49f9-968e-1d66cd60d567', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Figure 1. Machine learning-guided optimization of ChRs.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2eb442fb-62c4-4f6a-b8e2-b18dab8de8ef', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0e789ba6-3b3d-43b2-bb4b-b76ea36dbaff', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0a31e94e-fa96-4e5a-8f45-eacf88e3644d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9de5d786-92f3-47cd-a5da-bbd82b5b9723', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bc50d223-d67b-497b-82f1-39e850fa5f23', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3f30f2a1-9613-4173-989d-af38f4a89ee1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='55a4bfec-eb8d-47b7-a747-7d44184314a3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='747c2e22-1cc5-4f9a-a71e-5a03e97e4afa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d78ff8f8-cc5b-4361-84ad-d00b8a896af8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4013167f-ebb7-4aba-b678-bd220289a6dc', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The figure contains several parts:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='26f4bcf5-53cc-47a3-b63d-6f10f52d73e3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f3e658f3-13a2-434e-a8a2-21ba5d31d3e0', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='682b3782-3575-4cd0-a61b-ee209bc543b2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bf14379b-8c58-4979-bf36-ddcbaf7e6fcb', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='710c6759-6aa6-486d-8416-354425d974d1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bf0a44d9-4aaa-4e31-9920-b822a7960340', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='c. Train regression models graphs for current strength, off-kinetics, and spectral properties.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c90c5cba-0876-46b0-8685-89c5600c1c72', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a5394cd9-c6fc-4deb-b87e-e8be3d516b0a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='331b22e1-5ad8-4bf0-91fa-99b86227c6f4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='aea915d8-751d-4f07-a1ce-245f855a7e35', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b0aba028-55f4-4519-a3c1-a892d8e44b42', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='19e1f9dd-9b58-4d77-8902-cb1090a2fa2d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c210e9e5-bb3c-4787-bc80-5b8739d61549', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='10c5d31f-fadb-43bf-9db0-17d1cdc70f56', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ffa54c33-089f-48ee-99a5-bd09276b25f2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='547de6b1-a425-4fc3-9399-451577867097', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 2', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b4e8c093-705b-45cc-b8c2-49d904c468ff', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='05b282d9-3026-492e-bdcc-d444974b1b85', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Figure 2.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ec519d37-ac8d-45cc-b5ef-6bf53fbba540', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f7817eea-34b6-4ccb-a56f-e5701adf1899', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='17af16db-636e-4041-8fe3-9d31e952aa49', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f9e33914-7fc1-457d-8865-e9ea9e95eea8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='00c11160-53c7-4bb6-a2a6-fe4409b857a9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4be63a51-8788-46a1-a70b-1fb463eebc4a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='d903c070-5feb-459b-973d-b69e8de48125', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3f2659f4-007f-488e-9790-e205e435b256', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3efa0829-5443-4f8d-b799-f4d38dd5c67a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(d) Peak and steady-state photocurrent strength', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='978a89b8-397b-4c72-88fc-1ee3ffba48aa', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7a46407d-6d1b-471a-858d-329d2d715f23', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The figure contains several panels:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b37bef2d-d540-480a-8d39-942ba341cd26', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ebaad2c0-6859-44bb-9d79-9fabcc3e1125', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='a) A series of current traces and corresponding cell images for different ChR variants.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='c628f43d-1b4a-4ec0-b289-a8e451d5fb8e', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='0954c2e9-31fb-42ce-8451-3b6236832e84', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='23c82828-12e1-4493-b173-fc338e3ddaf6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f588e89e-38b3-4234-b438-f6f43262389a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='422b188b-195c-45c8-9c24-f5feda7a9e8c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='5653ff23-56e3-4065-89c7-808aeb6fc196', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='d) Line graphs showing normalized photocurrent vs wavelength.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2e969435-a823-4fe6-9127-92b6f9fe65a4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e7f54701-1a5e-4658-b675-481e8289017c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='e) Line graphs showing peak and steady-state photocurrent vs light intensity.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a8bff709-0910-467f-a0c0-a1bf492590af', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='855397fa-f872-4307-8313-cd8caba1c12b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Each panel contains detailed data comparing various ChR variants across different properties and conditions.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='04c2b943-e553-4f72-9161-a563fb2a7c02', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7e2cee87-508b-4a63-a9b5-83cb656802f8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Figure 3.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8805bc94-da0c-40b6-8b1a-07a081f89ecb', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b6c73607-2155-4bcb-92ae-19413c841f36', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='143cd2ec-1635-415f-b472-59509edf1e3a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6a234cc1-9880-47fb-a048-c1686468e873', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7f834542-65d1-44a6-8c70-7c3acd2a9307', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f860182d-ca93-435d-999f-cbe55aaa64cd', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e2bc2c06-13a9-4e13-ada6-f8022bd94fff', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='f46f4ef0-c004-4023-a54a-87fa8ab19785', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4e3cb592-011f-4e80-b30b-a2d2e697a079', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='2c3b859c-ce6a-49a0-9bca-47c17fe586fe', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='01332574-2273-43a5-be6d-71c67603c453', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='9bd81bd2-94b8-44db-996d-e09a9aeb5c54', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='08ef712c-23e4-45c4-b51f-ef997bcac39c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7ca513c6-abc8-4d14-8d58-b79294469ec2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='7d3f56a9-8573-4617-9c5a-104af1b3a209', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='210426a7-071b-4853-a916-649aef5d2023', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='bb340ad2-4b15-4e6d-82d7-6e57b116bf3c', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e69a80f0-8b42-49d2-ad41-ebfd0116c1e9', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4510c607-5fcd-4bb3-a29b-ca03e6e80aa3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(i) Brain slice images showing systemic delivery (5x10^11 VG).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a07bfae2-4ea8-4e8e-a1d5-84f5908e51c3', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b87f836c-6bad-40a7-819d-18a7104a23db', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4c433c0b-6f22-4d73-b3bf-a884aed379d8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='dfda9a27-0d96-42ca-9ce3-0664c39f948d', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='3202c42b-66c3-4ce8-a447-e3edf3adcbaf', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='4b2f92ac-6ad7-44c1-9273-2afc925e273f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='96e8b975-e883-431b-9ecc-405fc36e1c02', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='e853ca84-dd80-479c-8183-7dcc72d6aaf6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='b3929c33-7caf-484a-911a-b756559a6b6a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='42e37449-294a-481b-bc67-8406979c6bb4', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8fd2eb18-17d9-4741-9de1-122acbbfe86a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ad60bad4-9792-4b83-9ec4-435eec41b083', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='## Figure 4.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='8f14298b-4132-4881-be3a-27ba661d9ba2', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='fa67dc68-1f07-4e5e-afc3-cda07665641f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ce95c67c-30fe-469d-9801-e5c43f6392ac', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='ec168c78-fc51-43b1-a681-f230261db6e7', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='29de4bfe-a79a-43ce-b080-aa7a426e105a', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='936da182-16cf-4888-a59f-92334fada7ca', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='58ceccc9-2e0c-425e-9043-970b26fb5813', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='43ea844c-59f4-4069-800c-a035d54a83e8', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The figure contains several panels illustrating the experimental setup and results:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='81024f73-b1e2-4d13-99e6-87e403687fc6', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='6ef4c204-60a8-4420-9901-1a9cb1d567da', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='a7a5db97-4f36-4a94-86ab-6afd729a462b', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='861e93d4-ab7b-4f7b-892b-15b9341f3ead', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='826331d8-1f3d-4420-b67e-6ce83c651ef5', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='d. Systemic delivery and fiber on skull surface diagrams for WT mice.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='93304d2f-04ae-4e11-9878-eb385ce07406', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='549b5000-936a-4854-b302-8e64f2531d4f', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='1844ad77-180f-4f74-8773-8ef2cc59b2da', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='g. Time-lapse images showing mouse movement in response to optical stimulation.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='519abda4-a438-420c-a446-def2eaeb79f1', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " Document(id_='04e90033-1c6c-4476-8c0f-7bf6b6595a10', embedding=None, metadata={'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf', 'file_name': 'nihms-1538039.pdf', 'file_type': 'application/pdf', 'file_size': 1991485, 'creation_date': '2024-09-20', 'last_modified_date': '2024-09-18'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, text='The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n')]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llama_parse import LlamaParse\n",
    "\n",
    "from plotreader.utils.document import _DEFAULT_PARSING_PROMPT\n",
    "\n",
    "\n",
    "parser = LlamaParse(\n",
    "    result_type=\"markdown\",\n",
    "    parsing_instruction=_DEFAULT_PARSING_PROMPT,\n",
    "    # use_vendor_multimodal_model=True,\n",
    "    # vendor_multimodal_model_name='anthropic-sonnet-3.5',\n",
    "    premium_mode=True,\n",
    "    split_by_page=True,\n",
    "    page_separator=\"\\n\",\n",
    "    take_screenshot=True,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started parsing the file under job_id 6f087e76-e0dd-470f-a0aa-d16fed3c55f2\n"
     ]
    }
   ],
   "source": [
    "json_objs = parser.get_json_result(\"/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(json_objs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'pages': [{'page': 1,\n",
       "    'text': '                                     HHS Public Access\\n                                     Author manuscript\\nAuthor Manuscript                    Nat Methods. Author manuscript; available in PMC 2020 April 14.\\n                       Published in final edited form as:\\n                         Nat Methods. 2019 November ; 16(11): 1176–1184. doi:10.1038/s41592-019-0583-8.\\n\\n\\n\\n                       Machine learning-guided channelrhodopsin engineering enables\\n                       minimally-invasive optogenetics\\n                       Claire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana\\n                       Gradinaru1,*, Frances H. Arnold1,2,*\\n                       1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena;\\nAuthor ManuscriptCalifornia; USA\\n                       2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena;\\n                       California; USA\\n\\n\\n\\n                       Abstract\\n                             We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity\\n                             enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the\\n                             mammalian brain require intracranial surgery for transgene delivery and implantation of invasive\\n                             fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate\\n                             expansive optogenetics without the need for invasive implants, our engineering approach leverages\\n                             the significant literature of ChR variants to train statistical models for the design of new, high-\\nAuthor Manuscriptperformance ChRs. With Gaussian Process models trained on a limited experimental set of 102\\n                             functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light\\n                             sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via\\n                             minimally-invasive systemic transgene delivery, not possible previously due to low per-cell\\n                             transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation\\n                             without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables\\n                             minimally-invasive optogenetics.\\n\\n\\n\\n                       Introduction\\n                                           Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae.\\n                                           Transgenic expression of ChRs in the brain enables light-dependent neuronal activation1.\\nAuthor Manuscript                          These channels are widely applied as tools in neuroscience research2; however, functional\\n\\n\\n\\n                       Users may view, print, copy, and download text and data-mine the content in such documents, for the purposes of academic research,\\n                       *subject always to the full Conditions of use:http://www.nature.com/authors/editorial_policies/license.html#terms\\n                       †K.K.Y. & J.E.R. contributed equally to this work.\\n                                                                         Corresponding: viviana@caltech.edu (V.G.) & fha@cheme.caltech.edu (F.H.A).\\n                       Author Contributions\\n                       C.N.B., K.K.Y., V.G., and F.H.A. conceptualized the project. C.N.B. coordinated all experiments and data analysis. C.N.B. and K.K.Y.\\n                       built machine-learning models. C.N.B. performed construct design and cloning. C.N.B. and E.D.M. performed AAV production.\\n                       E.D.M. prepared cultured neurons. C.N.B and J.E.R. conducted electrophysiology. C.N.B. and J.E.R. performed injections. J.E.R.\\n                       performed fiber cannula implants and behavioral experiments. C.N.B. performed all data analysis. C.N.B. wrote the manuscript with\\n                       input and editing from all authors. V.G. supervised optogenetics/electrophysiology, and F.H.A. supervised the protein engineering.\\n                       Competing interests\\n                       A provisional patent application (CIT File No.: CIT-8092-P) has been filed by Caltech based on these results. C.N.B., K.K.Y., V.G.,\\n                       and F.H.A. are inventors on this provisional patent.',\n",
       "    'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\\n\\nClaire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n## Abstract\\n\\nWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\\n\\n## Introduction\\n\\nChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "    'images': [{'name': 'img_p0_1.png',\n",
       "      'height': 102,\n",
       "      'width': 102,\n",
       "      'x': 61.5,\n",
       "      'y': 22.5,\n",
       "      'original_width': 152,\n",
       "      'original_height': 152},\n",
       "     {'name': 'page_1.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'heading',\n",
       "      'lvl': 1,\n",
       "      'value': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "      'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "      'bBox': {'x': 90, 'y': 142, 'w': 429.0159999999997, 'h': 14}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "      'md': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "      'bBox': {'x': 90, 'y': 233, 'w': 68.30000000000004, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Abstract',\n",
       "      'md': '## Abstract',\n",
       "      'bBox': {'x': 89, 'y': 295, 'w': 48.672, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "      'md': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "      'bBox': {'x': 90, 'y': 162, 'w': 394.5899999999998, 'h': 14}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Introduction',\n",
       "      'md': '## Introduction',\n",
       "      'bBox': {'x': 90, 'y': 520, 'w': 69.756, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "      'md': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "      'bBox': {'x': 90, 'y': 520, 'w': 345.41999999999985, 'h': 12}}]},\n",
       "   {'page': 2,\n",
       "    'text': '                   Bedbrook et al.                                                                                                   Page 2\\n\\n\\n\\nAuthor Manuscript                    limitations of available ChRs prohibit a number of optogenetic applications. These channels\\n                                    have broad activation spectra in the visible range and require high-intensity light for\\n                                    activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring\\n                                     approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to\\n                                     produce sufficient light-dependent depolarization to induce neuronal activation3. When\\n                                     applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target\\n                                     cell population to reliably activate action potentials4–6. This confines light-dependent\\n                                     activation to a small volume of brain tissue [~1 mm37. Enabling optogenetics for large brain]\\n                                     volumes without the need to implant invasive optical fibers for light delivery would be\\n                                     highly desirable for neuroscience applications.\\n\\n\\n\\n                                     Engineering ChRs to overcome limits in conductance and light sensitivity and extend the\\n                                     reach of optogenetic experiments requires overcoming three major challenges. First,\\nAuthor Manuscript                    rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the\\n                                     sequence and structural determinants of membrane protein expression and plasma-\\n                                     membrane localization are highly constrained and poorly understood8,9. Second, because\\n                                     properties of interest for neuroscience applications are assayed using low-throughput\\n                                     techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not\\n                                     feasible10. And third,in vivo applications require retention and optimization of multiple\\n                                     properties, for example, localization in mammalian cells while simultaneously tuning\\n                                     kinetics, photocurrents, and spectral properties6.\\n                                     Diverse ChRs have been published, including variants discovered from nature11,12, variants\\n                                     engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting\\n                                     from rational design16. Studies of these coupled with structural information17 and molecular\\nAuthor Manuscript                    dynamic simulations18 have established some understanding of the mechanics and sequence\\n                                     features important for specific ChR properties1,16. Despite this, it is still not possible to\\n                                     predict functional properties of new ChR sequences.\\n\\n\\n\\n                                     Our approach has been to leverage the significant literature of ChRs to train statistical\\n                                     models that enable design of highly-functional ChRs. These models take as their input the\\n                                     sequence and structural information for a given ChR variant and then predict its functional\\n                                     properties. The models use training data to learn how sequence and structural elements map\\n                                     to ChR function. Once known, that mapping can be used to predict the behavior of untested\\n                                     ChR variants.\\n\\n\\n\\n                                     We trained models in this manner and found that they accurately predict the properties of\\n                                     untested ChR sequences. We used these models to engineer 30 ‘designer’ ChR variants with\\nAuthor Manuscript                    specific combinations of desired properties, a number of which have unprecedented\\n                                     photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-\\n                                     photocurrent ChRs for applications in the mammalian brain and demonstrate their potential\\n                                     for minimally-invasive activation of populations of neurons in the brain enabled by systemic\\n                                     transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This\\n                                     work demonstrates how a machine learning-guided approach can enable engineering of\\n                                     proteins that have been challenging to engineer.\\n\\n\\n\\n                                            Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "    'images': [{'name': 'page_2.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "      'md': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 371.86999999999983, 'h': 13}}]},\n",
       "   {'page': 3,\n",
       "    'text': '                   Bedbrook et al.                                                                                                          Page 3\\n\\n\\n\\nAuthor ManuscriptResults\\n                       Functional characterization of ChR variants for machine learning\\n                                          In previous work, we explored structure-guided recombination20,21 of three highly-\\n                                          functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing\\n                                          two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310)\\n                                          chimeric variants9. These recombination libraries are a rich source of functionally diverse\\n                                          sequences9. Previously, we generated 102 ChR variants selected from these recombination\\n                                          libraries8,9, 76 of these were previously characterized for photocurrent properties (with\\n                                          patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR\\n                                          recombination variants provide the primary dataset used for model training. We\\n                                          supplemented this dataset with data from other published sources including 19 ChR variants\\nAuthor Manuscript                         from nature, 14 single-mutant ChR variants, and 28 recombination variants from other\\n                                          libraries (Dataset 1). As the data produced by other labs were not collected under the same\\n                                          experimental conditions as data collected in our hands, they cannot be used for comparison\\n                                          for absolute ChR properties (i.e. photocurrent strength); however, these data do provide\\n                                          useful binary information on whether a sequence variant is functional or not. Thus, we used\\n                                          published data from other sources when training binary classification models for ChR\\n                                          function.\\n\\n\\n\\n                                          Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-\\n                                          kinetics, we used these measured properties to train machine-learning models (Figure 1a).\\n                                          Enhancing ChR photocurrent strength would enable reliable neuronal activation even under\\n                                          low-light conditions. Different off-rates can be useful for specific applications: fast off-\\n                                          kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with\\nAuthor Manuscript                         increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant\\n                                          depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it\\n                                          is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for\\n                                          ChR function8.\\n\\n\\n\\n                       Training Gaussian process (GP) classification and regression models\\n                                          Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process\\n                                          (GP) classification and regression models (Figure 1). GP models successfully predicted\\n                                          thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and,\\n                                          more recently, ChR membrane localization8. For a detailed description of the GP model\\n                                          architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive\\n                                          values for new sequences from training examples by assuming that similar inputs (ChR\\nAuthor Manuscript                         sequence variants) will have similar outputs (photocurrent properties). To quantify the\\n                                          relatedness of inputs (ChR sequence variants), we compared both sequence and structure.\\n                                          ChR sequence information is encoded in the amino acid sequence. For structural\\n                                          comparisons, we convert the 3D crystal-structural information into a ‘contact map’ that is\\n                                          convenient for modeling. Two residues are considered to be in contact and potentially\\n                                          important for structural and functional integrity if they have any non-hydrogen atoms within\\n                                          4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\\n\\n\\n\\n                                                 Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"# Results\\n\\n## Functional characterization of ChR variants for machine learning\\n\\nIn previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\\n\\n## Training Gaussian process (GP) classification and regression models\\n\\nUsing the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "    'images': [{'name': 'page_3.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'heading',\n",
       "      'lvl': 1,\n",
       "      'value': 'Results',\n",
       "      'md': '# Results',\n",
       "      'bBox': {'x': 89, 'y': 75, 'w': 43.344, 'h': 12}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Functional characterization of ChR variants for machine learning',\n",
       "      'md': '## Functional characterization of ChR variants for machine learning',\n",
       "      'bBox': {'x': 90, 'y': 95, 'w': 307.2499999999999, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.',\n",
       "      'md': 'In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.',\n",
       "      'bBox': {'x': 150, 'y': 110, 'w': 370.22, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Training Gaussian process (GP) classification and regression models',\n",
       "      'md': '## Training Gaussian process (GP) classification and regression models',\n",
       "      'bBox': {'x': 90, 'y': 485, 'w': 329.47, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "      'md': \"Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "      'bBox': {'x': 150, 'y': 503, 'w': 371.8799999999999, 'h': 13}}]},\n",
       "   {'page': 4,\n",
       "    'text': '                   Bedbrook et al.                                                                                                        Page 4\\n\\n\\n\\nAuthor Manuscript                          similarity between two variants by aligning them and counting the number of positions and\\n                                           contacts at which they are identical24.\\n                                          We trained a binary classification model to predict if a ChR sequence will be functional\\n                                          using all 102 training sequences from the recombination library (Dataset 2) as well as data\\n                                          from 61 variants published by others (Dataset 1). We then used this trained classification\\n                                          model to predict whether uncharacterized ChR sequence variants were functional (Figure\\n                                          1b). To test prediction accuracy, we performed 20-fold cross validation on the training data\\n                                          set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good\\n                                          predictive power (Supplemental Table 1).\\n\\n\\n\\n                                          Next, we trained three regression models, one for each of the ChR photocurrent properties of\\n                                          interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics\\nAuthor Manuscript                         (Figure 1c). Once trained, these models were used to predict photocurrent properties of new,\\n                                          untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross\\n                                          validation on the training dataset and observed high correlation between predicted and\\n                                          measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental\\n                                          Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or\\n                                          C1Chrimson crystal structure26 perform as well as models built with a contact map from the\\n                                          C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps\\n                                           share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental\\n                                           Figure 1a,b).\\n\\n\\n\\n                       Selection of designer ChRs using trained models\\n                                           To select ‘designer’ ChRs (i.e. ChRs predicted to have a useful combination of properties),\\nAuthor Manuscript                          we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to\\n                                           not localize to the plasma membrane or predicted to be non-functional. To do this, we used\\n                                           classification models of ChR localization8 and function to predict the probability of\\n                                           localization and function for each ChR sequence in the 120,000-variant recombination\\n                                           library. Not surprisingly, most ChR variants were predicted to not localize and not function.\\n                                           To focus on ChR variants predicted to localize and function, we set a threshold for the\\n                                           product of the predicted probabilities of localization and function (Figure 1b); any ChR\\n                                           sequence above that threshold would be considered for the next tier of the process. We\\n                                           selected a conservative threshold of 0.4.\\n\\n\\n\\n                                           The training data made clear that the higher the mutation distance from one of the three\\n                                           parents, the less likely it was that a sequence would be functional; however, we expect that\\nAuthor Manuscript                          more diverse sequences would also have the more diverse functional properties. To explore\\n                                           diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4\\n                                           threshold and were multi-block-swap sequences containing on average 70 mutations from\\n                                           the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their\\n                                           photocurrent properties were measured with patch-clamp electrophysiology. 59% of the\\n                                           tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap\\n                                           sequences randomly selected (i.e., not selected by the model) and having comparable\\n                                           average mutation level. This validates the classification model’s ability to make useful\\n                                           predictions about novel functional sequences, even for sequences that are very distant from\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\\n\\n## Selection of designer ChRs using trained models\\n\\nTo select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "    'images': [{'name': 'page_4.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).',\n",
       "      'md': 'similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).',\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 371.5999999999999, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Selection of designer ChRs using trained models',\n",
       "      'md': '## Selection of designer ChRs using trained models',\n",
       "      'bBox': {'x': 90, 'y': 383, 'w': 233.3900000000001, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "      'md': \"To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "      'bBox': {'x': 150, 'y': 415, 'w': 366.18999999999977, 'h': 13}}]},\n",
       "   {'page': 5,\n",
       "    'text': '                   Bedbrook et al.                                                                                                           Page 5\\n\\n\\n\\nAuthor Manuscript                          those previously tested. We updated the models by including data from these 22 sequences\\n                                           for future rounds of predictions.\\n\\n\\n\\n                                           From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed\\n                                           the conservative 0.4 predicted localization and function threshold (Figure 1). For the second\\n                                           tier of the selection process, we used the three regression models trained on all functional\\n                                           variants collected up to this point to predict the photocurrent strength, wavelength sensitivity\\n                                           of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3).\\n                                           We selected 28 designer ChRs predicted to be highly functional with different combinations\\n                                           of properties including those predicted to have the highest photocurrent strength, most red-\\n                                           shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow\\n                                           (Supplemental Figure 2–3).\\n\\n\\n\\nAuthor Manuscript                          Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK\\n                                           cells, and characterized for their photocurrent properties with patch-clamp\\n                                           electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected\\n                                           using the updated classification model above the 0.4 threshold both localize and function.\\n                                           For each of the designer ChR variants, the measured photocurrent properties correlated well\\n                                           with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This\\n                                           outstanding performance on a novel set of sequences demonstrates the power of this data-\\n                                           driven predictive method for engineering designer ChRs. As a negative control, we selected\\n                                           two ChR variant sequences from the recombination library that the model predicted would\\n                                           be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-\\n                                           block swap from two of the most highly functional ChR recombination variants tested. As\\n                                           predicted, these sequences were non-functional (Figure 2b), which shows that ChR\\nAuthor Manuscript                          functionality can be attenuated by incorporating even minimal diversity at certain positions.\\n                       Sequence and structural determinants of ChR functional properties\\n                                           We used L1-regularized linear regression models to identify a limited set of residues and\\n                                           structural contacts that strongly influence ChR photocurrent strength, spectral properties,\\n                                           and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of\\n                                           these sequence and structural features by weighting their contributions using L2-regularized\\n                                           linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we\\n                                           identified a set of important residues and contacts and their respective weights. A specific\\n                                           residue or contact at a given position is weighted as likely to lead to, e.g., low (negative\\n                                           weight) or high (positive weight) photocurrents. A number of residues and contacts most\\n                                           important for tuning spectral properties are proximal to the retinal-binding pocket, including\\nAuthor Manuscript                          the blue-shifting contact between A206 and F269 and the blue-shifting contact between\\n                                           F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the\\n                                           red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR\\n                                           parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-\\n                                           kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental\\n                                           Figure 4), a residue that is part of the DC-gate1. Mutation of either the aspartic acid or\\n                                           cysteine within the DC-gate has been shown to decrease off-kinetic speed14,27. While the\\n                                           cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\\n\\n## Sequence and structural determinants of ChR functional properties\\n\\nWe used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "    'images': [{'name': 'page_5.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.',\n",
       "      'md': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.',\n",
       "      'bBox': {'x': 150, 'y': 71, 'w': 370.5199999999998, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Sequence and structural determinants of ChR functional properties',\n",
       "      'md': '## Sequence and structural determinants of ChR functional properties',\n",
       "      'bBox': {'x': 90, 'y': 439, 'w': 320.06, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "      'md': 'We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "      'bBox': {'x': 150, 'y': 457, 'w': 369.2999999999998, 'h': 10}}]},\n",
       "   {'page': 6,\n",
       "    'text': '                   Bedbrook et al.                                                                                                          Page 6\\n\\n\\n\\nAuthor Manuscript                         195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at\\n                                          that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly\\n                                          to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal\\n                                          contribute strongly to photocurrent strength. For example, the most heavily weighted contact\\n                                          includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that\\n                                          covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and\\n                                          CheRiff.\\n\\n\\n\\n                      Machine-guided search identifies ChRs with a range of useful functional properties\\n                                          We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer\\n                                          ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-\\n                                          predicted ChRs, we also assessed the two top-performing ChRs from the classification\\nAuthor Manuscript                         models’ predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-\\n                                          predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the\\n                                          30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated\\n                                          photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants\\n                                          exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants\\n                                          have larger red-light activated photocurrents when compared with the blue-light activated\\n                                          parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants\\n                                          predicted to be non-functional by the models produce <30 pA currents.\\n\\n\\n\\n                                          Designer ChRs’ off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure\\n                                          2c). This range is quite remarkable given that all designer ChRs are built from sequence\\n                                          blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5\\n                                          designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower\\nAuthor Manuscript                         off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms\\n                                          and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics\\n                                          with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5\\n                                          s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and\\n                                          ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\n\\n\\n                                          Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure\\n                                          5). ChR_28_10’s red-shifted spectrum matches that of CsChrimR, demonstrating that\\n                                          incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a\\n                                          red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties:\\n                                          ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar\\n                                          steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA)\\nAuthor Manuscript                         when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow\\n                                          activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\n\\n\\n                                          We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR,\\n                                          CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity\\n                                          of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal\\n                                          decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm\\n                                          −2), suggesting that photocurrents were saturated at these intensities and would only\\n                                          attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\\n\\n\\n\\n                                                 Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\\n\\n## Machine-guided search identifies ChRs with a range of useful functional properties\\n\\nWe assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "    'images': [{'name': 'page_6.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': '195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.',\n",
       "      'md': '195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.',\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 370.64999999999986, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Machine-guided search identifies ChRs with a range of useful functional properties',\n",
       "      'md': '## Machine-guided search identifies ChRs with a range of useful functional properties',\n",
       "      'bBox': {'x': 89, 'y': 181, 'w': 394.2299999999999, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "      'md': \"We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "      'bBox': {'x': 149, 'y': 199, 'w': 371.80999999999995, 'h': 13}}]},\n",
       "   {'page': 7,\n",
       "    'text': '                  Bedbrook et al.                                                                                                           Page 7\\n\\n\\n\\nAuthor Manuscript                        expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating\\n                                         that the improved photocurrent strength of these ChRs is not solely due to improved\\n                                         expression (Supplemental Figure 6–7).\\n\\n\\n\\n                                         We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with\\n                                         ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used forin\\n                                         vivo optogenetics, and CoChR (fromChloromonas oogama)11, reported to be one of the\\n                                         highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–\\n                                         6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2)\\n                                         481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-\\n                                         intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR\\n                                         produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much\\n                                         lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x\\nAuthor Manuscript                        larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x\\n                                         larger steady-state photocurrents than CoChR when exposed to low-intensity light\\n                                         (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of\\n                                         these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to\\n                                         the increased accumulation of the open state under low-light conditions14.\\n\\n\\n\\n                      Validation of designer ChRs for neuroscience applications\\n                                         For further validation we selected three of the top high-conductance ChRs, ChR_9_4,\\n                                         ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3,\\n                                         respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin\\n                                         (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust\\n                                         membrane localization and expression throughout the neuron soma and neurites (Figure 3b).\\nAuthor Manuscript                        The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-\\n                                         intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of\\n                                         magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation\\n                                         (Figure 3d, Supplemental Figure 8h).\\n\\n\\n\\n                                         Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of\\n                                         rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter\\n                                         (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in\\n                                         ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the\\n                                         ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA\\n                                         photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits\\n                                         undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW\\nAuthor Manuscript                        mm−2 light, a four-fold improvement over ChR2(H134R)’s irradiance-matched\\n                                         photocurrents (Figure 3g). Our characterization of ChR2(H134R)’s light sensitivity and\\n                                         photocurrent strength is consistent with previous results from other labs6,22.\\n\\n\\n\\n                      Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n                                         We investigated whether these light-sensitive, high-photocurrent ChRs could provide\\n                                         optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of\\n                                         ‘non-invasive optogenetics’ relied on invasive intracranial virus delivery, which results in\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\\n\\n## Validation of designer ChRs for neuroscience applications\\n\\nFor further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\\n\\n## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n\\nWe investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "    'images': [{'name': 'page_7.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.',\n",
       "      'md': 'expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.',\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 371.9, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Validation of designer ChRs for neuroscience applications',\n",
       "      'md': '## Validation of designer ChRs for neuroscience applications',\n",
       "      'bBox': {'x': 89, 'y': 331, 'w': 276.85000000000014, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\",\n",
       "      'md': \"For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\",\n",
       "      'bBox': {'x': 149, 'y': 137, 'w': 371.1199999999997, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation',\n",
       "      'md': '## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation',\n",
       "      'bBox': {'x': 89, 'y': 637, 'w': 400.65000000000003, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "      'md': \"We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "      'bBox': {'x': 149, 'y': 655, 'w': 358.21999999999986, 'h': 10}}]},\n",
       "   {'page': 8,\n",
       "    'text': '                   Bedbrook et al.                                                                                                  Page 8\\n\\n\\n\\nAuthor Manuscript                   many copies of virus per cell and thus very high expression levels of the injected\\n                                    construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad\\n                                    transduction throughout the central nervous system with a single minimally-invasive\\n                                    intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in\\n                                    brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-\\n                                    PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of\\n                                    infection with systemically delivered viral vectors resulting in insufficient opsin expression\\n                                    and light-evoked currents to control neuronal firing with commonly-used channels (e.g.\\n                                    ChR2).\\n\\n\\n\\n                                    We hypothesized that the ChRgers could overcome this limitation and allow large-volume\\n                                    optogenetic excitation following systemic transgene delivery. We systemically delivered\\n                                    rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the\\nAuthor Manuscript                   hSyn promoter and observed broad expression throughout the brain (Figure 3i). We\\n                                    measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for\\n                                    light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced\\n                                    light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing\\n                                    neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced\\n                                    activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared\\n                                    with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l).\\n                                    ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or\\n                                    ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10;\\n                                    Supplemental Table 6). These results demonstrate the need for light-sensitive and high-\\n                                    photocurrent opsins for applications where systemic delivery is desired.\\n\\n\\n\\nAuthor Manuscript                   We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa\\n                                    promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to\\n                                    results observed after direct injection into the PFC (Figure 3g). When expressed in\\n                                    pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced\\n                                    firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency\\n                                    stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while\\n                                    ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher\\n                                    frequency stimulation (20–40 Hz) (Figure 3m).\\n\\n\\n\\n                                    We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using\\n                                    optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral\\n                                    tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-\\nAuthor Manuscript                   floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into\\n                                    Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery\\n                                    and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an\\n                                    operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose\\n                                    poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a\\n                                    frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz)\\n                                    and higher light power (up to 10 mW) promoted greater maximum operant response rates\\n                                    (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in\\n                                    ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "    'images': [{'name': 'page_8.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "      'md': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 370.2699999999998, 'h': 13}}]},\n",
       "   {'page': 9,\n",
       "    'text': '                   Bedbrook et al.                                                                                                  Page 9\\n\\n\\n\\nAuthor Manuscript                   acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy\\n                                    number produced by systemic delivery for robust neuronal activation.\\n\\n\\n\\n                                    In order to determine if ChRger2 would enable both minimally-invasive transgene delivery\\n                                    and minimally-invasive optical excitation, we assayed directional control of locomotion in\\n                                    freely moving animals by optogenetic stimulation of the right secondary motor cortex\\n                                    (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the\\n                                    contralateral lower extremities, causing mice to turn away from the stimulation side. We\\n                                    systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under\\n                                    a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex\\n                                    (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex\\n                                    for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We\\n                                    secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without\\nAuthor Manuscript                   puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to\\n                                    be minimally invasive. Despite the presence of the highly optically scattering calavarial\\n                                    bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals\\n                                    expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and\\n                                    Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical\\n                                    stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10\\n                                    mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by\\n                                    the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside\\n                                    the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both\\n                                    ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is\\n                                    compatible with minimally-invasive systemic gene delivery and can enable minimally-\\n                                    invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion\\nAuthor Manuscript                   nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic\\n                                    transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\\n\\n\\n\\n                      Discussion\\n                                    We demonstrated a data-driven approach to engineering ChR properties that enables efficient\\n                                    discovery of highly functional ChR variants based on data from relatively few variants. In\\n                                    this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric\\n                                    ChRs and use it to efficiently search sequence space and select top-performing variants for a\\n                                    given property10,24,34. By first eliminating the vast majority of non-functional sequences, we\\n                                    can focus on local peaks scattered throughout the landscape. Then, using regression models,\\n                                    we predict which sequences lie on the fitness peaks.\\nAuthor Manuscript                   Machine learning provides a platform for simultaneous optimization of multiple ChR\\n                                    properties that follow engineering specifications. We were able to generate variants with\\n                                    large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the\\n                                    parental or other commonly used ChRs. We also use the machine-learning models to identify\\n                                    the residues and contacts most important for ChR function. Application of this machine-\\n                                    learning pipeline (limited data collection from diverse sequences, model training and\\n                                    validation, and prediction and testing of new sequences) has great potential to optimize other\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\\n\\n## Discussion\\n\\nWe demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "    'images': [{'name': 'page_9.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.',\n",
       "      'md': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.',\n",
       "      'bBox': {'x': 150, 'y': 71, 'w': 371.3699999999998, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Discussion',\n",
       "      'md': '## Discussion',\n",
       "      'bBox': {'x': 89, 'y': 461, 'w': 64.01999999999998, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': 'We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "      'md': 'We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "      'bBox': {'x': 149, 'y': 481, 'w': 371.9299999999996, 'h': 13}}]},\n",
       "   {'page': 10,\n",
       "    'text': '                  Bedbrook et al.                                                                                                                 Page 10\\n\\n\\n\\nAuthor Manuscript                        neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and\\n                                         AAVs30.\\n                                         We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and\\n                                         validated ChRger2’s application forin vivo optogenetics. The high-photocurrent properties\\n                                        of these ChRs overcome the limitation of low per-cell copy number after systemic delivery.\\n                                        ChRger2 enabled neuronal excitation with high temporal precision without invasive\\n                                        intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas,\\n                                        extending what is currently possible for optogenetics experiments.\\n\\n\\n\\n                      Online methods\\n                      Construct design and cloning\\nAuthor Manuscript                       The design, construction, and characterization of the recombination library of chimeras is\\n                                         described in detail in Bedbrooket al.9. The 10-block contiguous and 10-block\\n                                        noncontiguous recombination libraries were designed and built using SCHEMA\\n                                        recombination9. Software packages for calculating SCHEMA energies are openly available\\n                                        at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these\\n                                        libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217,\\n                                        and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9\\n                                        blocks of one parent and a single block from one of the other two parents) and multi-block-\\n                                        swap chimera sequences.\\n\\n\\n\\n                                        Selected ChR variant genes were inserted into a constant vector backbone [pFCK from\\n                                        Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS)\\nAuthor Manuscript                       sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR\\n                                        variants contain the SpyTag sequence following the N-terminal signal peptide for the\\n                                        SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36.\\n                                        The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR\\n                                        variant sequences used in this study are documented in Dataset 2. All selected ChR genes\\n                                        were synthesized and cloned in the pFCK mammalian expression vector by Twist\\n                                        Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were\\n                                        created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n                                        For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3,\\n                                        CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene\\n                                        plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a\\n                                        pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR\\nAuthor Manuscript                       was inserted with a TS sequence5 and fluorescent protein (eYFP).\\n\\n\\n\\n                      HEK293T cell and primary neuronal cultures\\n                                        The culturing and characterization ChRs in HEK cells is described in Bedbrook                     et al.9,36.\\n                                        Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with\\n                                        10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were\\n                                        transfected with purified ChR variant DNA using FuGENE®6 reagent according to the\\n                                        manufacturer’s (Promega) recommendations. Cells were given 48 hours to express the ChRs\\n\\n\\n\\n                                               Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\\n\\n## Online methods\\n\\n### Construct design and cloning\\n\\nThe design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\\n\\n### HEK293T cell and primary neuronal cultures\\n\\nThe culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "    'images': [{'name': 'page_10.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\",\n",
       "      'md': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\",\n",
       "      'bBox': {'x': 149, 'y': 68, 'w': 368.50999999999976, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Online methods',\n",
       "      'md': '## Online methods',\n",
       "      'bBox': {'x': 89, 'y': 209, 'w': 90.68399999999997, 'h': 12}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 3,\n",
       "      'value': 'Construct design and cloning',\n",
       "      'md': '### Construct design and cloning',\n",
       "      'bBox': {'x': 89, 'y': 229, 'w': 140.93, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).',\n",
       "      'md': 'The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).',\n",
       "      'bBox': {'x': 78, 'y': 45, 'w': 370.03999999999996, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 3,\n",
       "      'value': 'HEK293T cell and primary neuronal cultures',\n",
       "      'md': '### HEK293T cell and primary neuronal cultures',\n",
       "      'bBox': {'x': 89, 'y': 615, 'w': 209.97000000000014, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "      'md': \"The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "      'bBox': {'x': 78, 'y': 45, 'w': 365.77, 'h': 13.514623181074967}}]},\n",
       "   {'page': 11,\n",
       "    'text': '                  Bedbrook et al.                                                                                                      Page 11\\n\\n\\n\\nAuthor Manuscript                        before photocurrent measurements. Primary hippocampal neuronal cultures were prepped\\n                                         from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River\\n                                         Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented\\n                                         with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB\\n                                         packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings\\n                                         were performed 5–10 days after transduction.\\n\\n\\n\\n                      Patch-clamp electrophysiology\\n                                         Whole-cell patch-clamp and cell-attached recordings were performed in transfected HEK\\n                                         cells, transduced cultured neurons, and acute brain slices to measure light-activated inward\\n                                         currents or neuronal firing. For electrophysiological recordings, cultured cells were\\n                                         continuously perfused with extracellular solution at room temperature (in mM: 140 NaCl, 5\\nAuthor Manuscript                        KCl, 10 HEPES, 2 MgCl2, 2 CaCl2, 10 glucose; pH 7.35) while mounted on the microscope\\n                                         stage. For slice recordings, 32 °C artificial cerebrospinal fluid (ACSF) was continuously\\n                                         perfused over slices. ACSF contained 127 mM NaCl, 2.5 mM KCl, 25 mM NaHCO3, 1.25\\n                                         mM NaH2PO4, 12 mMd-glucose, 0.4 mM sodium ascorbate, 2 mM CaCl2, and 1 mM\\n                                         MgCl2 and was bubbled continuously with 95% oxygen / 5% CO2. Firing and photocurrent\\n                                         measurements were performed in the presence of 3 mM kynurenic acid and 100 μM\\n                                         picrotoxin to block optically evoked ionotropic glutamatergic and GABAergic currents,\\n                                         respectively.\\n\\n\\n\\n                                         Patch pipettes were fabricated from borosilicate capillary glass tubing (1B150–4; World\\n                                         Precision Instruments) using a model P-2000 laser puller (Sutter Instruments) to resistances\\n                                         of 3–6 MΩ. Pipettes were filled with K-gluconate intracellular solution containing the\\n                                         following (in mM): 134 K gluconate, 5 EGTA, 10 HEPES, 2 MgCl2, 0.5 CaCl2, 3 ATP, and\\nAuthor Manuscript                        0.2 GTP. Whole-cell patch-clamp and cell-attached recordings were made using a\\n                                         Multiclamp 700B amplifier (Molecular Devices), a Digidata 1440 digitizer (Molecular\\n                                         Devices), and a PC running pClamp (version 10.4) software (Molecular Devices) to generate\\n                                         current injection waveforms and to record voltage and current traces.\\n\\n\\n\\n                                         Photocurrents were recorded from cells in voltage clamp held at −60 mV. Neuronal firing\\n                                         was measured in current clamp mode with current injection for a −60 mV holding potential.\\n                                         Access resistance (Ra) and membrane resistance (Rm) were monitored throughout recording,\\n                                         and cells were discarded if Ra or Rm changed more than 15%. During ChR variant\\n                                         functional screening in HEK cells, photocurrents were only recorded from cells that passed\\n                                         our recording criteria: Rm > 200 MΩ and holding current >−100 pA. Our measured\\n                                         membrane properties of ChR expressing neurons were consistent with previous literature of\\nAuthor Manuscript                        opsin-expressing cells11 and are also consistent with previous reports of properties of\\n                                         cultured hippocampal neurons38,39 and PFC neurons in slice40,41 (Supplemental Figures 10).\\n                                         For cell culture experiments, the experimenter was blinded to the identity of the ChR being\\n                                         patched but not to the fluorescence level of the cells. For acute slice recordings, the\\n                                         experimenter was not blinded to the identity of the ChR.\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'NO_CONTENT_HERE',\n",
       "    'images': [{'name': 'page_11.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'NO_CONTENT_HERE',\n",
       "      'md': 'NO_CONTENT_HERE',\n",
       "      'bBox': {'x': 0, 'y': 0, 'w': 612}}]},\n",
       "   {'page': 12,\n",
       "    'text': '                  Bedbrook et al.                                                                                                      Page 12\\n\\n\\n\\nAuthor ManuscriptLight delivery and imaging\\n                                         Patch-clamp recordings were done with short light pulses to measure photocurrents. Light\\n                                         pulse duration, wavelength, and power were varied depending on the experiment (as\\n                                         described in the text). Light pulses were generated using a Lumencor SPECTRAX light\\n                                         engine. The illumination/output spectra for each color were measured (Supplemental Figure\\n                                         5). To evaluate normalized green photocurrent, we measured photocurrent strength at three\\n                                         wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm,\\n                                         and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these\\n                                         measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640\\n                                         nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we\\n                                         measured photocurrents at seven different wavelengths (peak ± half width half maximum):\\n                                         (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481\\nAuthor Manuscript                        ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for\\n                                         each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n                                         Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5\\n                                         sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR\\n                                         expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880\\n                                         confocal microscope and Zen software.\\n\\n\\n\\n                      Electrophysiology data analysis\\n                                         Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and\\n                                         custom data-processing scripts written using open-source packages in the Python\\n                                         programming language to perform baseline adjustments, find the peak and steady state\\nAuthor Manuscript                        inward currents, perform monoexponential fits of photocurrent decay for off-kinetic\\n                                         properties, and quantify spike fidelity. Only neurons with an uncompensated series\\n                                         resistance between 5 and 25 MΩ,Rm > 90 MΩ, and holding current >−150 pA (holding at\\n                                         −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent\\n                                         amplitude was not adjusted for expression level since both expression and conductance\\n                                         contribute to thein vivo utility of the tool. However, comparisons of expression with\\n                                         photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7.\\n                                         As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure\\n                                         1a). As a metric for the ChR activation spectrum, the normalized current strength induced by\\n                                         exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to\\n                                         characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the\\n                                         photocurrent decay rate, τoff (Figure 1a).\\nAuthor ManuscriptAAV production and purification\\n                                         Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE,\\n                                         pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X =\\n                                         ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods\\n                                         described in Devermanet al.42 and Challiset al.31. Briefly, triple transfection of HEK293T\\n                                         cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested\\n                                         from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556)\\n                                         step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': '## Light delivery and imaging\\n\\nPatch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\\n\\n## Electrophysiology data analysis\\n\\nElectrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\\n\\n## AAV production and purification\\n\\nProduction of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "    'images': [{'name': 'page_12.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Light delivery and imaging',\n",
       "      'md': '## Light delivery and imaging',\n",
       "      'bBox': {'x': 90, 'y': 71, 'w': 126.70000000000002, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.',\n",
       "      'md': 'Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.',\n",
       "      'bBox': {'x': 149, 'y': 89, 'w': 369.5999999999999, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Electrophysiology data analysis',\n",
       "      'md': '## Electrophysiology data analysis',\n",
       "      'bBox': {'x': 89, 'y': 349, 'w': 151.73, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).',\n",
       "      'md': 'Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).',\n",
       "      'bBox': {'x': 149, 'y': 367, 'w': 371.0999999999998, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'AAV production and purification',\n",
       "      'md': '## AAV production and purification',\n",
       "      'bBox': {'x': 89, 'y': 589, 'w': 152.90000000000003, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "      'md': 'Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "      'bBox': {'x': 149, 'y': 607, 'w': 369.63999999999976, 'h': 10}}]},\n",
       "   {'page': 13,\n",
       "    'text': '                  Bedbrook et al.                                                                                                          Page 13\\n\\n\\n\\nAuthor Manuscript                        phosphate buffered saline (PBS). Virus titers were determined by measuring the number of\\n                                         DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\\n\\n\\n\\n                      Animals\\n                                         All procedures were approved by the California Institute of Technology Institutional Animal\\n                                         Care and Use Committee (IACUC).Dat-Cre mice (006660) and C57Bl/6J mice (000664)\\n                                         were purchased from Jackson Laboratory.\\n\\n\\n\\n                      Intravenous injections, stereotactic injections, and cannula implantation\\n                                         Intravenous administration of rAAV vectors was performed by injecting the virus into the\\n                                         retro-orbital sinus at viral titers indicated in the text. There were no observed health issues\\n                                         with animals after systemic injection of virus at the titers presented in the paper. Mice\\nAuthor Manuscript                        remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice\\n                                         electrophysiology, there was no observed indication of poor cell health due to viral-mediated\\n                                         expression, which was quantified by measuring the membrane resistance [Rm], leak current\\n                                         [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local\\n                                         expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl\\n                                         of purified AAV vectors at 5×1012 vg ml−1 targeting the following coordinates: anterior-\\n                                         posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For\\n                                         stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses,\\n                                         MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the\\n                                         VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV,\\n                                         4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm\\n                                         mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were\\nAuthor Manuscript                        surgically secured to the surface of the skull above M2 (unilaterally) targeted to the\\n                                         following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a\\n                                         standard drill to create a level surface for the fiber-skull interface. Light was delivered from\\n                                         either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-\\n                                         LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-\\n                                         ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull\\n                                         with Metabond (Parkel, SKU S396) and dental cement.\\n\\n\\n\\n                                         Analysis of behavioral experiments was performed using the open-source MATLAB\\n                                         program OptiMouse43 to track mouse nose, body, and tail position while the mouse was\\n                                         running on the treadmill. Optogenetic intracranial self-stimulation was performed using a\\n                                         mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR\\n                                         nose port (Model 80116TM).\\nAuthor Manuscript\\n                      Gaussian process modeling\\n                                         Both the GP regression and classification modeling methods applied in this paper are based\\n                                         on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple\\n                                         Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/\\n                                         muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2\\n                                         sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high\\n                                         sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\\n\\n## Animals\\n\\nAll procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\\n\\n## Intravenous injections, stereotactic injections, and cannula implantation\\n\\nIntravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\\n\\n## Gaussian process modeling\\n\\nBoth the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "    'images': [{'name': 'page_13.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.',\n",
       "      'md': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.',\n",
       "      'bBox': {'x': 150, 'y': 71, 'w': 368.00999999999976, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Animals',\n",
       "      'md': '## Animals',\n",
       "      'bBox': {'x': 90, 'y': 111, 'w': 38.900000000000006, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.',\n",
       "      'md': 'All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.',\n",
       "      'bBox': {'x': 150, 'y': 129, 'w': 369.77999999999975, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Intravenous injections, stereotactic injections, and cannula implantation',\n",
       "      'md': '## Intravenous injections, stereotactic injections, and cannula implantation',\n",
       "      'bBox': {'x': 90, 'y': 183, 'w': 343.18, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).',\n",
       "      'md': 'Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).',\n",
       "      'bBox': {'x': 90,\n",
       "       'y': 111,\n",
       "       'w': 371.3499999999996,\n",
       "       'h': 10.514623181075024}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Gaussian process modeling',\n",
       "      'md': '## Gaussian process modeling',\n",
       "      'bBox': {'x': 89, 'y': 601, 'w': 133.18000000000006, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "      'md': 'Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "      'bBox': {'x': 149, 'y': 619, 'w': 370.3499999999998, 'h': 10}}]},\n",
       "   {'page': 14,\n",
       "    'text': '                   Bedbrook et al.                                                                                                     Page 14\\n\\n\\n\\nAuthor Manuscript                   the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact\\n                                    architecture observed in the C1C2 crystal structure. Models built using structural encodings\\n                                    built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb)\\n                                    performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The\\n                                    models are robust to differences in contact maps because they use both sequence and\\n                                    structural information, which is somewhat redundant.\\n\\n\\n\\n                                    For a given ChR, the contact map is simply a list of contacting amino acids with their\\n                                    positions. For example, a contact between alanine at position 134 and methionine at position\\n                                    1 of the amino acid sequence would be encoded by [(‘A134’), (‘M1’)]. Both sequence and\\n                                    structural information were one-hot encoded. Regression models for ChR properties were\\n                                    trained to predict the logarithm of the measured properties. All training data was normalized\\n                                    to have mean zero and standard deviation one.\\nAuthor Manuscript\\n                                    Gaussian process regression and classification models require kernel functions that measure\\n                                    the similarity between protein sequences. Learning involves optimizing the form of the\\n                                    kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be\\n                                    optimal for all ChR properties (Supplemental Table 1).\\n\\n\\n\\n                                    For classification model training, all 102 functionally characterized ChR variants from our\\n                                    recombination libraries (Dataset 2) were used as well as data from 61 sequence variants\\n                                    published by others (Dataset 1). The model was then updated with data collected from the\\n                                    22 additional ChR recombination variants with high sequence diversity (~70 mutations from\\n                                    the closest parent) and predicted to be functional (Figure 1d). For training the regression\\n                                    models, all 102 functionally characterized training sequences (Dataset 2) were initially used\\nAuthor Manuscript                   and then the models were updated with data collected from the 22 additional ChR variants\\n                                    (Figure 1d).\\n\\n\\n\\n                                    GP regression—In regression, the goal is to infer the value of an unknown functionf(x) at\\n                                    a novel point x*given observationsy at inputsX. Assuming that the observations are subject\\n                                    to independent and identically distributed Gaussian noise with variance σn2, the posterior\\n                                    distribution of f * = f x* for Gaussian process regression is Gaussian with mean\\n\\n\\n\\n                                                                                   f * = k*T K + σn2I −1y                                    (1)\\n\\n\\n\\n                                    and variance\\n\\n\\n\\nAuthor Manuscript                                                         v* = k x*, x* − k*T K + σn2I −1k*                                  (2)\\n\\n\\n\\n                                    WhereK is the symmetric, square covariance matrix for the training set:Kij =k(x,x) forxii        j\\n                                    andxj in the training set. k*is the vector of covariances between the novel input and each\\n                                    input in the training set, and k*i = k x*, xi . The hyperparameters in the kernel functions and\\n                                    the noise hyperparameterσn were determined by maximizing the log marginal likelihood:\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\\n\\n## GP regression\\n\\nIn regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\",\n",
       "    'images': [{'name': 'page_14.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\",\n",
       "      'md': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\",\n",
       "      'bBox': {'x': 149, 'y': 71, 'w': 369.9499999999997, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'GP regression',\n",
       "      'md': '## GP regression',\n",
       "      'bBox': {'x': 477, 'y': 619, 'w': 2.224, 'h': 8.411698544860009}},\n",
       "     {'type': 'text',\n",
       "      'value': 'In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:',\n",
       "      'md': 'In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:',\n",
       "      'bBox': {'x': 149, 'y': 479, 'w': 352.27, 'h': 14}}]},\n",
       "   {'page': 15,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                                                                                                                                                                                                 Page 15\\n\\n\\n\\nAuthor Manuscript                                                                                                                                                                             1yT K + σn2I −1y − 2 1log K + σn2I − 2 nlog 2π\\n                                                                                                                    log p y|X = − 2                                                                                                                                                                                                         (3)\\n                                      where nis the dimensionality of the inputs. Regression was implemented using open-source\\n                                      packages in the SciPy ecosystem44–46.\\n\\n\\n\\n                                    GP classification—In binary classification, instead of continuous outputsy, the outputs\\n                                    are class labelsyi ∈ {+1, −1}, and the goal is to use the training data to make probabilistic\\n                                      predictions π x* = p y* = + 1 x* . We use Laplace’s method to approximate the posterior\\n                                    distribution. Hyperparameters in the kernels are found by maximizing the marginal\\n                                    likelihood. Classification was implemented using open-source packages in the SciPy\\n                                    ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is\\nAuthor Manuscript                   or is not functional. A ChR sequence was considered to be functional if its photocurrents\\n                                    were >100 pA upon light exposure, a threshold set as an approximate lower bound for\\n                                    current necessary for neuronal activation.\\n\\n\\n\\n                                    GP kernels for modeling proteins—Gaussian process regression and classification\\n                                    models require kernel functions that measure the similarity between protein sequences. A\\n                                    protein sequences of lengthL is defined by the amino acid present at each location. This can\\n                                    be encoded as a binary feature vector                                                                                                                                                xse that indicates the presence or absence of each\\n                                      amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids).\\n                                      Likewise, the protein’s structure can be represented as a residue-residue contact map. The\\n                                      contact map can be encoded as a binary feature vectorxst that indicates the presence or\\n                                    absence of each possible contacting pair. Both the sequence and structure feature vectors\\nAuthor Manuscript                   were used by concatenating them to form a sequence-structure feature vector.\\n                                    Three types of kernel functions                                                                                                                        k(s,s) were considered: polynomial kernels, squaredij\\n                                    exponential kernels, and Matérn kernels. These different forms represent possible functions\\n                                    for the protein’s fitness landscape. The polynomial kernel is defined as:\\n\\n\\n\\n                                                                                                                                                                                                   k s, s′ =                                                       σ02 + σp2 xT x′ d                                                        (4)\\n\\n\\n\\n                                      whereσ0 andσp are hyperparameters. We considered polynomial kernels with                                                                                                                                                                                                         d = 3. The\\n                                      squared exponential kernel is defined as:\\n\\n\\n\\n                                                                                                                                                                                                                                              σ2 exp − x − x′ 22\\n                                                                                                                                                                                    k s, s′ =                                                         p                                    l                                                (5)\\nAuthor Manuscript\\n                                      wherel and                                                   σp are also hyperparameters and |⋅|2 is the L2 norm. Finally, the Matérn kernel\\n                                      with v=                                     5is defined as:\\n                                                                                  2\\n\\n\\n\\n                                                                                                                                                                                         5 x − x′ 2+ 5 x − x′ 22                                                                              2  exp − 5 x − x′ 22\\n                                                                                                    k s, s′ =                                                       1 +                                                                                                               3l2                                                   (6)\\n                                                                                                                                                                                                                l                                                                                                   l\\n\\n\\n\\n                                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\\n\\n## GP classification\\n\\nIn binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\\n\\n## GP kernels for modeling proteins\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "    'images': [{'name': 'page_15.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': '$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.',\n",
       "      'md': '$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.',\n",
       "      'bBox': {'x': 150, 'y': 128, 'w': 153.5, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'GP classification',\n",
       "      'md': '## GP classification',\n",
       "      'bBox': {'x': 289, 'y': 438, 'w': 4.032, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\",\n",
       "      'md': \"In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\",\n",
       "      'bBox': {'x': 149, 'y': 128, 'w': 366.64999999999986, 'h': 14}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'GP kernels for modeling proteins',\n",
       "      'md': '## GP kernels for modeling proteins',\n",
       "      'bBox': {'x': 289, 'y': 438, 'w': 4.032, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "      'md': \"Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "      'bBox': {'x': 149,\n",
       "       'y': 314,\n",
       "       'w': 369.75998000000004,\n",
       "       'h': 10.514623181075024}}]},\n",
       "   {'page': 16,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                              Page 16\\n\\n\\n\\nAuthor Manuscript                          Wherel is once again a hyperparameter.\\n                                           L1 regression feature identification and weighting—L1 regression was used to\\n                                           identify residues and contacts in the ChR structure most important for each ChR functional\\n                                           property of interest. First, residues and contacts that covary were identified using the\\n                                           concatenated sequence and structure binary feature vector for each of the training set ChR\\n                                           variants. Each set of covarying residues and contacts was combined into a single feature. L1\\n                                           linear regression was used to select the features that contribute most to each ChR functional\\n                                           property of interest. The level of regularization was chosen by maximizing the log marginal\\n                                           likelihood of the Gaussian process regression model trained on the features selected at that\\n                                           level of regularization. We then performed Bayesian ridge regression on the selected features\\n                                           using the default settings in scikit-learn47. Residues and contacts with the largest absolute\\nAuthor Manuscript                          Bayesian ridge linear regression weights were plotted onto the C1C2 structure\\n                                           (Supplemental Figure 4). For feature identification and weighting, models were trained on\\n                                           both the training set and also the test set of 28 ChR sequences predicted to have useful\\n                                           combinations of diverse properties.\\n\\n\\n\\n                       Statistical analysis\\n                                           Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01.\\n                                           For statistical comparisons, we first performed a D’Agostino & Pearson normality test. If the\\n                                           p-value of a D’Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-\\n                                           Wallis test with Dunn’s multiple comparisonspost hoc test was used. If the data passed the\\n                                           normality test, a one-way ANOVA was used.\\n\\n\\n\\n                       Data availability\\nAuthor Manuscript                          The authors declare that data supporting the findings of this study are available within the\\n                                           paper and its supplementary information files. Source data for classification model training\\n                                           are provided in Dataset 1 and Dataset 2. Source data for regression model training are\\n                                           provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution\\n                                           at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\\n\\n\\n\\n                       Code availability\\n                                           Code used to train classification and regression models can be found at: https://github.com/\\n                                           fhalab/channels.\\n\\n\\n\\n                       Supplementary Material\\nAuthor Manuscript                          Refer to Web version on PubMed Central for supplementary material.\\n\\n\\n\\n                       Acknowledgements\\n                                           We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech\\n                                           Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer\\n                                           measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for\\n                                           helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative\\n                                           Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN\\n                                           R01MH117069, NIH Director’s Pioneer Award DP1OD025535, NIH Director’s New Innovator Award\\n                                           DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"Where l is once again a hyperparameter.\\n\\n## L1 regression feature identification and weighting\\n\\nL1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\\n\\n## Statistical analysis\\n\\nPlotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\\n\\n## Data availability\\n\\nThe authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\\n\\n## Code availability\\n\\nCode used to train classification and regression models can be found at: https://github.com/fhalab/channels.\\n\\n## Supplementary Material\\n\\nRefer to Web version on PubMed Central for supplementary material.\\n\\n## Acknowledgements\\n\\nWe thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "    'images': [{'name': 'page_16.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'Where l is once again a hyperparameter.',\n",
       "      'md': 'Where l is once again a hyperparameter.',\n",
       "      'bBox': {'x': 0, 'y': 0, 'w': 612}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'L1 regression feature identification and weighting',\n",
       "      'md': '## L1 regression feature identification and weighting',\n",
       "      'bBox': {'x': 0, 'y': 0, 'w': 612}},\n",
       "     {'type': 'text',\n",
       "      'value': 'L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.',\n",
       "      'md': 'L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.',\n",
       "      'bBox': {'x': 149, 'y': 111, 'w': 371.4499999999998, 'h': 13}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Statistical analysis',\n",
       "      'md': '## Statistical analysis',\n",
       "      'bBox': {'x': 90, 'y': 305, 'w': 89.33999999999997, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': \"Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\",\n",
       "      'md': \"Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\",\n",
       "      'bBox': {'x': 90, 'y': 305, 'w': 363.16999999999996, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Data availability',\n",
       "      'md': '## Data availability',\n",
       "      'bBox': {'x': 90, 'y': 405, 'w': 75.24000000000001, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).',\n",
       "      'md': 'The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).',\n",
       "      'bBox': {'x': 150, 'y': 423, 'w': 369.5499999999997, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Code availability',\n",
       "      'md': '## Code availability',\n",
       "      'bBox': {'x': 90, 'y': 505, 'w': 78.57000000000002, 'h': 10}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.',\n",
       "      'md': 'Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.',\n",
       "      'bBox': {'x': 150, 'y': 523, 'w': 364.45000000000005, 'h': 10}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Supplementary Material',\n",
       "      'md': '## Supplementary Material',\n",
       "      'bBox': {'x': 90, 'y': 567, 'w': 135.49200000000002, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Refer to Web version on PubMed Central for supplementary material.',\n",
       "      'md': 'Refer to Web version on PubMed Central for supplementary material.',\n",
       "      'bBox': {'x': 90, 'y': 567, 'w': 251.10000000000025, 'h': 12}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Acknowledgements',\n",
       "      'md': '## Acknowledgements',\n",
       "      'bBox': {'x': 90, 'y': 618, 'w': 112.392, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': \"We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "      'md': \"We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "      'bBox': {'x': 149, 'y': 636, 'w': 368.2239999999998, 'h': 8}}]},\n",
       "   {'page': 17,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                   Page 17\\n\\n\\n\\n                                    grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the\\nAuthor Manuscript                   Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector\\n                                    Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content\\n                                    of the information does not necessarily reflect the position or the policy of the Government, and no official\\n                                    endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards\\n                                    F31MH102913. J.E.R. is supported by the Children’s Tumor Foundation (Young Investigator Award 2016-01-006).\\n\\n\\n\\n                        References\\n                                    1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/\\n                                       science.aan5544 (2017).\\n                                    2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems.\\n                                       Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n                                    3. Lin JY A user’s guide to channelrhodopsin variants: features, limitations and future developments.\\n                                       Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed:\\n                                       20621963]\\nAuthor Manuscript                   4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain\\n                                       structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n                                    5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics.\\n                                       Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n                                    6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of\\n                                       microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed:\\n                                       22179551]\\n                                    7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature\\n                                       neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n                                    8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral\\n                                       membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane\\n                                       localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017).\\n                                       [PubMed: 29059183]\\n                                    9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric\\nAuthor Manuscript                      channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of\\n                                       America (2017).\\n                                    10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature\\n                                         reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n                                    11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods\\n                                         11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n                                    12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion\\n                                         channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi:\\n                                         10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n                                    13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of\\n                                         channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–\\n                                         1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n                                    14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches.\\n                                         Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n                                    15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin\\nAuthor Manuscript                        variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/\\n                                         j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n                                    16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion\\n                                         selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016).\\n                                         [PubMed: 26699459]\\n                                    17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482,\\n                                         369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n                                    18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344,\\n                                         409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\\n\\n## References\\n\\n1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "    'images': [{'name': 'page_17.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\",\n",
       "      'md': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\",\n",
       "      'bBox': {'x': 150, 'y': 68, 'w': 367.84000000000003, 'h': 8}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'References',\n",
       "      'md': '## References',\n",
       "      'bBox': {'x': 90, 'y': 144, 'w': 64.572, 'h': 12}},\n",
       "     {'type': 'text',\n",
       "      'value': \"1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "      'md': \"1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "      'bBox': {'x': 150, 'y': 162, 'w': 371.8170000000001, 'h': 9}}]},\n",
       "   {'page': 18,\n",
       "    'text': '                   Bedbrook et al.                                                                                                            Page 18\\n\\n\\n\\nAuthor Manuscript                   19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and\\n                                         peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017).\\n                                         [PubMed: 28671695]\\n                                    20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related\\n                                         proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202\\n                                         (2013). [PubMed: 23225662]\\n                                    21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by\\n                                         recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed:\\n                                         12042875]\\n                                    22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered\\n                                         microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed:\\n                                         24952910]\\n                                    23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi:\\n                                         10.1038/nn.2495 (2010). [PubMed: 20081849]\\n                                    24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian\\nAuthor Manuscript                        processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013).\\n                                         [PubMed: 23277561]\\n                                    25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi:\\n                                         10.1126/science.aan8862 (2017).\\n                                    26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature\\n                                         communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n                                    27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of\\n                                         channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/\\n                                         bi901634p (2010). [PubMed: 20000562]\\n                                    28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans\\n                                         triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.\\n                                         2005.11.032 (2005). [PubMed: 16360690]\\n                                    29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated\\n                                         optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\nAuthor Manuscript                   30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and\\n                                         Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-\\n                                         neuro-080317-062048 (2018).\\n                                    31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents.\\n                                         Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n                                    32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron\\n                                         Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.\\n                                         2015.10.017 (2015). [PubMed: 26586182]\\n                                    33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in\\n                                         vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–\\n                                         14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n                                    34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering.\\n                                         Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n                                    35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in\\n                                         mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/\\nAuthor Manuscript                        ncomms5894 (2014).\\n                                    36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-\\n                                         Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.\\n                                         2015.06.020 (2015). [PubMed: 26211362]\\n                                    37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server.\\n                                         Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n                                    38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic\\n                                         Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular\\n                                         neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "    'images': [{'name': 'page_18.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "      'md': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "      'bBox': {'x': 150, 'y': 69, 'w': 371.17800000000034, 'h': 9}}]},\n",
       "   {'page': 19,\n",
       "    'text': '                   Bedbrook et al.                                                                                                          Page 19\\n\\n\\n\\nAuthor Manuscript                   39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in\\n                                         hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n                                    40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development\\n                                         of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9,\\n                                         5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n                                    41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal\\n                                         neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/\\n                                         bht278 (2015). [PubMed: 24108807]\\n                                    42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to\\n                                         the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n                                    43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis\\n                                         of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017).\\n                                         [PubMed: 28506280]\\n                                    44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical\\n                                         computation. Computing in Science and Engineering 13, 22–30 (2011).\\nAuthor Manuscript                   45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–\\n                                         95 (2007).\\n                                    46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20\\n                                         (2007).\\n                                    47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning\\n                                         Research 12, 2825–2830 (2011).\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "    'images': [{'name': 'page_19.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "      'md': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "      'bBox': {'x': 150, 'y': 69, 'w': 370.071, 'h': 9}}]},\n",
       "   {'page': 20,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                                                                  Page 20\\n                                       Photocurrent properties                                Irain Gassilicalion mnoclols                                                Irain rogression mocels\\nAuthor Manuscript                            Current sirengh                  Ihreshold out nn-{unclianai ChRs Irom Ihe @brary                      approximate Ihe litness lantscape {ordifferent properties\\n                                                                             H                                               1,161 ChRs            Current strength             Olf-klnetlcs          Spectral properlies\\n                                                      Sleajy                                                                     ahove\\n                                                                                                                               Uhroslold     1\\n                                              \\'crk-kinebcs                                                                     28 top\\n                                                                             L                                                 VatiaIs\\n                                                                                                                             pp > 0.4        1\\n                                          Wavelengih sensilivity                    0,0                            0,8                                                                                  \"25\\n                                                  acilyaticn                                                                                           mea sured                 mea sured                 ICasurod\\n                                                                                      Preclcted probabililty of locallzing\\n                                                                                Traning dala                                 Traning cata\\n                                                                                 (102 variants -                               124 vaiants}\\n                                                                                      Kanons\\n                                    !                                                                 Cla ssilicalion22 varantsvarials                Regrossion                                Chractorizalion\\n                                         400 Wavclengh [nm]500600                                       moclels                                       morlels                                        valiclalion\\n                                                                             Training set                                                                                              Z00 R = 0.92\\nAuthor Manuscript                   2L z0o0                                                                                                                                2           TE00\\n                                                                                                                                                                                       I0o\\n                                          IOOO                                                                                                                                          F00\\n                                                                                                                                                                                                 Tooo Cu\\n                                       2                                                                                       1                                                              R = 0.97\\n                                    9                                                                                                                                      8             20\\n                                       I                                                                                                                                      I     L 075110olR = 0.96\\n                                       {                                                                                        :                                              1       0 501\\n                                       1                                                                                                                                      1\\nAuthor Manuscript                                       parents            single-block             IuIli-block           classilicalion        regression\\n                                                                               suap                    seap                                                                                         Measured\\n                                            mutalions:\\n                                            Figure 1.\\n                                           Machine learning-guided optimization of ChRs. (a) Upon light exposure, ChRs open and\\n                                           reach a        peak inward current and then desensitize reaching a lowersteady-state current. We\\n                                           use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR\\n                                           off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the\\n                                           time to reach 50% of the light-exposed current after light removal. As a metric for\\n                                           wavelength sensitivity of activation, we used the normalized photocurrent with green (546\\n                                           nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and\\n                                           red-shifted ChRs (peak activation: ~520–650 nm). (b) We trained classification models to\\nAuthor Manuscript                          predict whether ChRs would localize correctly to the plasma membrane and function (i.e.,\\n                                           ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of\\n                                           localization and function). (c) We then trained regression models to approximate the fitness\\n                                           landscape for each property of interest for the recombination library (inset show hypothetical\\n                                           fitness landscapes). (b–c) Models were trained with photocurrent properties for each ChR in\\n                                           the training set (plots show 20-fold cross validation on the training set). Sequences predicted\\n                                           to localize and function by the classification models and predicted to have an optimized set\\n                                           of functional properties by the regression models were selected for further characterization,\\n\\n\\n\\n                                                     Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'Figure 1\\n\\n## Figure 1. Machine learning-guided optimization of ChRs.\\n\\n(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "    'images': [{'name': 'img_p19_1.png',\n",
       "      'height': 766,\n",
       "      'width': 873,\n",
       "      'x': 120.13676,\n",
       "      'y': 61.999998000000005,\n",
       "      'original_width': 1800,\n",
       "      'original_height': 1579},\n",
       "     {'name': 'page_20.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'Figure 1',\n",
       "      'md': 'Figure 1',\n",
       "      'bBox': {'x': 140.41257364375716,\n",
       "       'y': 102.04741652741515,\n",
       "       'w': 237.37537924398623,\n",
       "       'h': 32.63122991122715}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Figure 1. Machine learning-guided optimization of ChRs.',\n",
       "      'md': '## Figure 1. Machine learning-guided optimization of ChRs.',\n",
       "      'bBox': {'x': 140.41257364375716,\n",
       "       'y': 89.68710216710183,\n",
       "       'w': 97.4228118980527,\n",
       "       'h': 48.4524322924282}},\n",
       "     {'type': 'text',\n",
       "      'value': '(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "      'md': '(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "      'bBox': {'x': 133.48912508247423,\n",
       "       'y': 67.93294889295039,\n",
       "       'w': 371.8399999999998,\n",
       "       'h': 48.4524322924282}}]},\n",
       "   {'page': 21,\n",
       "    'text': '                   Bedbrook et al.                                                                                                Page 21\\n\\n\\n\\nAuthor Manuscript                   e.g., the 28 top variants. (d) The classification model was trained with 102 recombination\\n                                    variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression\\n                                    models were trained with 124 recombination variants. (e) Measurements of training set ChR\\n                                    and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n\\n                                    = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set\\n                                    data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model\\n                                    predictions vs measured photocurrent property for each of the 28 designer ChRs. R\\n                                    represents the Pearson correlation coefficient.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "    'images': [{'name': 'page_21.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "      'md': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "      'bBox': {'x': 149,\n",
       "       'y': 71,\n",
       "       'w': 370.94000000000005,\n",
       "       'h': 10.51462318107501}}]},\n",
       "   {'page': 22,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                             Page 22\\n                                        CheRiff     CsChrimR       C1C2        11_10                                  10 10        15_10       28_10        21_10\\nAuthor Manuscript\\n                                                                                                                                                                            ZumM\\n                                                              Photocurrent strength with different wavelength excitation                                 Tos (ms]\\n                                                                                                                                                       102   103  104\\n                                    89 000500         Peak\\n                                                      Steady state                                                                                            Parenis\\n                                                                                                                                                                      21_10\\n                                                                                                                                         CsChiimR                        10Q ms\\n                                      3 2000                                                                                               CneRill\\nAuthor Manuscript                   20 1000\\n                                                                                                                                                                      Cheril\\n                                        4000\\n                                    8 2000\\n                                    ?1\\n                                   # 20001000\\n                                                                                3    8e9 83 e9\\n                                     [ 0.75                                               2500                                          2500\\nAuthor Manuscript                                                                      1 2000                                       0 2000\\n                                                                                     2    1500                                       8 1500\\n                                                                                          1000                                      82  1000\\n                                        0.25                                               500                                       2   500\\n                                        0.00\\n                                              400     Wavelength [nm] 600500                      10-1 Intensity [mW mm?]                             Intensity [mW mm?]\\n                                           Figure 2.\\n                                           The model-predicted ChRs exhibit a large range of functional properties often far exceeding\\n                                           the parents. (a) Representative current traces after 0.5 s light exposure for select designer\\n                                           ChR variants with corresponding expression and localization in HEK cells. Vertical colored\\n                                           scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents\\nAuthor Manuscript                          250 ms. The variant color presented in (a) is constant throughout panels. (b) Measured peak\\n                                           and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8\\n                                           cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm\\n                                           light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. (c) Off-kinetics decay rate (τoff)\\n                                           following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2).\\n                                           Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1\\n                                           ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly,\\n                                           ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in\\n                                           photocurrent 0.5 s after the light exposure. (d) Peak and steady-state photocurrent strength\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'Figure 2\\n\\n## Figure 2.\\nThe model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "    'images': [{'name': 'img_p21_1.png',\n",
       "      'height': 859,\n",
       "      'width': 873,\n",
       "      'x': 120,\n",
       "      'y': 62.000001999999995,\n",
       "      'original_width': 1800,\n",
       "      'original_height': 1770},\n",
       "     {'name': 'page_22.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'Figure 2',\n",
       "      'md': 'Figure 2',\n",
       "      'bBox': {'x': 126.9278350515464,\n",
       "       'y': 240.0302772968568,\n",
       "       'w': 11.876288659793813,\n",
       "       'h': 17.803027529685682}},\n",
       "     {'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Figure 2.',\n",
       "      'md': '## Figure 2.',\n",
       "      'bBox': {'x': 126.9278350515464,\n",
       "       'y': 240.0302772968568,\n",
       "       'w': 34.335,\n",
       "       'h': 17.803027529685682}},\n",
       "     {'type': 'text',\n",
       "      'value': 'The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "      'md': 'The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "      'bBox': {'x': 126.9278350515464,\n",
       "       'y': 69.41793013736903,\n",
       "       'w': 369.28,\n",
       "       'h': 25.715484209545984}}]},\n",
       "   {'page': 23,\n",
       "    'text': '                  Bedbrook et al.                                                                                                Page 23\\n\\n\\n\\nAuthor Manuscript                  with varying light irradiances compared with parental ChRs (CheRiff,n = 5 cells;\\n                                   CsChrimR,n = 5 cells; C1C2,n = 4 cells; 28_10,n = 5 cells; 11_10,n = 5 cells; 25_9,n = 5\\n                                   cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs\\n                                   (CheRiff,n = 6 cells; CsChrimR,n = 5 cells; C1C2,n = 4 cells; 11_10,n = 6 cells; 12_10,n\\n                                   = 7 cells; 25_9,n = 5 cells; 10_10,n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and\\n                                   ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data\\n                                   are mean±SEM.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                          Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "    'images': [{'name': 'page_23.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "      'md': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "      'bBox': {'x': 150, 'y': 99, 'w': 369.76999999999975, 'h': 10}}]},\n",
       "   {'page': 24,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                                Page 24\\n                                        Camkil                          [PCPCE                                                                                         Light intensity:\\n                                                ChR     BYFP WPREOGHpa                            peak    steady slale                                               3x102 (Mw mm?]\\nAuthor Manuscript                             88683                                         2500  8x10 J mW mm?                   Chageri\\n                                                                                                                                  ChAgerz\\n                                               CoCnr                                    3 2o0]                              {     ChRger3                            50 mv ChR2\\n                                                ChAz                                     L 1500                                   CoChR\\n                                                                                                                                  Chaz\\n                                        ChR?                 ChRgert                        1000                            8                                             CoChR\\n                                                                                                                                   Light intensity (mw mm]                ChRgeri\\n                                                                                           4ooo\\n                                                                                         1 3000             Mm\\n                                                             IMum                                                                                       ChR2              ChRgerz\\n                                        ChRger?              ChRger3                                                                                 ChRgerz]\\n                                                                                                                                                              ChRgeri\\n                                                                                            2000                            {\\n                                                                                            1000-                                                                         ChRger3\\n                                                                                                                            3\\n                                                                                                 CoChR ChRgerzChRgert ChRgera           Frequency (Hz]\\n                                               Direci intracranial injeclion Systemic delivery (Sx10\" VC)          Systemic delivery (Ixio\" VG)   m        Systemic delivery (Ix1O\" VG)\\nAuthor Manuscript                              Systemlc delivery Sx10\" VG)                                         CnRgerz\\n                                   L 1500 chRgerz\\n                                                 2 2000 ChAgori                                             1 1000 CaCha                             1\\n                                            ChRgor3\\n                                      1000 ChRZ                                                                500                                          ChAigerz\\n                                                                                                                                                                    CoChR\\n                                       500                                                                                                           8      ChAger3\\n                                           10\"Light intensity [mW mm*7]                                              Light intensity [mW mm?]                     Frequency [Hz]\\n                                                                                                                   Systemic delivery (Ix10\" VG)         ChRger1   ChRger2    ChRger3\\n                                    0.5 5 VYV I &r10 \\'                                                      2       ChAgerz\\n                                                             Light intensity[IW mm ?]                               CoChR                           2 Hz\\n                                                                             Systemic delivery (1x10\" VG)   1\\n                                                                                 Light excitable cells (%0)\\n                                                                                                            3                                       5 Hz                           50 mV\\n                                                                Ixio\\'      ChR2\\n                                  1000 PA                                 CoChR                                                                         L\\n                                                                         ChRagi                                                                    20 Hz\\n                                                                 2x0\"    ChAgerz\\n                                                                         ChRger3                                    10 *Light intensity [mW mm ]\\n                                            Figure 3.\\nAuthor Manuscript                          ChRger variants in cultured neurons and in acute brain slices outperform the commonly used\\n                                           ChR2(H134R) and the high performance CoChR. (a) ChRs were cloned into an AAV vector\\n                                           with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression\\n                                           in culture and      in vivo. (b) Cultured neurons expressing ChRgers and ChR2(H134R) under\\n                                           the hSyn promoter (repeated independently six times per construct with similar results). (c)\\n                                           Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate-\\n                                            intensity (0.8 mW mm−2) light in cultured neurons (ChR2,n = 16 cells; CoChR,n = 17\\n                                            cells; ChRger1,n = 9 cells; ChRger2,n = 24 cells; ChRger3,n = 9 cells). (d) Spike fidelity\\n                                            with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1,n = 6\\n                                            cells; ChRger2,n = 6 cells; ChRger3,n = 6 cells; CoChR,n = 7 cells; ChR2,n = 7 cells). (e)\\n                                            Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured\\n                                            neurons (ChRger1,n = 9 cells; ChRger2,n = 12 cells; ChRger3,n = 7 cells; ChR2,n = 8\\nAuthor Manuscript                           cells). (f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms\\n                                            pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing\\n                                            for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced\\n                                            depolarization. (g) Photocurrent strength with varying light irradiances in acute brain slice\\n                                            after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC\\n                                            (ChRger1,n = 11 cells; ChRger2,n = 11 cells; ChRger3,n = 11 cells; ChR2,n = 9 cells) or\\n                                            after systemic delivery of CamKIIa-ChRger2 (ChRger2,n = 6 cells; 5×1011 vg/animal). (h)\\n                                            Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at\\n\\n\\n\\n                                                   Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "    'images': [{'name': 'img_p23_1.png',\n",
       "      'height': 646,\n",
       "      'width': 873,\n",
       "      'x': 120.16205,\n",
       "      'y': 61.99999699999995,\n",
       "      'original_width': 1800,\n",
       "      'original_height': 1333},\n",
       "     {'name': 'page_24.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "      'md': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "      'bBox': {'x': 149,\n",
       "       'y': 70.4126283947368,\n",
       "       'w': 371.63999999999976,\n",
       "       'h': 20.289287481424154}}]},\n",
       "   {'page': 25,\n",
       "    'text': '                   Bedbrook et al.                                                                                                    Page 25\\n\\n\\n\\nAuthor Manuscript                   varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of\\n                                    rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression\\n                                    throughout the cortex (5×1011 vg/animal; repeated independently five times per construct\\n                                    with similar results). (j) The fraction of light excitable neurons in the PFC after systemic\\n                                    delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only\\n                                    neurons expressing the eYFP marker (1×1011 vg/animal). Peak (solid line) and steady-state\\n                                    (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in\\n                                    acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells)\\n                                    and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying\\n                                    stimulation frequency in acute brain slice after systemic delivery (1×1011 vg/animal)\\n                                    (CoChR,      n = 15 cells; ChRger1,n = 9 cells; ChRger2,n = 5 cells; ChRger3,n = 8 cells) with\\n                                    1 mW mm−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW\\nAuthor Manuscript                   mm−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "    'images': [{'name': 'page_25.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'text',\n",
       "      'value': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "      'md': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "      'bBox': {'x': 149,\n",
       "       'y': 71,\n",
       "       'w': 370.5099999999998,\n",
       "       'h': 13.514623181075024}}]},\n",
       "   {'page': 26,\n",
       "    'text': '                   Bedbrook et al.                                                                                                                             Page 26\\n                                               Systemic delivery               Fiber implantation       0 Fiber    anti-eYFP                 Intracrania self-stimulation\\nAuthor Manuscript                                      843    WPAE ngHpa                                                               10 Hz stimulation         mW power\\n                                                     ChRgerz                                                      VTA          1 400 ChR2ChRger2\\n                                                          TAAV PHPeB                                   ChRz-TS-eYFP            1 300\\n                                                                                                            Fiber                200\\n                                               Dat-Cre                                                                         2 100\\n                                                                                                        ChRgerz TS- eYFP300 Hm           Power [mW] 10 20  1 3  Simulation\\n                                                                                                                                                               irequency [Hz]\\n                                                Systemic delivery               Fiber on skull surface                     Minimally invasive optical control of locomotion\\n                                       Camkila           #YFP  WPRE hGHpa                                   8                 ChBger?                          ChR?\\n                                               ChAgerz          TAAV-PHPeB                       skull                  671 nm\\n                                                                                                                              447 Dmt               O00 447 nin\\n                                                                                                                                                        671 nm\\n                                                                                                           2 I\\n                                                                                                           8\\nAuthor Manuscript\\n                                                                                                                              Frames                          Frames\\n                                                                                                            ChRgerz\\n                                                                                                           1 Frame                   Frame                 Frame\\n                                                                                                               Frame                 Frame                 Frame\\n                                  anti-eYFP         50 Mum          ChRqerz TS eYFP 50 m                    8\\n                                  ChR2-IS eYFP\\n                                           Figure 4.\\n                                           Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral\\n                                           modulation. (a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP\\n                                           or ChR2(H134R)-TS-eYFP (3×1011 vg/mouse) intoDat-Cre animals coupled with fiber\\n                                           optic implantation above the VTA enabled blue light-induced intracranial self-stimulation\\nAuthor Manuscript                          (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light\\n                                           power and varying stimulation frequencies. ChRger2,n = 4 animals; ChR2(H134R),                                    n = 4\\n                                           animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and\\n                                           ChRger2 (bottom). (b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged\\n                                           CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×1011 vg/mouse) into wild type\\n                                           (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the\\n                                           skull above the right M2 that had been thinned to create a level surface for the fiber-skull\\n                                           interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed\\n                                           velocity. Coronal slices show expression throughout cortex with higher magnification image\\n                                           of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation\\n                                           of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz\\n                                           stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2,                          n = 5 animals;\\nAuthor Manuscript                          ChR2(H134R),n = 5 animals. No turning behavior was observed in any animal with 10 Hz\\n                                           stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral\\n                                           genomes.\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "    'md': '## Figure 4.\\n\\nValidation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "    'images': [{'name': 'img_p25_1.png',\n",
       "      'height': 543,\n",
       "      'width': 873,\n",
       "      'x': 120,\n",
       "      'y': 62.00000300000005,\n",
       "      'original_width': 1800,\n",
       "      'original_height': 1118},\n",
       "     {'name': 'page_26.jpg',\n",
       "      'height': 792,\n",
       "      'width': 612,\n",
       "      'x': 0,\n",
       "      'y': 0,\n",
       "      'type': 'full_page_screenshot'}],\n",
       "    'items': [{'type': 'heading',\n",
       "      'lvl': 2,\n",
       "      'value': 'Figure 4.',\n",
       "      'md': '## Figure 4.',\n",
       "      'bBox': {'x': 150, 'y': 351, 'w': 34.335, 'h': 9}},\n",
       "     {'type': 'text',\n",
       "      'value': 'Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "      'md': 'Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "      'bBox': {'x': 121.48453608247422,\n",
       "       'y': 63.4824339779006,\n",
       "       'w': 369.5099999999998,\n",
       "       'h': 27.672044920810322}}]}],\n",
       "  'job_metadata': {'credits_used': 1950.0,\n",
       "   'job_credits_usage': 0,\n",
       "   'job_pages': 0,\n",
       "   'job_is_cache_hit': True},\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf'}]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "json_objs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "26"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(json_objs[0][\"pages\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\n",
      "\n",
      "[The rest of the introduction is cut off in the image]Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.# Results\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs areexpressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "## Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize otherneuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "## Online methods\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRsNO_CONTENT_HERE## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated inphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "## Animals\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) usethe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "## GP regression\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "## GP classification\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "## L1 regression feature identification and weighting\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "## Data availability\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "## Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hubgrant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "## References\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\n",
      "\n",
      "30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\n",
      "\n",
      "31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\n",
      "\n",
      "32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\n",
      "\n",
      "33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\n",
      "\n",
      "34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\n",
      "\n",
      "35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\n",
      "\n",
      "36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\n",
      "\n",
      "37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\n",
      "\n",
      "38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\n",
      "\n",
      "40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\n",
      "\n",
      "41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\n",
      "\n",
      "42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\n",
      "\n",
      "43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\n",
      "\n",
      "44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\n",
      "\n",
      "45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\n",
      "\n",
      "46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).Figure 1\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\n",
      "\n",
      "The figure contains several parts:\n",
      "\n",
      "a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\n",
      "\n",
      "b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\n",
      "\n",
      "c. Train regression models graphs for current strength, off-kinetics, and spectral properties.\n",
      "\n",
      "d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\n",
      "\n",
      "e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\n",
      "\n",
      "f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\n",
      "\n",
      "The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.Figure 2\n",
      "\n",
      "## Figure 2.\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strength\n",
      "\n",
      "The figure contains several panels:\n",
      "\n",
      "a) A series of current traces and corresponding cell images for different ChR variants.\n",
      "\n",
      "b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\n",
      "\n",
      "c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\n",
      "\n",
      "d) Line graphs showing normalized photocurrent vs wavelength.\n",
      "\n",
      "e) Line graphs showing peak and steady-state photocurrent vs light intensity.\n",
      "\n",
      "Each panel contains detailed data comparing various ChR variants across different properties and conditions.with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.## Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n"
     ]
    }
   ],
   "source": [
    "full_text = \"\".join([page[\"md\"] for page in json_objs[0][\"pages\"]])\n",
    "print(full_text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> Image for page 1: [{'name': 'img_p0_1.png', 'height': 102, 'width': 102, 'x': 61.5, 'y': 22.5, 'original_width': 152, 'original_height': 152}, {'name': 'page_1.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 2: [{'name': 'page_2.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 3: [{'name': 'page_3.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 4: [{'name': 'page_4.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 5: [{'name': 'page_5.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 6: [{'name': 'page_6.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 7: [{'name': 'page_7.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 8: [{'name': 'page_8.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 9: [{'name': 'page_9.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 10: [{'name': 'page_10.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 11: [{'name': 'page_11.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 12: [{'name': 'page_12.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 13: [{'name': 'page_13.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 14: [{'name': 'page_14.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 15: [{'name': 'page_15.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 16: [{'name': 'page_16.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 17: [{'name': 'page_17.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 18: [{'name': 'page_18.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 19: [{'name': 'page_19.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 20: [{'name': 'img_p19_1.png', 'height': 766, 'width': 873, 'x': 120.13676, 'y': 61.999998000000005, 'original_width': 1800, 'original_height': 1579}, {'name': 'page_20.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 21: [{'name': 'page_21.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 22: [{'name': 'img_p21_1.png', 'height': 859, 'width': 873, 'x': 120, 'y': 62.000001999999995, 'original_width': 1800, 'original_height': 1770}, {'name': 'page_22.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 23: [{'name': 'page_23.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 24: [{'name': 'img_p23_1.png', 'height': 646, 'width': 873, 'x': 120.16205, 'y': 61.99999699999995, 'original_width': 1800, 'original_height': 1333}, {'name': 'page_24.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 25: [{'name': 'page_25.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n",
      "> Image for page 26: [{'name': 'img_p25_1.png', 'height': 543, 'width': 873, 'x': 120, 'y': 62.00000300000005, 'original_width': 1800, 'original_height': 1118}, {'name': 'page_26.jpg', 'height': 792, 'width': 612, 'x': 0, 'y': 0, 'type': 'full_page_screenshot'}]\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "storage_dir = \"./storage/tmp\"\n",
    "\n",
    "data_images_dir = os.path.join(storage_dir, \"data_images\")\n",
    "if os.path.exists(data_images_dir):\n",
    "    for filename in os.listdir(data_images_dir):\n",
    "        file_path = os.path.join(data_images_dir, filename)\n",
    "        try:\n",
    "            if os.path.isfile(file_path):\n",
    "                os.unlink(file_path)\n",
    "        except Exception as e:\n",
    "            print(f\"Error deleting file {file_path}: {e}\")\n",
    "\n",
    "image_dicts = parser.get_images(json_objs, download_path=data_images_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Estimated token count: 17099\n",
      "Exact token count: 19298\n"
     ]
    }
   ],
   "source": [
    "from anthropic import Anthropic\n",
    "\n",
    "def estimate_anthropic_tokens(text: str) -> int:\n",
    "    client = Anthropic()\n",
    "    return client.count_tokens(text)\n",
    "\n",
    "def get_exact_token_count(text: str, model: str = \"claude-3-sonnet-20240229\") -> int:\n",
    "    client = Anthropic()\n",
    "    response = client.messages.create(\n",
    "        model=model,\n",
    "        max_tokens=1,\n",
    "        messages=[{\"role\": \"user\", \"content\": text}]\n",
    "    )\n",
    "    return response.usage.input_tokens\n",
    "\n",
    "# Example usage\n",
    "estimated_count = estimate_anthropic_tokens(full_text)\n",
    "print(f\"Estimated token count: {estimated_count}\")\n",
    "\n",
    "exact_count = get_exact_token_count(full_text)\n",
    "print(f\"Exact token count: {exact_count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Estimated token count: 481\n",
      "Exact token count: 530\n",
      "Estimated token count: 643\n",
      "Exact token count: 704\n",
      "Estimated token count: 685\n",
      "Exact token count: 752\n",
      "Estimated token count: 716\n",
      "Exact token count: 788\n",
      "Estimated token count: 752\n",
      "Exact token count: 849\n",
      "Estimated token count: 913\n",
      "Exact token count: 1053\n",
      "Estimated token count: 889\n",
      "Exact token count: 988\n",
      "Estimated token count: 841\n",
      "Exact token count: 938\n",
      "Estimated token count: 738\n",
      "Exact token count: 820\n",
      "Estimated token count: 755\n",
      "Exact token count: 873\n",
      "Estimated token count: 5\n",
      "Exact token count: 13\n",
      "Estimated token count: 858\n",
      "Exact token count: 1038\n",
      "Estimated token count: 856\n",
      "Exact token count: 972\n",
      "Estimated token count: 616\n",
      "Exact token count: 689\n",
      "Estimated token count: 711\n",
      "Exact token count: 797\n",
      "Estimated token count: 663\n",
      "Exact token count: 745\n",
      "Estimated token count: 1283\n",
      "Exact token count: 1497\n",
      "Estimated token count: 1258\n",
      "Exact token count: 1486\n",
      "Estimated token count: 431\n",
      "Exact token count: 518\n",
      "Estimated token count: 544\n",
      "Exact token count: 582\n",
      "Estimated token count: 151\n",
      "Exact token count: 180\n",
      "Estimated token count: 450\n",
      "Exact token count: 505\n",
      "Estimated token count: 204\n",
      "Exact token count: 258\n",
      "Estimated token count: 651\n",
      "Exact token count: 720\n",
      "Estimated token count: 325\n",
      "Exact token count: 384\n",
      "Estimated token count: 677\n",
      "Exact token count: 787\n"
     ]
    }
   ],
   "source": [
    "pages = []\n",
    "\n",
    "for page in json_objs[0][\"pages\"]:\n",
    "\n",
    "    single_page_text = page[\"md\"]\n",
    "\n",
    "    estimated_count = estimate_anthropic_tokens(single_page_text)\n",
    "    print(f\"Estimated token count: {estimated_count}\")\n",
    "\n",
    "    exact_count = get_exact_token_count(single_page_text)\n",
    "    print(f\"Exact token count: {exact_count}\")\n",
    "\n",
    "    pages.append({\n",
    "        \"estimated_token_count\": estimated_count,\n",
    "        \"exact_token_count\": exact_count,\n",
    "    })\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "from llmsherpa.readers import LayoutPDFReader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "llmsherpa_api_url = \"http://localhost:5010/api/parseDocument?renderFormat=all\"\n",
    "pdf_url = \"/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf\" # also allowed is a file path e.g. /home/downloads/xyz.pdf\n",
    "# pdf_url = \"/Users/loyalshababo/dev/plotreader/sandbox/storage/README.pdf\"\n",
    "pdf_reader = LayoutPDFReader(llmsherpa_api_url)\n",
    "doc = pdf_reader.read_pdf(pdf_url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import llmsherpa\n",
    "    # ... existing code ...\n",
    "\n",
    "def to_markdown(doc):\n",
    "    \"\"\"\n",
    "    Converts the document to a markdown formatted string.\n",
    "    \"\"\"\n",
    "    markdown = \"\"\n",
    "    \n",
    "    for section in doc.sections():\n",
    "        # Add section header\n",
    "        markdown += f\"{'#' * (section.level + 1)} {section.to_text()}\\n\\n\"\n",
    "        \n",
    "        for child in section.children:\n",
    "            if isinstance(child, llmsherpa.readers.layout_reader.Paragraph):\n",
    "                markdown += f\"{child.to_text()}\\n\\n\"\n",
    "            elif isinstance(child, llmsherpa.readers.layout_reader.ListItem):\n",
    "                markdown += f\"- {child.to_text()}\\n\"\n",
    "            elif isinstance(child, llmsherpa.readers.layout_reader.Table):\n",
    "                markdown += child.to_text() + \"\\n\\n\"\n",
    "    \n",
    "    return markdown.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "doc_sherpa = to_markdown(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "from typing import List, Dict, Tuple\n",
    "from difflib import SequenceMatcher\n",
    "\n",
    "from plotreader.utils.base import BasicAnthropicLLM\n",
    "\n",
    "class MarkdownMerger:\n",
    "    def __init__(self, version1: str, version2: str):\n",
    "        self.version1 = version1\n",
    "        self.version2 = version2\n",
    "\n",
    "    def parse_markdown(self, markdown: str) -> List[Tuple[int, str, str]]:\n",
    "        \"\"\"\n",
    "        Parse markdown into a list of (level, header, content) tuples.\n",
    "        \"\"\"\n",
    "        sections = []\n",
    "        current_level = 0\n",
    "        current_header = \"\"\n",
    "        current_content = []\n",
    "\n",
    "        for line in markdown.split('\\n'):\n",
    "            if line.startswith('#'):\n",
    "                if current_header:\n",
    "                    sections.append((current_level, current_header, '\\n'.join(current_content)))\n",
    "                current_level = len(line.split()[0])  # Count the number of '#'\n",
    "                current_header = line.lstrip('#').strip()\n",
    "                current_content = []\n",
    "            else:\n",
    "                current_content.append(line)\n",
    "\n",
    "        if current_header:\n",
    "            sections.append((current_level, current_header, '\\n'.join(current_content)))\n",
    "\n",
    "        return sections\n",
    "\n",
    "    def similarity_score(self, a: str, b: str) -> float:\n",
    "        return SequenceMatcher(None, a, b).ratio()\n",
    "\n",
    "    def match_sections(self, sections1: List[Tuple[int, str, str]], \n",
    "                       sections2: List[Tuple[int, str, str]]) -> List[Tuple[Tuple[int, str, str], Tuple[int, str, str]]]:\n",
    "        \"\"\"\n",
    "        Match sections based on header similarity.\n",
    "        \"\"\"\n",
    "        matched = []\n",
    "        for s1 in sections1:\n",
    "            best_match = max(sections2, key=lambda s2: self.similarity_score(s1[1], s2[1]))\n",
    "            if self.similarity_score(s1[1], best_match[1]) > 0.6:  # Threshold for considering a match\n",
    "                matched.append((s1, best_match))\n",
    "        return matched\n",
    "    \n",
    "    def merge_content(self, header: str,content1: str, content2: str) -> str:\n",
    "        \"\"\"\n",
    "        Merge content from two versions.\n",
    "        \"\"\"\n",
    "        try:\n",
    "            return self._merge_content_llm(header, content1, content2)\n",
    "        except Exception as e:\n",
    "            print(e)\n",
    "            return self._merge_content_length_based(header, content1, content2)\n",
    "    \n",
    "    def _merge_content_length_based(self, header: str,content1: str, content2: str) -> str:\n",
    "        \"\"\"\n",
    "        Merge content from two versions, preferring longer paragraphs.\n",
    "        \"\"\"\n",
    "\n",
    "        paragraphs1 = content1.split('\\n\\n')\n",
    "        paragraphs2 = content2.split('\\n\\n')\n",
    "        \n",
    "        merged = []\n",
    "        for p1, p2 in zip(paragraphs1, paragraphs2):\n",
    "            merged.append(p1 if len(p1) >= len(p2) else p2)\n",
    "        \n",
    "        # Add any remaining paragraphs from the longer version\n",
    "        merged.extend(paragraphs1[len(paragraphs2):])\n",
    "        merged.extend(paragraphs2[len(paragraphs1):])\n",
    "        \n",
    "        return '\\n\\n'.join(merged)\n",
    "\n",
    "    def _merge_content_llm(self, header: str,content1: str, content2: str) -> str:\n",
    "        \"\"\"\n",
    "        Merge content from two versions, using an LLM.\n",
    "        \"\"\"\n",
    "        \n",
    "\n",
    "        claude = BasicAnthropicLLM(model=\"claude-3-haiku-20240307\")\n",
    "\n",
    "        prompt = f\"\"\"Below are two different markdown strings extraced from a scientific paper via different parsers.\n",
    "        Your job is to merge the two markdown strings into a single markdown string that is your best estimate of the original text.\n",
    "        The goal would be for each figure caption to be its own section.\n",
    "        \n",
    "        SECTION HEADER TITLE:\n",
    "        {header}\n",
    "\n",
    "        PARSED CONTENT 1:\n",
    "        {content1}\n",
    "\n",
    "        PARSED CONTENT 2:\n",
    "        {content2}\n",
    "\n",
    "        Return only the final markdown string without any other text.\n",
    "        \"\"\"\n",
    "\n",
    "        response = claude.message(prompt)\n",
    "        raw_text = response.content[0].text\n",
    "        split_text = raw_text.split(\"\\n\\n\")\n",
    "        header = split_text[0]\n",
    "        header = header.replace(\"#\", \"\").strip()\n",
    "        content = \"\\n\\n\".join(split_text[1:])\n",
    "        return header, content\n",
    "\n",
    "    def merge(self) -> str:\n",
    "        \"\"\"\n",
    "        Perform the entire merge process and return the final markdown.\n",
    "        \"\"\"\n",
    "        sections1 = self.parse_markdown(self.version1)\n",
    "        sections2 = self.parse_markdown(self.version2)\n",
    "        \n",
    "        matched_sections = self.match_sections(sections1, sections2)\n",
    "        \n",
    "        merged_sections = []\n",
    "        for s1, s2 in matched_sections:\n",
    "            level = min(s1[0], s2[0])  # Use the lower level (higher in hierarchy)\n",
    "            header = s1[1] if len(s1[1]) >= len(s2[1]) else s2[1]\n",
    "            header, content = self.merge_content(f\"{'#' * level} {header}\", s1[2], s2[2])\n",
    "            merged_sections.append((level, header, content))\n",
    "        \n",
    "        # Sort merged sections based on their order in version1\n",
    "        merged_sections.sort(key=lambda x: sections1.index(x[0]) if x[0] in sections1 else len(sections1))\n",
    "        \n",
    "        # Generate final markdown\n",
    "        final_markdown = \"\"\n",
    "        for level, header, content in merged_sections:\n",
    "            final_markdown += f\"{'#' * level} {header}\\n\\n{content}\\n\\n\"\n",
    "        \n",
    "        return final_markdown.strip()\n",
    "\n",
    "# Usage\n",
    "# merger = MarkdownMerger(version1, version2)\n",
    "# final_markdown = merger.merge()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation<sup>3</sup>. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials<sup>4–6</sup>. This confines light-dependent activation to a small volume of brain tissue [~1 mm<sup>3</sup>]<sup>7</sup>. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood<sup>8,9</sup>. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible<sup>10</sup>. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties<sup>6</sup>.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature<sup>11,12</sup>, variants engineered through recombination<sup>9,13</sup> and mutagenesis<sup>14,15</sup>, as well as variants resulting from rational design<sup>16</sup>. Studies of these coupled with structural information<sup>17</sup> and molecular dynamic simulations<sup>18</sup> have established some understanding of the mechanics and sequence features important for specific ChR properties<sup>1,16</sup>. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB<sup>19</sup>. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "## Results\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly- functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off- kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Figure 2\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs areexpressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "## Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "## Online methods\n",
      "\n",
      "### Figure 1\n",
      "\n",
      "Schematic representation of the experimental setup. The sample is illuminated by a laser beam, and the scattered light is collected by a lens and detected by a photodetector. The signal from the photodetector is then processed by a data acquisition system and analyzed using a computer.\n",
      "\n",
      "### Figure 2\n",
      "\n",
      "Typical time-resolved fluorescence decay curve obtained from the sample. The experimental data (black dots) are fitted with a multi-exponential decay function (red line) to extract the fluorescence lifetime components.\n",
      "\n",
      "### Figure 3\n",
      "\n",
      "Fluorescence lifetime imaging (FLIM) map of the sample. The color scale represents the fluorescence lifetime at each pixel, providing information about the local environment and molecular interactions within the sample.\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs before photocurrent measurements. Primary hippocampal neuronal cultures were prepped from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings were performed 5–10 days after transduction.\n",
      "\n",
      "## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "## Animals\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23.\n",
      "For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/ muscle/).\n",
      "For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2).\n",
      "Structural encodings (i.e., the contact map) use the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure.\n",
      "Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d).\n",
      "The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions.\n",
      "For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')].\n",
      "Both sequence and structural information were one-hot encoded.\n",
      "Regression models for ChR properties were trained to predict the logarithm of the measured properties.\n",
      "All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences.\n",
      "Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2).\n",
      "The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1).\n",
      "The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d).\n",
      "For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "GP regression—In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn 2, the posterior distribution of f\n",
      "*\n",
      "= f x* for Gaussian process regression is Gaussian with mean f and variance\n",
      "\n",
      "- * = k* T\n",
      "K\n",
      "+ σn 2I −1y\n",
      "v* = k x*, x* − k* T\n",
      "K\n",
      "+ σn 2I −1k*\n",
      "\n",
      "- (2)\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set.\n",
      "k* is the vector of covariances between the novel input and each input in the training set, and k*i = k x*, xi.\n",
      "The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\n",
      "\n",
      "log p y|X =\n",
      "−\n",
      "1 2 yT K + σn 2I −1y − 1 2log K + σn 2I − n 2log 2π (3) k s, s′ = σp 2 exp −\n",
      "\n",
      "where n is the dimensionality of the inputs.\n",
      "Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "GP classification—In binary classification, instead of continuous outputs y, the outputs are class labels yi ∈ {+1, −1}, and the goal is to use the training data to make probabilistic predictions π x* = p y* =\n",
      "+\n",
      "1 x*.\n",
      "We use Laplace's method to approximate the posterior distribution.\n",
      "Hyperparameters in the kernels are found by maximizing the marginal likelihood.\n",
      "Classification was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "The binary classification model was trained to predict if a ChR sequence is or is not functional.\n",
      "A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "GP kernels for modeling proteins—Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences.\n",
      "A protein sequence s of length L is defined by the amino acid present at each location.\n",
      "This can be encoded as a binary feature vector xse that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids).\n",
      "Likewise, the protein's structure can be represented as a residue-residue contact map.\n",
      "The contact map can be encoded as a binary feature vector xst that indicates the presence or absence of each possible contacting pair.\n",
      "Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions k(si, sj) were considered: polynomial kernels, squared exponential kernels, and Matérn kernels.\n",
      "These different forms represent possible functions for the protein's fitness landscape.\n",
      "The polynomial kernel is defined as: k s, s′ = σ0 2 + σp 2xTx′ d where σ0 and σp are hyperparameters.\n",
      "We considered polynomial kernels with d = 3.\n",
      "The squared exponential kernel is defined as:\n",
      "\n",
      "- (4)\n",
      "- x − x′ 2\n",
      "2 2 2\n",
      "\n",
      "l k s, s′ =\n",
      "1\n",
      "+ l\n",
      "\n",
      "- (5)\n",
      "where l and σp are also hyperparameters and |⋅|2 is the L2 norm.\n",
      "Finally, the Matérn kernel with v\n",
      "=\n",
      "5 2 is defined as:\n",
      "\n",
      "- (6)\n",
      "5\n",
      "x\n",
      "− x′ 2 2\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01.\n",
      "For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "## Data availability\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "## Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions.\n",
      "This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848.\n",
      "Additional funding includes the NSF NeuroNex Technology Hub grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.).\n",
      "The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred.\n",
      "C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913.\n",
      "J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "## References\n",
      "\n",
      "- 1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "- 2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "- 3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "- 4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "- 5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "- 6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "- 7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "- 8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "- 9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "- 10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "- 11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "- 12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "- 13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "- 14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "- 15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "- 16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "- 17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "- 18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\n",
      "- 19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "- 20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "- 21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "- 22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "- 23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "- 24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "- 25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "- 26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "- 27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "- 28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current\n",
      "\n",
      "## Figure 1.\n",
      "\n",
      "Machine learning-guided optimization of ChRs.\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function). (c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes). (b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization, e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.\n",
      "\n",
      "## Figure 2.\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "(d) Peak and steady-state photocurrent strength with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells).\n",
      "(e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.\n",
      "\n",
      "## Figure 3.\n",
      "\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "(a) ChRs were cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "(b) Cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter (repeated independently six times per construct with similar results).\n",
      "(c) Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate-intensity (0.8 mW mm−2) light in cultured neurons (ChR2, n = 16 cells; CoChR, n = 17 cells; ChRger1, n = 9 cells; ChRger2, n = 24 cells; ChRger3, n = 9 cells).\n",
      "(d) Spike fidelity with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1, n = 6 cells; ChRger2, n = 6 cells; ChRger3, n = 6 cells; CoChR, n = 7 cells; ChR2, n = 7 cells).\n",
      "(e) Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons (ChRger1, n = 9 cells; ChRger2, n = 12 cells; ChRger3, n = 7 cells; ChR2, n = 8 cells).\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "(g) Photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC (ChRger1, n = 11 cells; ChRger2, n = 11 cells; ChRger3, n = 11 cells; ChR2, n = 9 cells) or after systemic delivery of CamKIIa-ChRger2 (ChRger2, n = 6 cells; 5×1011 vg/animal).\n",
      "(h) Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at varying light irradiances in acute brain slice after direct injection.\n",
      "(i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×1011 vg/animal; repeated independently five times per construct with similar results).\n",
      "(j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×1011 vg/animal).\n",
      "(k) Peak (solid line) and steady-state (dashed line) photocurrent strength with varying light irradiances in acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons).\n",
      "(l) Spike fidelity with varying light irradiances in acute brain slice after systemic delivery (1×1011 vg/animal)\n"
     ]
    }
   ],
   "source": [
    "merger_test = MarkdownMerger(doc_sherpa, full_text)\n",
    "final_markdown = merger_test.merge()\n",
    "print(final_markdown)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "from typing import List, Dict, Tuple\n",
    "from difflib import SequenceMatcher\n",
    "import re\n",
    "from llama_index.core import Settings\n",
    "from llama_index.llms.anthropic import Anthropic\n",
    "\n",
    "class MarkdownNode:\n",
    "    def __init__(self, level: int, header: str, content: str = \"\"):\n",
    "        self.level = level\n",
    "        self.header = header\n",
    "        self.content = content\n",
    "        self.children = []\n",
    "\n",
    "    def add_child(self, child: 'MarkdownNode'):\n",
    "        self.children.append(child)\n",
    "\n",
    "class MarkdownMerger:\n",
    "    def __init__(self, version1: str, version2: str):\n",
    "        self.version1 = version1\n",
    "        self.version2 = version2\n",
    "        self.llm = Settings.llm or Anthropic(model=\"claude-3-haiku-20240307\")\n",
    "\n",
    "    def parse_markdown_to_tree(self, markdown: str) -> List[MarkdownNode]:\n",
    "        lines = markdown.split('\\n')\n",
    "        root = MarkdownNode(0, \"Root\")\n",
    "        stack = [root]\n",
    "        current_content = []\n",
    "\n",
    "        for line in lines:\n",
    "            if line.startswith('#'):\n",
    "                if current_content:\n",
    "                    stack[-1].content = '\\n'.join(current_content)\n",
    "                    current_content = []\n",
    "\n",
    "                level = len(line.split()[0])\n",
    "                header = line.lstrip('#').strip()\n",
    "                node = MarkdownNode(level, header)\n",
    "\n",
    "                while stack[-1].level >= level:\n",
    "                    stack.pop()\n",
    "                \n",
    "                stack[-1].add_child(node)\n",
    "                stack.append(node)\n",
    "            else:\n",
    "                current_content.append(line)\n",
    "\n",
    "        if current_content:\n",
    "            stack[-1].content = '\\n'.join(current_content)\n",
    "\n",
    "        return root.children\n",
    "\n",
    "    def text_similarity(self, text1: str, text2: str) -> float:\n",
    "        return SequenceMatcher(None, text1, text2).ratio()\n",
    "\n",
    "    def find_best_match(self, node: MarkdownNode, candidates: List[MarkdownNode]) -> Tuple[MarkdownNode, float]:\n",
    "        best_match = None\n",
    "        best_score = 0\n",
    "\n",
    "        for candidate in candidates:\n",
    "            node_text = node.header + \"\\n\\n\" + node.content\n",
    "            candidate_text = candidate.header + \"\\n\\n\" + candidate.content\n",
    "            similarity_score = self.text_similarity(node_text, candidate_text)\n",
    "\n",
    "            # if combined_score > 0.9:\n",
    "            #     continue\n",
    "            if similarity_score > best_score:\n",
    "                best_score = similarity_score\n",
    "                best_match = candidate\n",
    "\n",
    "        return best_match, best_score\n",
    "\n",
    "    def merge_trees(self, tree1: List[MarkdownNode], tree2: List[MarkdownNode]) -> List[MarkdownNode]:\n",
    "        merged = []\n",
    "        used2 = set()\n",
    "\n",
    "        if len(tree1) == 0:\n",
    "            if len(tree2) == 0:\n",
    "                return []\n",
    "            return tree2\n",
    "        if len(tree2) == 0:\n",
    "            return tree1\n",
    "\n",
    "        for node1 in tree1:\n",
    "            best_match, score = self.find_best_match(node1, tree2)\n",
    "            print(score)\n",
    "            if score > 0.2:  # Adjusted threshold\n",
    "                merged_node = self.merge_nodes(node1, best_match)\n",
    "                merged.append(merged_node)\n",
    "                used2.add(best_match)\n",
    "            else:\n",
    "                merged.append(node1)\n",
    "\n",
    "        merged = sorted(merged, key=lambda x: x.level)\n",
    "\n",
    "        # Add remaining nodes from tree2\n",
    "        unused = []\n",
    "        for node2 in tree2:\n",
    "            if node2 not in used2:\n",
    "                unused.append(node2)\n",
    "\n",
    "        if len(unused) == 0 or len(used2) == 0:\n",
    "            return merged\n",
    "        else:\n",
    "            return self.merge_trees(merged, unused)\n",
    "\n",
    "\n",
    "    def merge_nodes(self, node1: MarkdownNode, node2: MarkdownNode) -> MarkdownNode:\n",
    "        merged_header, merged_content = self.merge_content(node1.header, node1.content, node2.content)\n",
    "        merged_node = MarkdownNode(min(node1.level, node2.level), merged_header, merged_content)\n",
    "        merged_node.children = self.merge_trees(node1.children, node2.children)\n",
    "        return merged_node\n",
    "\n",
    "    def merge_content(self, header: str, content1: str, content2: str) -> Tuple[str, str]:\n",
    "        try:\n",
    "            return self._merge_content_llm(header, content1, content2)\n",
    "        except Exception as e:\n",
    "            print(f\"LLM-based merging failed: {e}\")\n",
    "            return header, self._merge_content_length_based(content1, content2)\n",
    "\n",
    "    def _merge_content_llm(self, header: str, content1: str, content2: str) -> Tuple[str, str]:\n",
    "        prompt = f\"\"\"Below are two different markdown strings extracted from a scientific paper via different parsers.\n",
    "        Your job is to merge the two markdown strings into a single markdown string that is your best estimate of the original text.\n",
    "        Keep your eye out for text that could be figure captions. The parsers struggle with these. You may need to add headers or recombine text to best represent figure captions.\n",
    "        The goal would be for each figure caption to be its own section.\n",
    "        \n",
    "        SECTION HEADER TITLE:\n",
    "        {header}\n",
    "\n",
    "        PARSED CONTENT 1:\n",
    "        {content1}\n",
    "\n",
    "        PARSED CONTENT 2:\n",
    "        {content2}\n",
    "\n",
    "        Return only the final markdown string without any other text.\n",
    "        \"\"\"\n",
    "\n",
    "        response = self.llm.complete(prompt)\n",
    "        raw_text = response.text\n",
    "        split_text = raw_text.split(\"\\n\\n\")\n",
    "        merged_header = split_text[0].replace(\"#\", \"\").strip()\n",
    "        merged_content = \"\\n\\n\".join(split_text[1:])\n",
    "        return merged_header, merged_content\n",
    "\n",
    "    def _merge_content_length_based(self, content1: str, content2: str) -> str:\n",
    "        paragraphs1 = content1.split('\\n\\n')\n",
    "        paragraphs2 = content2.split('\\n\\n')\n",
    "        \n",
    "        merged = []\n",
    "        for p1, p2 in zip(paragraphs1, paragraphs2):\n",
    "            merged.append(p1 if len(p1) >= len(p2) else p2)\n",
    "        \n",
    "        merged.extend(paragraphs1[len(paragraphs2):])\n",
    "        merged.extend(paragraphs2[len(paragraphs1):])\n",
    "        \n",
    "        return '\\n\\n'.join(merged)\n",
    "\n",
    "    def tree_to_markdown(self, nodes: List[MarkdownNode], level: int = 0) -> str:\n",
    "        markdown = \"\"\n",
    "        for node in nodes:\n",
    "            markdown += f\"{'#' * node.level} {node.header}\\n\\n{node.content}\\n\\n\"\n",
    "            if node.children:\n",
    "                markdown += self.tree_to_markdown(node.children, level + 1)\n",
    "        return markdown\n",
    "\n",
    "    def merge(self) -> str:\n",
    "        tree1 = self.parse_markdown_to_tree(self.version1)\n",
    "        tree2 = self.parse_markdown_to_tree(self.version2)\n",
    "        merged_tree = self.merge_trees(tree1, tree2)\n",
    "        return self.tree_to_markdown(merged_tree)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.29936305732484075\n",
      "0.9618768328445748\n",
      "0.8208304657031429\n",
      "0.045004500450045004\n",
      "0.04619565217391304\n",
      "0.026059763724808897\n",
      "0.04895416110369381\n",
      "0.03146853146853147\n",
      "0.07520564042303172\n",
      "0.029097239492663518\n",
      "0.992176962503372\n",
      "1.0\n",
      "0.9951592356687898\n",
      "0.7317654063709438\n",
      "0.07590132827324478\n",
      "0.34151472650771386\n",
      "0.07590132827324478\n",
      "0.04267931238885596\n",
      "0.07111756168359942\n",
      "0.08636363636363636\n",
      "0.18412698412698414\n",
      "0.05026782035434693\n",
      "0.04686899731079524\n",
      "0.03941730934018852\n",
      "0.10304449648711944\n",
      "0.03837471783295711\n",
      "0.06486486486486487\n",
      "0.1111111111111111\n",
      "0.15129151291512916\n",
      "0.3438914027149321\n",
      "1.0\n",
      "0.992377495462795\n",
      "0.9424529909674487\n",
      "0.03523439832785906\n",
      "0.012759827124922823\n",
      "0.019455252918287938\n",
      "0.004704775346977182\n",
      "0.012091898428053204\n",
      "0.006868560724320949\n",
      "0.00516795865633075\n",
      "0.007182761372705506\n",
      "0.006676403087836428\n",
      "0.010198878123406425\n",
      "0.004279382546232615\n",
      "0.0096514745308311\n",
      "0.042042042042042045\n",
      "0.01588271227855834\n",
      "0.014047410008779631\n",
      "0.08362369337979095\n",
      "0.008190618019359643\n",
      "0.009783368273934312\n",
      "0.014025245441795231\n",
      "0.019766397124887692\n",
      "0.009473060982830076\n",
      "0.01287001287001287\n",
      "0.020486555697823303\n",
      "0.02746566791510612\n",
      "0.049019607843137254\n",
      "0.10101010101010101\n",
      "0.009237875288683603\n",
      "0.0034626038781163434\n",
      "0.006097560975609756\n",
      "0.0021559468199784403\n",
      "0.04710144927536232\n"
     ]
    }
   ],
   "source": [
    "merger = MarkdownMerger(full_text, doc_sherpa)\n",
    "final_markdown = merger.merge()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>†</sup>These authors contributed equally to this work\n",
      "\n",
      "<sup>*</sup>Corresponding authors\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 10<sup>5</sup>–10<sup>6</sup> functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation<sup>3</sup>. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials<sup>4–6</sup>. This confines light-dependent activation to a small volume of brain tissue [~1 mm<sup>3</sup>]<sup>7</sup>. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood<sup>8,9</sup>. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible<sup>10</sup>. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties<sup>6</sup>.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature<sup>11,12</sup>, variants engineered through recombination<sup>9,13</sup> and mutagenesis<sup>14,15</sup>, as well as variants resulting from rational design<sup>16</sup>. Studies of these coupled with structural information<sup>17</sup> and molecular dynamic simulations<sup>18</sup> have established some understanding of the mechanics and sequence features important for specific ChR properties<sup>1,16</sup>. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB<sup>19</sup>. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs areexpressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "\n",
      "## Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "## As there is no content provided in either PARSED CONTENT 1 or PARSED CONTENT 2, I cannot merge or create a final markdown string. The only information available is the section header title \"Online methods\". Without any actual content to work with, I cannot generate a meaningful response that meets the requested criteria.\n",
      "\n",
      "\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience.\n",
      "\n",
      "## Figure Caption\n",
      "Supplemental Figure 1c,d: For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37.\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization of ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs before photocurrent measurements.\n",
      "\n",
      "Primary hippocampal neuronal cultures were prepped from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings were performed 5–10 days after transduction.\n",
      "\n",
      "## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5).\n",
      "\n",
      "To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated inphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "\n",
      "## Animals\n",
      "\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) usethe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "\n",
      "## GP regression\n",
      "\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "\n",
      "## GP classification\n",
      "\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "\n",
      "## L1 regression feature identification and weighting\n",
      "\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "\n",
      "## Data availability\n",
      "\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "\n",
      "## Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions.\n",
      "\n",
      "This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848.\n",
      "\n",
      "Additional funding includes the NSF NeuroNex Technology Hub grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.).\n",
      "\n",
      "The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred.\n",
      "\n",
      "C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913.\n",
      "\n",
      "J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "## References\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\n",
      "\n",
      "19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near\n",
      "\n",
      "### 45. Hunter JD Matplotlib: A 2D Graphics Environment\n",
      "\n",
      "\n",
      "Hunter JD Matplotlib: A 2D Graphics Environment.\n",
      "Computing in Science and Engineering 9, 90– 95 (2007).\n",
      "\n",
      "\n",
      "### 46. Oliphant TE Python for Scientific Computing\n",
      "\n",
      "\n",
      "Oliphant TE Python for Scientific Computing.\n",
      "Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "- 47.\n",
      "Pedregosa F et al.\n",
      "Scikit-learn: Machine Learning in Python.\n",
      "Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "### Figure 1.\n",
      "\n",
      "\n",
      "Machine learning-guided optimization of ChRs.\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current.\n",
      "We use both peak and steady-state current as metrics for photocurrent strength.\n",
      "To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal.\n",
      "As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set).\n",
      "Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization, e.g., the 28 top variants.\n",
      "(d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants.\n",
      "(e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2).\n",
      "Each gray-colored point is a ChR variant.\n",
      "Training set data are shaded in blue.\n",
      "Mean number of mutations for each set is below the plots.\n",
      "(f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs.\n",
      "R represents the Pearson correlation coefficient.\n",
      "\n",
      "Figure 2.\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells.\n",
      "Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2).\n",
      "397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2).\n",
      "Parent ChRs are highlighted in light gray.\n",
      "Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "(d) Peak and steady-state photocurrent strength with varying light irradiances compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "5 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 28_10, n\n",
      "=\n",
      "5 cells; 11_10, n\n",
      "=\n",
      "5 cells; 25_9, n\n",
      "=\n",
      "5 cells).\n",
      "(e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "6 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 11_10, n\n",
      "=\n",
      "6 cells; 12_10, n\n",
      "=\n",
      "7 cells; 25_9, n\n",
      "=\n",
      "5 cells; 10_10, n\n",
      "=\n",
      "4 cells).\n",
      "Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 3.\n",
      "\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "(a) ChRs were cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "(b) Cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter (repeated independently six times per construct with similar results).\n",
      "(c) Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate- intensity (0.8 mW mm−2) light in cultured neurons (ChR2, n = 16 cells; CoChR, n = 17 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 24 cells; ChRger3, n\n",
      "=\n",
      "9 cells).\n",
      "(d) Spike fidelity with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1, n\n",
      "=\n",
      "6 cells; ChRger2, n\n",
      "=\n",
      "6 cells; ChRger3, n\n",
      "=\n",
      "6 cells; CoChR, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "7 cells).\n",
      "(e) Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons (ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 12 cells; ChRger3, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "8 cells).\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "(g) Photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC (ChRger1, n = 11 cells; ChRger2, n = 11 cells; ChRger3, n = 11 cells; ChR2, n\n",
      "=\n",
      "9 cells) or after systemic delivery of CamKIIa-ChRger2 (ChRger2, n\n",
      "=\n",
      "6 cells; 5×1011 vg/animal).\n",
      "(h) Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at varying light irradiances in acute brain slice after direct injection.\n",
      "(i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×1011 vg/animal; repeated independently five times per construct with similar results).\n",
      "(j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×1011 vg/animal).\n",
      "Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons).\n",
      "(m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×1011 vg/animal) (CoChR, n = 15 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n\n",
      "=\n",
      "5 cells; ChRger3, n\n",
      "=\n",
      "8 cells) with 1 mW mm−2 intensity light.\n",
      "(n) Representative voltage traces with blue light–driven (1 mW mm−2) spiking at the indicated frequencies.\n",
      "vg, viral genomes.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×1011 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies.\n",
      "ChRger2, n\n",
      "=\n",
      "4 animals; ChR2(H134R), n\n",
      "=\n",
      "4 animals.\n",
      "Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×1011 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface.\n",
      "Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity.\n",
      "Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right).\n",
      "Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW).\n",
      "ChRger2, n\n",
      "=\n",
      "5 animals; ChR2(H134R), n\n",
      "=\n",
      "5 animals.\n",
      "No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW).\n",
      "Plotted data are mean±SEM.\n",
      "vg, viral genomes.\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\n",
      "\n",
      "The figure contains several parts:\n",
      "\n",
      "a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\n",
      "\n",
      "b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\n",
      "\n",
      "c. Train regression models graphs for current strength, off-kinetics, and spectral properties.\n",
      "\n",
      "d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\n",
      "\n",
      "e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\n",
      "\n",
      "f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\n",
      "\n",
      "The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.Figure 2\n",
      "\n",
      "\n",
      "## Figure 2.\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strength\n",
      "\n",
      "The figure contains several panels:\n",
      "\n",
      "a) A series of current traces and corresponding cell images for different ChR variants.\n",
      "\n",
      "b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\n",
      "\n",
      "c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\n",
      "\n",
      "d) Line graphs showing normalized photocurrent vs wavelength.\n",
      "\n",
      "e) Line graphs showing peak and steady-state photocurrent vs light intensity.\n",
      "\n",
      "Each panel contains detailed data comparing various ChR variants across different properties and conditions.with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.## Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(final_markdown)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "merger = MarkdownMerger(final_markdown, final_markdown)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_markdown_self = merger.merge()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# HHS Public Access\n",
      "\n",
      "\n",
      "\n",
      "## Nat Methods. Author manuscript; available in PMC 2020 April 14.\n",
      "\n",
      "\n",
      "\n",
      "Published in final edited form as: Nat Methods.\n",
      "2019 November ; 16(11): 1176–1184.\n",
      "doi:10.1038/s41592-019-0583-8.\n",
      "\n",
      "\n",
      "\n",
      "# Since both PARSED CONTENT 1 and PARSED CONTENT 2 are empty in this case, there is no content to merge or process. As a result, there is no final markdown string to return.\n",
      "\n",
      "\n",
      "\n",
      "## Nat Methods. Author manuscript; available in PMC 2020 April 14.\n",
      "\n",
      "\n",
      "\n",
      "Published in final edited form as: Nat Methods.\n",
      "2019 November ; 16(11): 1176–1184.\n",
      "doi:10.1038/s41592-019-0583-8.\n",
      "\n",
      "\n",
      "\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "\n",
      "## Claire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana Gradinaru1,*, Frances H. Arnold1,2,*\n",
      "\n",
      "\n",
      "\n",
      "1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA 2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "\n",
      "\n",
      "## Abstract\n",
      "\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "\n",
      "## Introduction\n",
      "\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 10<sup>5</sup>–10<sup>6</sup> functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation<sup>3</sup>. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials<sup>4–6</sup>. This confines light-dependent activation to a small volume of brain tissue [~1 mm<sup>3</sup>]<sup>7</sup>. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood<sup>8,9</sup>. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible<sup>10</sup>. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties<sup>6</sup>.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature<sup>11,12</sup>, variants engineered through recombination<sup>9,13</sup> and mutagenesis<sup>14,15</sup>, as well as variants resulting from rational design<sup>16</sup>. Studies of these coupled with structural information<sup>17</sup> and molecular dynamic simulations<sup>18</sup> have established some understanding of the mechanics and sequence features important for specific ChR properties<sup>1,16</sup>. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB<sup>19</sup>. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "\n",
      "## As there is no content provided in either PARSED CONTENT 1 or PARSED CONTENT 2, I cannot merge or create a final markdown string. The result would simply be the section header:\n",
      "\n",
      "\n",
      "\n",
      "# As there is no content provided in either PARSED CONTENT 1 or PARSED CONTENT 2, I cannot merge or create a final markdown string. The result would simply be the section header:\n",
      "\n",
      "# Results\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Figure 1: Gaussian process (GP) classification and regression models for ChR engineering\n",
      "\n",
      "#### 4.5 Å in the C1C2 crystal structure (3UG9\n",
      "\n",
      "\n",
      "\n",
      "We defined the sequence and structural similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1).\n",
      "We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b).\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c).\n",
      "Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants.\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2).\n",
      "Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red- shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data- driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single- block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents.\n",
      "\n",
      "A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate1. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed14,27. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position 195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position.\n",
      "\n",
      "Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2).\n",
      "\n",
      "In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10).\n",
      "\n",
      "Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9).\n",
      "\n",
      "## Figure 3b\n",
      "When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites.\n",
      "\n",
      "## Figure 3c\n",
      "The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures.\n",
      "\n",
      "## Figure 3d, Supplemental Figure 8h\n",
      "The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation.\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices.\n",
      "\n",
      "## Figure 3g,h\n",
      "We observed greater light sensitivity for the ChRgers compared with ChR2(H134R). The ChRgers exhibit >200 pA photocurrent at 10^-3 mW mm^-2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10^-2 mW mm^-2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents.\n",
      "\n",
      "Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Discussion\n",
      "\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "\n",
      "## Since both PARSED CONTENT 1 and PARSED CONTENT 2 are empty in this case, there is no content to merge or process. The only information provided is the section header title. Therefore, the final markdown string would simply be:\n",
      "\n",
      "\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions.\n",
      "\n",
      "This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848.\n",
      "\n",
      "Additional funding includes the NSF NeuroNex Technology Hub grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.).\n",
      "\n",
      "The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred.\n",
      "\n",
      "C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913.\n",
      "\n",
      "J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "\n",
      "## References\n",
      "\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi:10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\n",
      "\n",
      "19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi:10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi:10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrare\n",
      "\n",
      "\n",
      "### 45. Hunter JD Matplotlib: A 2D Graphics Environment\n",
      "\n",
      "\n",
      "\n",
      "Hunter JD Matplotlib: A 2D Graphics Environment.\n",
      "Computing in Science and Engineering 9, 90– 95 (2007).\n",
      "\n",
      "\n",
      "\n",
      "### 46. Oliphant TE Python for Scientific Computing\n",
      "\n",
      "\n",
      "\n",
      "Oliphant TE Python for Scientific Computing.\n",
      "Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "- 47.\n",
      "Pedregosa F et al.\n",
      "Scikit-learn: Machine Learning in Python.\n",
      "Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "\n",
      "### Figure 1.\n",
      "\n",
      "\n",
      "\n",
      "Machine learning-guided optimization of ChRs.\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current.\n",
      "We use both peak and steady-state current as metrics for photocurrent strength.\n",
      "To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal.\n",
      "As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set).\n",
      "Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization, e.g., the 28 top variants.\n",
      "(d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants.\n",
      "(e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2).\n",
      "Each gray-colored point is a ChR variant.\n",
      "Training set data are shaded in blue.\n",
      "Mean number of mutations for each set is below the plots.\n",
      "(f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs.\n",
      "R represents the Pearson correlation coefficient.\n",
      "\n",
      "Figure 2.\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells.\n",
      "Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2).\n",
      "397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2).\n",
      "Parent ChRs are highlighted in light gray.\n",
      "Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "(d) Peak and steady-state photocurrent strength with varying light irradiances compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "5 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 28_10, n\n",
      "=\n",
      "5 cells; 11_10, n\n",
      "=\n",
      "5 cells; 25_9, n\n",
      "=\n",
      "5 cells).\n",
      "(e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "6 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 11_10, n\n",
      "=\n",
      "6 cells; 12_10, n\n",
      "=\n",
      "7 cells; 25_9, n\n",
      "=\n",
      "5 cells; 10_10, n\n",
      "=\n",
      "4 cells).\n",
      "Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 3.\n",
      "\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "(a) ChRs were cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "(b) Cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter (repeated independently six times per construct with similar results).\n",
      "(c) Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate- intensity (0.8 mW mm−2) light in cultured neurons (ChR2, n = 16 cells; CoChR, n = 17 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 24 cells; ChRger3, n\n",
      "=\n",
      "9 cells).\n",
      "(d) Spike fidelity with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1, n\n",
      "=\n",
      "6 cells; ChRger2, n\n",
      "=\n",
      "6 cells; ChRger3, n\n",
      "=\n",
      "6 cells; CoChR, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "7 cells).\n",
      "(e) Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons (ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 12 cells; ChRger3, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "8 cells).\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "(g) Photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC (ChRger1, n = 11 cells; ChRger2, n = 11 cells; ChRger3, n = 11 cells; ChR2, n\n",
      "=\n",
      "9 cells) or after systemic delivery of CamKIIa-ChRger2 (ChRger2, n\n",
      "=\n",
      "6 cells; 5×1011 vg/animal).\n",
      "(h) Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at varying light irradiances in acute brain slice after direct injection.\n",
      "(i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×1011 vg/animal; repeated independently five times per construct with similar results).\n",
      "(j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×1011 vg/animal).\n",
      "Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons).\n",
      "(m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×1011 vg/animal) (CoChR, n = 15 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n\n",
      "=\n",
      "5 cells; ChRger3, n\n",
      "=\n",
      "8 cells) with 1 mW mm−2 intensity light.\n",
      "(n) Representative voltage traces with blue light–driven (1 mW mm−2) spiking at the indicated frequencies.\n",
      "vg, viral genomes.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×1011 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies.\n",
      "ChRger2, n\n",
      "=\n",
      "4 animals; ChR2(H134R), n\n",
      "=\n",
      "4 animals.\n",
      "Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×1011 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface.\n",
      "Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity.\n",
      "Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right).\n",
      "Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW).\n",
      "ChRger2, n\n",
      "=\n",
      "5 animals; ChR2(H134R), n\n",
      "=\n",
      "5 animals.\n",
      "No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW).\n",
      "Plotted data are mean±SEM.\n",
      "vg, viral genomes.\n",
      "\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated inphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "\n",
      "\n",
      "## Animals\n",
      "\n",
      "\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) usethe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "\n",
      "\n",
      "## GP regression\n",
      "\n",
      "\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "\n",
      "\n",
      "## GP classification\n",
      "\n",
      "\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "\n",
      "\n",
      "## L1 regression feature identification and weighting\n",
      "\n",
      "\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "\n",
      "\n",
      "## Data availability\n",
      "\n",
      "\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "\n",
      "\n",
      "## Code availability\n",
      "\n",
      "\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\n",
      "\n",
      "The figure contains several parts:\n",
      "\n",
      "a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\n",
      "\n",
      "b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\n",
      "\n",
      "c. Train regression models graphs for current strength, off-kinetics, and spectral properties.\n",
      "\n",
      "d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\n",
      "\n",
      "e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\n",
      "\n",
      "f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\n",
      "\n",
      "The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.Figure 2\n",
      "\n",
      "\n",
      "\n",
      "## Figure 2.\n",
      "\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strength\n",
      "\n",
      "The figure contains several panels:\n",
      "\n",
      "a) A series of current traces and corresponding cell images for different ChR variants.\n",
      "\n",
      "b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\n",
      "\n",
      "c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\n",
      "\n",
      "d) Line graphs showing normalized photocurrent vs wavelength.\n",
      "\n",
      "e) Line graphs showing peak and steady-state photocurrent vs light intensity.\n",
      "\n",
      "Each panel contains detailed data comparing various ChR variants across different properties and conditions.with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.## Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n",
      "\n",
      "\n",
      "\n",
      "### Functional characterization of ChR variants for machine learning\n",
      "\n",
      "\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly- functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9.\n",
      "These recombination libraries are a rich source of functionally diverse sequences9.\n",
      "Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here.\n",
      "Together, these 102 ChR recombination variants provide the primary dataset used for model training.\n",
      "We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1).\n",
      "As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not.\n",
      "Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off- kinetics, we used these measured properties to train machine-learning models (Figure 1a).\n",
      "Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions.\n",
      "Different off-rates can be useful for specific applications: fast off- kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14).\n",
      "In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "\n",
      "\n",
      "### Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery.\n",
      "Previous reports of ‘non-invasive optogenetics’ relied on invasive intracranial virus delivery, which results in many copies of virus per cell and thus very high expression levels of the injected construct29.\n",
      "Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31.\n",
      "Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31.\n",
      "Use of rAAV- PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery.\n",
      "We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i).\n",
      "We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j).\n",
      "Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity.\n",
      "With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l).\n",
      "ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6).\n",
      "These results demonstrate the need for light-sensitive and high- photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter.\n",
      "With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g).\n",
      "When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n).\n",
      "ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst.\n",
      "CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32.\n",
      "We systemically delivered rAAV-PHP.eB packaging a double- floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5).\n",
      "Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke.\n",
      "Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner.\n",
      "Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a).\n",
      "Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33.\n",
      "In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side.\n",
      "We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5).\n",
      "We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11).\n",
      "We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive.\n",
      "Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2).\n",
      "Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1).\n",
      "Behavioral effects were seen at powers as low as 10 mW.\n",
      "To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins.\n",
      "20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R).\n",
      "Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally- invasive optogenetic excitation.\n",
      "Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "The culturing and characterization of ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs.\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization of ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs.\n",
      "\n",
      "## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36.\n",
      "\n",
      "Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs before photocurrent measurements.\n",
      "\n",
      "Primary hippocampal neuronal cultures were prepped from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings were performed 5–10 days after transduction.\n",
      "\n",
      "Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "\n",
      "### Patch-clamp electrophysiology\n",
      "\n",
      "\n",
      "\n",
      "Whole-cell patch-clamp and cell-attached recordings were performed in transfected HEK cells, transduced cultured neurons, and acute brain slices to measure light-activated inward currents or neuronal firing.\n",
      "For electrophysiological recordings, cultured cells were continuously perfused with extracellular solution at room temperature (in mM: 140 NaCl, 5 KCl, 10 HEPES, 2 MgCl2, 2 CaCl2, 10 glucose; pH 7.35) while mounted on the microscope stage.\n",
      "For slice recordings, 32 °C artificial cerebrospinal fluid (ACSF) was continuously perfused over slices.\n",
      "ACSF contained 127 mM NaCl, 2.5 mM KCl, 25 mM NaHCO3, 1.25 mM NaH2PO4, 12 mM d-glucose, 0.4 mM sodium ascorbate, 2 mM CaCl2, and 1 mM MgCl2 and was bubbled continuously with 95% oxygen / 5% CO2.\n",
      "Firing and photocurrent measurements were performed in the presence of 3 mM kynurenic acid and 100 μM picrotoxin to block optically evoked ionotropic glutamatergic and GABAergic currents, respectively.\n",
      "\n",
      "Patch pipettes were fabricated from borosilicate capillary glass tubing (1B150–4; World Precision Instruments) using a model P-2000 laser puller (Sutter Instruments) to resistances of 3–6 MΩ.\n",
      "Pipettes were filled with K-gluconate intracellular solution containing the following (in mM): 134 K gluconate, 5 EGTA, 10 HEPES, 2 MgCl2, 0.5 CaCl2, 3 ATP, and 0.2 GTP Whole-cell patch-clamp and cell-attached recordings were made using a Multiclamp 700B amplifier (Molecular Devices), a Digidata 1440 digitizer (Molecular Devices), and a PC running pClamp (version 10.4) software (Molecular Devices) to generate current injection waveforms and to record voltage and current traces.\n",
      "\n",
      "Photocurrents were recorded from cells in voltage clamp held at −60 mV.\n",
      "Neuronal firing was measured in current clamp mode with current injection for a −60 mV holding potential.\n",
      "Access resistance (Ra) and membrane resistance (Rm) were monitored throughout recording, and cells were discarded if Ra or Rm changed more than 15%.\n",
      "During ChR variant functional screening in HEK cells, photocurrents were only recorded from cells that passed our recording criteria: Rm > 200 MΩ and holding current >−100 pA. Our measured membrane properties of ChR expressing neurons were consistent with previous literature of opsin-expressing cells11 and are also consistent with previous reports of properties of cultured hippocampal neurons38,39 and PFC neurons in slice40,41 (Supplemental Figures 10).\n",
      "For cell culture experiments, the experimenter was blinded to the identity of the ChR being patched but not to the fluorescence level of the cells.\n",
      "For acute slice recordings, the experimenter was not blinded to the identity of the ChR.\n",
      "\n",
      "\n",
      "\n",
      "### Supplemental Figure 5\n",
      "\n",
      "The illumination/output spectra for each color were measured.\n",
      "\n",
      "\n",
      "### Figure 2e\n",
      "\n",
      "Full spectra measurements of photocurrents at seven different wavelengths.\n",
      "\n",
      "\n",
      "### Electrophysiology data analysis\n",
      "\n",
      "\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity.\n",
      "Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10).\n",
      "The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool.\n",
      "However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7.\n",
      "As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a).\n",
      "As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a).\n",
      "Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "\n",
      "\n",
      "### AAV production and purification\n",
      "\n",
      "\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31.\n",
      "Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI).\n",
      "Viral particles were harvested from the media and cells.\n",
      "Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%).\n",
      "Viruses were concentrated and formulated in phosphate buffered saline (PBS).\n",
      "Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "\n",
      "\n",
      "### Animals\n",
      "\n",
      "\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC).\n",
      "Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "\n",
      "\n",
      "### Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text.\n",
      "There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper.\n",
      "Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers.\n",
      "With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10).\n",
      "Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×1012 vg ml−1 targeting the following coordinates: anterior- posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2.\n",
      "For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm.\n",
      "For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm.\n",
      "The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface.\n",
      "Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H- LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC- ZF1.25) coupled to the fiber-optic cannula(e).\n",
      "Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse43 to track mouse nose, body, and tail position while the mouse was running on the treadmill.\n",
      "Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "\n",
      "\n",
      "### Gaussian process modeling\n",
      "\n",
      "\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23.\n",
      "For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/ muscle/).\n",
      "For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2).\n",
      "Structural encodings (i.e., the contact map) use the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure.\n",
      "Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d).\n",
      "The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions.\n",
      "For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [(‘A134’), (‘M1’)].\n",
      "Both sequence and structural information were one-hot encoded.\n",
      "Regression models for ChR properties were trained to predict the logarithm of the measured properties.\n",
      "All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences.\n",
      "Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2).\n",
      "The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1).\n",
      "The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d).\n",
      "For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "GP regression—In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn 2, the posterior distribution of f\n",
      "*\n",
      "= f x* for Gaussian process regression is Gaussian with mean f and variance\n",
      "\n",
      "- * = k* T\n",
      "K\n",
      "+ σn 2I −1y\n",
      "v* = k x*, x* − k* T\n",
      "K\n",
      "+ σn 2I −1k*\n",
      "\n",
      "- (2)\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set.\n",
      "k* is the vector of covariances between the novel input and each input in the training set, and k*i = k x*, xi.\n",
      "The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\n",
      "\n",
      "log p y|X =\n",
      "−\n",
      "1 2 yT K + σn 2I −1y − 1 2log K + σn 2I − n 2log 2π (3) k s, s′ = σp 2 exp −\n",
      "\n",
      "where n is the dimensionality of the inputs.\n",
      "Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "GP classification—In binary classification, instead of continuous outputs y, the outputs are class labels yi ∈ {+1, −1}, and the goal is to use the training data to make probabilistic predictions π x* = p y* =\n",
      "+\n",
      "1 x*.\n",
      "We use Laplace’s method to approximate the posterior distribution.\n",
      "Hyperparameters in the kernels are found by maximizing the marginal likelihood.\n",
      "Classification was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "The binary classification model was trained to predict if a ChR sequence is or is not functional.\n",
      "A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "GP kernels for modeling proteins—Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences.\n",
      "A protein sequence s of length L is defined by the amino acid present at each location.\n",
      "This can be encoded as a binary feature vector xse that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids).\n",
      "Likewise, the protein’s structure can be represented as a residue-residue contact map.\n",
      "The contact map can be encoded as a binary feature vector xst that indicates the presence or absence of each possible contacting pair.\n",
      "Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions k(si, sj) were considered: polynomial kernels, squared exponential kernels, and Matérn kernels.\n",
      "These different forms represent possible functions for the protein’s fitness landscape.\n",
      "The polynomial kernel is defined as: k s, s′ = σ0 2 + σp 2xTx′ d where σ0 and σp are hyperparameters.\n",
      "We considered polynomial kernels with d = 3.\n",
      "The squared exponential kernel is defined as:\n",
      "\n",
      "- (4)\n",
      "- x − x′ 2\n",
      "2 2 2\n",
      "\n",
      "l k s, s′ =\n",
      "1\n",
      "+ l\n",
      "\n",
      "- (5)\n",
      "where l and σp are also hyperparameters and |⋅|2 is the L2 norm.\n",
      "Finally, the Matérn kernel with v\n",
      "=\n",
      "5 2 is defined as:\n",
      "\n",
      "- (6)\n",
      "5\n",
      "x\n",
      "− x′ 2 2\n",
      "\n",
      "\n",
      "\n",
      "##### l +\n",
      "\n",
      "\n",
      "\n",
      "5\n",
      "x\n",
      "− x′ 3l2 exp −\n",
      "5\n",
      "x − x′ 2 2\n",
      "\n",
      "Where l is once again a hyperparameter.\n",
      "\n",
      "\n",
      "\n",
      "### Statistical analysis\n",
      "\n",
      "\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01.\n",
      "For statistical comparisons, we first performed a D’Agostino & Pearson normality test.\n",
      "If the p-value of a D’Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal- Wallis test with Dunn’s multiple comparisons post hoc test was used.\n",
      "If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "\n",
      "\n",
      "### Data availability\n",
      "\n",
      "\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files.\n",
      "Source data for classification model training are provided in Dataset 1 and Dataset 2.\n",
      "Source data for regression model training are provided in Dataset 2.\n",
      "DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "\n",
      "\n",
      "### Code availability\n",
      "\n",
      "\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/ fhalab/channels.\n",
      "\n",
      "\n",
      "\n",
      "# Since both PARSED CONTENT 1 and PARSED CONTENT 2 are empty in this case, there is no content to merge or process. The only information provided is the section header title. Therefore, the final markdown string would simply be:\n",
      "\n",
      "# Online methods\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions.\n",
      "\n",
      "This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848.\n",
      "\n",
      "Additional funding includes the NSF NeuroNex Technology Hub grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.).\n",
      "\n",
      "The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred.\n",
      "\n",
      "C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913.\n",
      "\n",
      "J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "\n",
      "## References\n",
      "\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi:10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\n",
      "\n",
      "19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi:10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi:10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrare\n",
      "\n",
      "\n",
      "### 45. Hunter JD Matplotlib: A 2D Graphics Environment\n",
      "\n",
      "\n",
      "\n",
      "Hunter JD Matplotlib: A 2D Graphics Environment.\n",
      "Computing in Science and Engineering 9, 90– 95 (2007).\n",
      "\n",
      "\n",
      "\n",
      "### 46. Oliphant TE Python for Scientific Computing\n",
      "\n",
      "\n",
      "\n",
      "Oliphant TE Python for Scientific Computing.\n",
      "Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "- 47.\n",
      "Pedregosa F et al.\n",
      "Scikit-learn: Machine Learning in Python.\n",
      "Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "\n",
      "### Figure 1.\n",
      "\n",
      "\n",
      "\n",
      "Machine learning-guided optimization of ChRs.\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current.\n",
      "We use both peak and steady-state current as metrics for photocurrent strength.\n",
      "To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal.\n",
      "As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set).\n",
      "Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization, e.g., the 28 top variants.\n",
      "(d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants.\n",
      "(e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2).\n",
      "Each gray-colored point is a ChR variant.\n",
      "Training set data are shaded in blue.\n",
      "Mean number of mutations for each set is below the plots.\n",
      "(f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs.\n",
      "R represents the Pearson correlation coefficient.\n",
      "\n",
      "Figure 2.\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells.\n",
      "Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2).\n",
      "397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2).\n",
      "Parent ChRs are highlighted in light gray.\n",
      "Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "(d) Peak and steady-state photocurrent strength with varying light irradiances compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "5 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 28_10, n\n",
      "=\n",
      "5 cells; 11_10, n\n",
      "=\n",
      "5 cells; 25_9, n\n",
      "=\n",
      "5 cells).\n",
      "(e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n\n",
      "=\n",
      "6 cells; CsChrimR, n\n",
      "=\n",
      "5 cells; C1C2, n\n",
      "=\n",
      "4 cells; 11_10, n\n",
      "=\n",
      "6 cells; 12_10, n\n",
      "=\n",
      "7 cells; 25_9, n\n",
      "=\n",
      "5 cells; 10_10, n\n",
      "=\n",
      "4 cells).\n",
      "Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 3.\n",
      "\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "(a) ChRs were cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "(b) Cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter (repeated independently six times per construct with similar results).\n",
      "(c) Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate- intensity (0.8 mW mm−2) light in cultured neurons (ChR2, n = 16 cells; CoChR, n = 17 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 24 cells; ChRger3, n\n",
      "=\n",
      "9 cells).\n",
      "(d) Spike fidelity with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1, n\n",
      "=\n",
      "6 cells; ChRger2, n\n",
      "=\n",
      "6 cells; ChRger3, n\n",
      "=\n",
      "6 cells; CoChR, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "7 cells).\n",
      "(e) Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons (ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n = 12 cells; ChRger3, n\n",
      "=\n",
      "7 cells; ChR2, n\n",
      "=\n",
      "8 cells).\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "(g) Photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC (ChRger1, n = 11 cells; ChRger2, n = 11 cells; ChRger3, n = 11 cells; ChR2, n\n",
      "=\n",
      "9 cells) or after systemic delivery of CamKIIa-ChRger2 (ChRger2, n\n",
      "=\n",
      "6 cells; 5×1011 vg/animal).\n",
      "(h) Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at varying light irradiances in acute brain slice after direct injection.\n",
      "(i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×1011 vg/animal; repeated independently five times per construct with similar results).\n",
      "(j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×1011 vg/animal).\n",
      "Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons).\n",
      "(m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×1011 vg/animal) (CoChR, n = 15 cells; ChRger1, n\n",
      "=\n",
      "9 cells; ChRger2, n\n",
      "=\n",
      "5 cells; ChRger3, n\n",
      "=\n",
      "8 cells) with 1 mW mm−2 intensity light.\n",
      "(n) Representative voltage traces with blue light–driven (1 mW mm−2) spiking at the indicated frequencies.\n",
      "vg, viral genomes.\n",
      "Plotted data are mean±SEM.\n",
      "\n",
      "Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×1011 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies.\n",
      "ChRger2, n\n",
      "=\n",
      "4 animals; ChR2(H134R), n\n",
      "=\n",
      "4 animals.\n",
      "Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×1011 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface.\n",
      "Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity.\n",
      "Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right).\n",
      "Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW).\n",
      "ChRger2, n\n",
      "=\n",
      "5 animals; ChR2(H134R), n\n",
      "=\n",
      "5 animals.\n",
      "No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW).\n",
      "Plotted data are mean±SEM.\n",
      "vg, viral genomes.\n",
      "\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Figure 1: Gaussian process (GP) classification and regression models for ChR engineering\n",
      "\n",
      "(a) Schematic of GP model training using ChR sequence/structure and functional data as inputs.\n",
      "(b) Binary classification model to predict ChR functionality.\n",
      "(c) Regression models for predicting photocurrent properties: strength, wavelength sensitivity, and off-kinetics.\n",
      "\n",
      "#### 4.5 Å in the C1C2 crystal structure (3UG9\n",
      "\n",
      "\n",
      "\n",
      "We defined the sequence and structural similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1).\n",
      "We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b).\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c).\n",
      "Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants.\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2).\n",
      "Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10^-3 mW mm^-2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10^-2 mW mm^-2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents.\n",
      "\n",
      "A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position.\n",
      "\n",
      "Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Figure 2: Functional characterization of designer ChRs\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs. In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10).\n",
      "\n",
      "Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties. ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities. These select designer ChRs are expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents.\n",
      "\n",
      "A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate1. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed14,27. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position 195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4).\n",
      "\n",
      "A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red- shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data- driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single- block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated inphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "\n",
      "\n",
      "## Animals\n",
      "\n",
      "\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) usethe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "\n",
      "\n",
      "## GP regression\n",
      "\n",
      "\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "\n",
      "\n",
      "## GP classification\n",
      "\n",
      "\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "\n",
      "\n",
      "## L1 regression feature identification and weighting\n",
      "\n",
      "\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "\n",
      "\n",
      "## Data availability\n",
      "\n",
      "\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "\n",
      "\n",
      "## Code availability\n",
      "\n",
      "\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\n",
      "\n",
      "The figure contains several parts:\n",
      "\n",
      "a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\n",
      "\n",
      "b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\n",
      "\n",
      "c. Train regression models graphs for current strength, off-kinetics, and spectral properties.\n",
      "\n",
      "d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\n",
      "\n",
      "e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\n",
      "\n",
      "f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\n",
      "\n",
      "The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.Figure 2\n",
      "\n",
      "\n",
      "\n",
      "## Figure 2.\n",
      "\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strength\n",
      "\n",
      "The figure contains several panels:\n",
      "\n",
      "a) A series of current traces and corresponding cell images for different ChR variants.\n",
      "\n",
      "b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\n",
      "\n",
      "c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\n",
      "\n",
      "d) Line graphs showing normalized photocurrent vs wavelength.\n",
      "\n",
      "e) Line graphs showing peak and steady-state photocurrent vs light intensity.\n",
      "\n",
      "Each panel contains detailed data comparing various ChR variants across different properties and conditions.with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.## Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n",
      "\n",
      "\n",
      "\n",
      "## Discussion\n",
      "\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "\n",
      "## Since both PARSED CONTENT 1 and PARSED CONTENT 2 are empty in this case, there is no content to merge or process. The only information provided is the section header title. Therefore, the final markdown string would simply be:\n",
      "\n",
      "\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36.\n",
      "\n",
      "Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs before photocurrent measurements.\n",
      "\n",
      "Primary hippocampal neuronal cultures were prepped from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings were performed 5–10 days after transduction.\n",
      "\n",
      "Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "\n",
      "### Patch-clamp electrophysiology\n",
      "\n",
      "\n",
      "\n",
      "Whole-cell patch-clamp and cell-attached recordings were performed in transfected HEK cells, transduced cultured neurons, and acute brain slices to measure light-activated inward currents or neuronal firing.\n",
      "For electrophysiological recordings, cultured cells were continuously perfused with extracellular solution at room temperature (in mM: 140 NaCl, 5 KCl, 10 HEPES, 2 MgCl2, 2 CaCl2, 10 glucose; pH 7.35) while mounted on the microscope stage.\n",
      "For slice recordings, 32 °C artificial cerebrospinal fluid (ACSF) was continuously perfused over slices.\n",
      "ACSF contained 127 mM NaCl, 2.5 mM KCl, 25 mM NaHCO3, 1.25 mM NaH2PO4, 12 mM d-glucose, 0.4 mM sodium ascorbate, 2 mM CaCl2, and 1 mM MgCl2 and was bubbled continuously with 95% oxygen / 5% CO2.\n",
      "Firing and photocurrent measurements were performed in the presence of 3 mM kynurenic acid and 100 μM picrotoxin to block optically evoked ionotropic glutamatergic and GABAergic currents, respectively.\n",
      "\n",
      "Patch pipettes were fabricated from borosilicate capillary glass tubing (1B150–4; World Precision Instruments) using a model P-2000 laser puller (Sutter Instruments) to resistances of 3–6 MΩ.\n",
      "Pipettes were filled with K-gluconate intracellular solution containing the following (in mM): 134 K gluconate, 5 EGTA, 10 HEPES, 2 MgCl2, 0.5 CaCl2, 3 ATP, and 0.2 GTP Whole-cell patch-clamp and cell-attached recordings were made using a Multiclamp 700B amplifier (Molecular Devices), a Digidata 1440 digitizer (Molecular Devices), and a PC running pClamp (version 10.4) software (Molecular Devices) to generate current injection waveforms and to record voltage and current traces.\n",
      "\n",
      "Photocurrents were recorded from cells in voltage clamp held at −60 mV.\n",
      "Neuronal firing was measured in current clamp mode with current injection for a −60 mV holding potential.\n",
      "Access resistance (Ra) and membrane resistance (Rm) were monitored throughout recording, and cells were discarded if Ra or Rm changed more than 15%.\n",
      "During ChR variant functional screening in HEK cells, photocurrents were only recorded from cells that passed our recording criteria: Rm > 200 MΩ and holding current >−100 pA. Our measured membrane properties of ChR expressing neurons were consistent with previous literature of opsin-expressing cells11 and are also consistent with previous reports of properties of cultured hippocampal neurons38,39 and PFC neurons in slice40,41 (Supplemental Figures 10).\n",
      "For cell culture experiments, the experimenter was blinded to the identity of the ChR being patched but not to the fluorescence level of the cells.\n",
      "For acute slice recordings, the experimenter was not blinded to the identity of the ChR.\n",
      "\n",
      "\n",
      "\n",
      "### Light delivery and imaging\n",
      "\n",
      "The culturing and characterization of ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs.\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "### Supplemental Figure 5\n",
      "\n",
      "The illumination/output spectra for each color were measured.\n",
      "\n",
      "\n",
      "### Figure 2e\n",
      "\n",
      "Full spectra measurements of photocurrents at seven different wavelengths.\n",
      "\n",
      "\n",
      "### Electrophysiology data analysis\n",
      "\n",
      "\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity.\n",
      "Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10).\n",
      "The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool.\n",
      "However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7.\n",
      "As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a).\n",
      "As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a).\n",
      "Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "\n",
      "\n",
      "### AAV production and purification\n",
      "\n",
      "\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31.\n",
      "Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI).\n",
      "Viral particles were harvested from the media and cells.\n",
      "Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%).\n",
      "Viruses were concentrated and formulated in phosphate buffered saline (PBS).\n",
      "Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "\n",
      "\n",
      "### Animals\n",
      "\n",
      "\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC).\n",
      "Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "\n",
      "\n",
      "### Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text.\n",
      "There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper.\n",
      "Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers.\n",
      "With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10).\n",
      "Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×1012 vg ml−1 targeting the following coordinates: anterior- posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2.\n",
      "For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm.\n",
      "For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm.\n",
      "The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface.\n",
      "Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H- LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC- ZF1.25) coupled to the fiber-optic cannula(e).\n",
      "Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse43 to track mouse nose, body, and tail position while the mouse was running on the treadmill.\n",
      "Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "\n",
      "\n",
      "### Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8.\n",
      "\n",
      "For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). Aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2).\n",
      "\n",
      "Structural encodings (i.e., the contact map) use the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "## GP regression\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn^2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean f and variance:\n",
      "\n",
      "f* = k*^T(K + σn^2I)^(-1)y\n",
      "v* = k(x*, x*) - k*^T(K + σn^2I)^(-1)k*\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\n",
      "\n",
      "log p(y|X) = -1/2 y^T(K + σn^2I)^(-1)y - 1/2 log|K + σn^2I| - n/2 log(2π)\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "## GP classification\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels yi ∈ {+1, −1}, and the goal is to use the training data to make probabilistic predictions π(x*) = p(y* = +1|x*). We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector xse that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector xst that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions k(si, sj) were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as: \n",
      "\n",
      "k(s, s') = (σ0^2 + σp^2x^Tx')^d \n",
      "\n",
      "where σ0 and σp are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "k(s, s') = σp^2 exp(-|x - x'|^2 / (2l^2))\n",
      "\n",
      "where l and σp are also hyperparameters and |⋅|2 is the L2 norm. Finally, the Matérn kernel with v = 5/2 is defined as:\n",
      "\n",
      "k(s, s') = σp^2(1 + √5|x - x'|/l + 5|x - x'|^2/(3l^2))exp(-√5|x - x'|/l)\n",
      "\n",
      "For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within a certain distance of each other.\n",
      "\n",
      "#### 4.5 Å in the C1C2 crystal structure (3UG9\n",
      "\n",
      "\n",
      "\n",
      "We defined the sequence and structural similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1).\n",
      "We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b).\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c).\n",
      "Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants.\n",
      "To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2).\n",
      "Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "\n",
      "\n",
      "##### l +\n",
      "\n",
      "\n",
      "\n",
      "5\n",
      "x\n",
      "− x′ 3l2 exp −\n",
      "5\n",
      "x − x′ 2 2\n",
      "\n",
      "Where l is once again a hyperparameter.\n",
      "\n",
      "\n",
      "\n",
      "### Statistical analysis\n",
      "\n",
      "\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01.\n",
      "For statistical comparisons, we first performed a D’Agostino & Pearson normality test.\n",
      "If the p-value of a D’Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal- Wallis test with Dunn’s multiple comparisons post hoc test was used.\n",
      "If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "\n",
      "\n",
      "### Data availability\n",
      "\n",
      "\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files.\n",
      "Source data for classification model training are provided in Dataset 1 and Dataset 2.\n",
      "Source data for regression model training are provided in Dataset 2.\n",
      "DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "\n",
      "\n",
      "### Code availability\n",
      "\n",
      "\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/ fhalab/channels.\n",
      "\n",
      "\n",
      "\n",
      "### Functional characterization of ChR variants for machine learning\n",
      "\n",
      "\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly- functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9.\n",
      "These recombination libraries are a rich source of functionally diverse sequences9.\n",
      "Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here.\n",
      "Together, these 102 ChR recombination variants provide the primary dataset used for model training.\n",
      "We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1).\n",
      "As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not.\n",
      "Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off- kinetics, we used these measured properties to train machine-learning models (Figure 1a).\n",
      "Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions.\n",
      "Different off-rates can be useful for specific applications: fast off- kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14).\n",
      "In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "\n",
      "\n",
      "### Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery.\n",
      "Previous reports of ‘non-invasive optogenetics’ relied on invasive intracranial virus delivery, which results in many copies of virus per cell and thus very high expression levels of the injected construct29.\n",
      "Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31.\n",
      "Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31.\n",
      "Use of rAAV- PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery.\n",
      "We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i).\n",
      "We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j).\n",
      "Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity.\n",
      "With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l).\n",
      "ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6).\n",
      "These results demonstrate the need for light-sensitive and high- photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter.\n",
      "With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g).\n",
      "When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n).\n",
      "ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst.\n",
      "CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32.\n",
      "We systemically delivered rAAV-PHP.eB packaging a double- floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5).\n",
      "Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke.\n",
      "Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner.\n",
      "Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a).\n",
      "Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33.\n",
      "In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side.\n",
      "We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5).\n",
      "We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11).\n",
      "We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive.\n",
      "Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2).\n",
      "Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1).\n",
      "Behavioral effects were seen at powers as low as 10 mW.\n",
      "To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins.\n",
      "20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R).\n",
      "Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally- invasive optogenetic excitation.\n",
      "Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "\n",
      "\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "\n",
      "## Claire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana Gradinaru1,*, Frances H. Arnold1,2,*\n",
      "\n",
      "\n",
      "\n",
      "1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA 2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "\n",
      "\n",
      "## Abstract\n",
      "\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "\n",
      "## Introduction\n",
      "\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 10<sup>5</sup>–10<sup>6</sup> functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation<sup>3</sup>. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials<sup>4–6</sup>. This confines light-dependent activation to a small volume of brain tissue [~1 mm<sup>3</sup>]<sup>7</sup>. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood<sup>8,9</sup>. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible<sup>10</sup>. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties<sup>6</sup>.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature<sup>11,12</sup>, variants engineered through recombination<sup>9,13</sup> and mutagenesis<sup>14,15</sup>, as well as variants resulting from rational design<sup>16</sup>. Studies of these coupled with structural information<sup>17</sup> and molecular dynamic simulations<sup>18</sup> have established some understanding of the mechanics and sequence features important for specific ChR properties<sup>1,16</sup>. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB<sup>19</sup>. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "\n",
      "## As there is no content provided in either PARSED CONTENT 1 or PARSED CONTENT 2, I cannot merge or create a final markdown string. The result would simply be the section header:\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(final_markdown_self)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "claude_mm = BasicAnthropicLLM(model=\"claude-3-5-sonnet-20240620\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'name': 'img_p0_1.png',\n",
       "  'height': 102,\n",
       "  'width': 102,\n",
       "  'x': 61.5,\n",
       "  'y': 22.5,\n",
       "  'original_width': 152,\n",
       "  'original_height': 152,\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p0_1.png',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 1},\n",
       " {'name': 'page_1.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_1.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 1},\n",
       " {'name': 'page_2.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_2.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 2},\n",
       " {'name': 'page_3.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_3.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 3},\n",
       " {'name': 'page_4.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_4.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 4},\n",
       " {'name': 'page_5.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_5.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 5},\n",
       " {'name': 'page_6.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_6.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 6},\n",
       " {'name': 'page_7.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_7.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 7},\n",
       " {'name': 'page_8.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_8.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 8},\n",
       " {'name': 'page_9.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_9.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 9},\n",
       " {'name': 'page_10.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_10.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 10},\n",
       " {'name': 'page_11.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_11.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 11},\n",
       " {'name': 'page_12.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_12.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 12},\n",
       " {'name': 'page_13.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_13.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 13},\n",
       " {'name': 'page_14.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_14.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 14},\n",
       " {'name': 'page_15.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_15.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 15},\n",
       " {'name': 'page_16.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_16.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 16},\n",
       " {'name': 'page_17.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_17.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 17},\n",
       " {'name': 'page_18.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_18.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 18},\n",
       " {'name': 'page_19.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_19.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 19},\n",
       " {'name': 'img_p19_1.png',\n",
       "  'height': 766,\n",
       "  'width': 873,\n",
       "  'x': 120.13676,\n",
       "  'y': 61.999998000000005,\n",
       "  'original_width': 1800,\n",
       "  'original_height': 1579,\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p19_1.png',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 20},\n",
       " {'name': 'page_20.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_20.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 20},\n",
       " {'name': 'page_21.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_21.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 21},\n",
       " {'name': 'img_p21_1.png',\n",
       "  'height': 859,\n",
       "  'width': 873,\n",
       "  'x': 120,\n",
       "  'y': 62.000001999999995,\n",
       "  'original_width': 1800,\n",
       "  'original_height': 1770,\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p21_1.png',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 22},\n",
       " {'name': 'page_22.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_22.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 22},\n",
       " {'name': 'page_23.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_23.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 23},\n",
       " {'name': 'img_p23_1.png',\n",
       "  'height': 646,\n",
       "  'width': 873,\n",
       "  'x': 120.16205,\n",
       "  'y': 61.99999699999995,\n",
       "  'original_width': 1800,\n",
       "  'original_height': 1333,\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p23_1.png',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 24},\n",
       " {'name': 'page_24.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_24.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 24},\n",
       " {'name': 'page_25.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_25.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 25},\n",
       " {'name': 'img_p25_1.png',\n",
       "  'height': 543,\n",
       "  'width': 873,\n",
       "  'x': 120,\n",
       "  'y': 62.00000300000005,\n",
       "  'original_width': 1800,\n",
       "  'original_height': 1118,\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p25_1.png',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 26},\n",
       " {'name': 'page_26.jpg',\n",
       "  'height': 792,\n",
       "  'width': 612,\n",
       "  'x': 0,\n",
       "  'y': 0,\n",
       "  'type': 'full_page_screenshot',\n",
       "  'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_26.jpg',\n",
       "  'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "  'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "  'page_number': 26}]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "image_dicts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 1\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Abstract', 'Introduction']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\n",
      "\n",
      "[The rest of the introduction is cut off in the image]\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 2\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 3\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Results', 'Functional characterization of ChR variants for machine learning', 'Training Gaussian process (GP) classification and regression models']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "# Results\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 4\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Selection of designer ChRs using trained models']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 5\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Sequence and structural determinants of ChR functional properties']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 6\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Machine-guided search identifies ChRs with a range of useful functional properties']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 7\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Validation of designer ChRs for neuroscience applications', 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 8\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 9\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Discussion']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "## Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 10\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Online methods', 'Construct design and cloning', 'HEK293T cell and primary neuronal cultures']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "## Online methods\n",
      "\n",
      "### Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "### HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 11\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Patch-clamp electrophysiology']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "NO_CONTENT_HERE\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 12\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Light delivery and imaging', 'Electrophysiology data analysis', 'AAV production and purification']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 13\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['Animals', 'Intravenous injections, stereotactic injections, and cannula implantation', 'Gaussian process modeling']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "## Animals\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 14\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['GP regression']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "## GP regression\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 15\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['GP classification', 'GP kernels for modeling proteins']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "## GP classification\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "## GP kernels for modeling proteins\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 16\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['L1 regression feature identification and weighting', 'Statistical analysis', 'Data availability', 'Code availability', 'Supplementary Material', 'Acknowledgements']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "Where l is once again a hyperparameter.\n",
      "\n",
      "## L1 regression feature identification and weighting\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "## Statistical analysis\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "## Data availability\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "## Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "## Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "## Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 17\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "['References']\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "## References\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 18\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\n",
      "\n",
      "30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\n",
      "\n",
      "31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\n",
      "\n",
      "32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\n",
      "\n",
      "33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\n",
      "\n",
      "34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\n",
      "\n",
      "35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\n",
      "\n",
      "36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\n",
      "\n",
      "37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\n",
      "\n",
      "38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 19\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\n",
      "\n",
      "40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\n",
      "\n",
      "41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\n",
      "\n",
      "42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\n",
      "\n",
      "43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\n",
      "\n",
      "44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\n",
      "\n",
      "45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\n",
      "\n",
      "46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 20\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "Figure 1\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\n",
      "\n",
      "The figure contains several parts:\n",
      "\n",
      "a. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\n",
      "\n",
      "b. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\n",
      "\n",
      "c. Train regression models graphs for current strength, off-kinetics, and spectral properties.\n",
      "\n",
      "d. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\n",
      "\n",
      "e. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\n",
      "\n",
      "f. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\n",
      "\n",
      "The image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 21\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 22\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "Figure 2\n",
      "\n",
      "## Figure 2.\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strength\n",
      "\n",
      "The figure contains several panels:\n",
      "\n",
      "a) A series of current traces and corresponding cell images for different ChR variants.\n",
      "\n",
      "b) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\n",
      "\n",
      "c) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\n",
      "\n",
      "d) Line graphs showing normalized photocurrent vs wavelength.\n",
      "\n",
      "e) Line graphs showing peak and steady-state photocurrent vs light intensity.\n",
      "\n",
      "Each panel contains detailed data comparing various ChR variants across different properties and conditions.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 23\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "# Figure 2\n",
      "# Figure 2\n",
      "## Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 24\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "# Figure 2\n",
      "# Figure 2\n",
      "## Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "The figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "False\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 25\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "# Figure 2\n",
      "# Figure 2\n",
      "## Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "# Figure 3\n",
      "# Figure 3\n",
      "## Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n",
      "Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
      "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
      "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
      "I've also included the previous page's headers to inform the levels of the target page.\n",
      "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
      "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
      "\n",
      "Page number: 26\n",
      "\n",
      "EXPECTED SECTION HEADERS:\n",
      "[]\n",
      "\n",
      "PREVIOUS PAGE HEADERS:\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Abstract\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "## Introduction\n",
      "# Results\n",
      "# Results\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "# Results\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "## Selection of designer ChRs using trained models\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "# Results\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "# Results\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "# Discussion\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Construct design and cloning\n",
      "# Online methods\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "## Patch-clamp electrophysiology\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "# Online methods\n",
      "## Electrophysiology data analysis\n",
      "# Online methods\n",
      "## AAV production and purification\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Animals\n",
      "# Online methods\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "### GP regression\n",
      "### GP regression\n",
      "### GP classification\n",
      "### GP kernels for modeling proteins\n",
      "# L1 regression feature identification and weighting\n",
      "# Statistical analysis\n",
      "# Data availability\n",
      "# Code availability\n",
      "# Supplementary Material\n",
      "# Acknowledgements\n",
      "# References\n",
      "# References\n",
      "# Figure 1\n",
      "# Figure 1\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "# Figure 2\n",
      "# Figure 2\n",
      "## Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "# Figure 3\n",
      "# Figure 3\n",
      "## Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "### Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "PARSED MARKDOWN:\n",
      "## Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n",
      "\n",
      "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
      "True\n",
      "\n",
      "\n",
      "Return only the final markdown string without any other text.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from PIL import Image\n",
    "from pydantic import BaseModel, Field\n",
    "from typing import Optional\n",
    "from llama_index.core.prompts import PromptTemplate\n",
    "from llama_index.core.output_parsers.pydantic import PydanticOutputParser\n",
    "from llama_index.core.schema import TextNode\n",
    "from llama_index.core.node_parser import MarkdownNodeParser\n",
    "\n",
    "import plotreader\n",
    "\n",
    "class Section(BaseModel):\n",
    "    header: str\n",
    "    level: int\n",
    "\n",
    "class PageMetadata(BaseModel):\n",
    "    page_number: int\n",
    "    figure_count: int\n",
    "    figure_names: List[str]\n",
    "    section_headers: List[str]\n",
    "    contains_fig_caption: bool = Field(..., description=\"Does any part of a figure caption exist on this page?\")\n",
    "    # contains_partial_fig_caption: bool = Field(..., description=\"Is there a figure caption that is partially contained in this page?\")\n",
    "\n",
    "output_parser = PydanticOutputParser(output_cls=PageMetadata)\n",
    "prompt_page_metadata = PromptTemplate(\n",
    "    \"The image provided is a page from a scientific paper. Is there a figure in this image? If so, what is the figure's name?\",\n",
    "    output_parser=output_parser\n",
    ").format(llm=plotreader._CLAUDE_SONNET35_MULTIMODAL)\n",
    "\n",
    "# You've also been provided with an image of the page and the page before and after (if they exist).\n",
    "revision_prompt = \"\"\"Below is a markdown string parsed from a signle page of a scientific paper PDF. \n",
    "Your job is to revise the levels of each part of the extracted markdown string so it matches the PDF.\n",
    "To aid you, I've provided the list of expected section headers for the page and information about whether or not to expecdt figure captions.\n",
    "I've also included the previous page's headers to inform the levels of the target page.\n",
    "Plese make each Figure caption its own section. Note that some figure captions may be split across two pages.\n",
    "Also, do not change any text. Only change the levels and organization of the sections to better match the PDF image.\n",
    "\n",
    "Page number: {page_number}\n",
    "\n",
    "EXPECTED SECTION HEADERS:\n",
    "{sections}\n",
    "\n",
    "PREVIOUS PAGE HEADERS:\n",
    "{previous_page_headers}\n",
    "\n",
    "PARSED MARKDOWN:\n",
    "{content1}\n",
    "\n",
    "CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
    "{contains_fig_caption}\n",
    "\n",
    "PREVIOUS PAGE CONTAINS WHOLE OR PARTIAL FIGURE CAPTION:\n",
    "{previous_page_contains_fig_caption}\n",
    "\n",
    "\n",
    "Return only the final markdown string without any other text.\n",
    "\"\"\"\n",
    "\n",
    "page_metadata_responses = []\n",
    "prev_page_contains_fig_caption = False\n",
    "prev_page_headers = []\n",
    "for image_dict in image_dicts:\n",
    "    if image_dict.get(\"type\") == \"full_page_screenshot\":\n",
    "\n",
    "        # get page metadata\n",
    "        page_num = image_dict.get(\"page_number\")\n",
    "        page_img = Image.open(image_dict.get(\"path\"))\n",
    "        response = claude_mm.message(prompt_page_metadata, images=[page_img])\n",
    "        page_metadata_responses.append(output_parser.parse(response.content[0].text))\n",
    "\n",
    "        # revise page markdown\n",
    "        try:\n",
    "            page_md = json_objs[0][\"pages\"][page_num-1][\"md\"]\n",
    "            revision_prompt_with_md = revision_prompt.format(\n",
    "                page_number=page_num, \n",
    "                sections=page_metadata_responses[page_num-1].section_headers, \n",
    "                content1=page_md,\n",
    "                previous_page_contains_fig_caption=prev_page_contains_fig_caption,\n",
    "                contains_fig_caption=page_metadata_responses[page_num-1].contains_fig_caption,\n",
    "                previous_page_headers=f\"\\n\".join(prev_page_headers),\n",
    "            )\n",
    "            print(revision_prompt_with_md)\n",
    "            response = claude_mm.message(revision_prompt_with_md, images=[page_img])\n",
    "            revised_md = response.content[0].text\n",
    "            json_objs[0][\"pages\"][page_num-1][\"md_revised\"] = revised_md\n",
    "\n",
    "            prev_page_contains_fig_caption = page_metadata_responses[page_num-1].contains_fig_caption\n",
    "            node = TextNode(text=revised_md)\n",
    "            nodes = MarkdownNodeParser().get_nodes_from_node(node)\n",
    "            for node in nodes:\n",
    "                for key in node.metadata.keys():\n",
    "                    if key.startswith(\"Header_\"):\n",
    "                        level = int(key.split(\"_\")[1])\n",
    "                        prev_page_headers.append(\"\".join([\"#\"]*level) + \" \" + node.metadata.get(key))\n",
    "        except Exception as e:\n",
    "            print(e)    \n",
    "            json_objs[0][\"pages\"][page_num-1][\"md_revised\"] = page_md\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functionalLimitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "# Results\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "## Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position\n",
      "\n",
      "## Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\n",
      "\n",
      "# Results\n",
      "## Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "## Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "# Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other\n",
      "\n",
      "# Online methods\n",
      "\n",
      "## Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\n",
      "\n",
      "## Patch-clamp electrophysiology\n",
      "\n",
      "# Online methods\n",
      "## Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "## Electrophysiology data analysis\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "## AAV production and purification\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in\n",
      "\n",
      "# Online methods\n",
      "\n",
      "phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "## Animals\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "## Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "## Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use\n",
      "\n",
      "# Online methods\n",
      "## Gaussian process modeling\n",
      "\n",
      "the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "### GP regression\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\n",
      "\n",
      "### GP regression\n",
      "\n",
      "$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "### GP classification\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "### GP kernels for modeling proteins\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "# L1 regression feature identification and weighting\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "# Statistical analysis\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "# Data availability\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "# Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "# Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "# Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hubgrant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "# References\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\n",
      "\n",
      "30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\n",
      "\n",
      "31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\n",
      "\n",
      "32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\n",
      "\n",
      "33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\n",
      "\n",
      "34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\n",
      "\n",
      "35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\n",
      "\n",
      "36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\n",
      "\n",
      "37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\n",
      "\n",
      "38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]\n",
      "\n",
      "# References\n",
      "\n",
      "39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\n",
      "\n",
      "40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\n",
      "\n",
      "41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\n",
      "\n",
      "42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\n",
      "\n",
      "43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\n",
      "\n",
      "44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\n",
      "\n",
      "45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\n",
      "\n",
      "46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "# Figure 1\n",
      "\n",
      "## Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.\n",
      "\n",
      "# Figure 2\n",
      "\n",
      "## Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strengthwith varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.\n",
      "\n",
      "# Figure 3\n",
      "## Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "### Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.\n",
      "\n",
      "# Figure 4\n",
      "\n",
      "## Figure 4. Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n"
     ]
    }
   ],
   "source": [
    "full_text = \"\"\n",
    "for page in json_objs[0][\"pages\"]:\n",
    "    if page[\"md_revised\"].startswith(\"#\"):\n",
    "        prefix = \"\\n\\n\"\n",
    "    else:\n",
    "        prefix = \"\"\n",
    "    full_text += prefix + page[\"md_revised\"]\n",
    "\n",
    "print(full_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "## Abstract\n",
      "\n",
      "We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\n",
      "\n",
      "## Introduction\n",
      "\n",
      "Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functionalLimitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\n",
      "\n",
      "Engineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\n",
      "\n",
      "Diverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\n",
      "\n",
      "Our approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\n",
      "\n",
      "We trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\n",
      "\n",
      "# Results\n",
      "\n",
      "## Functional characterization of ChR variants for machine learning\n",
      "\n",
      "In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\n",
      "\n",
      "Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\n",
      "\n",
      "## Training Gaussian process (GP) classification and regression models\n",
      "\n",
      "Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\n",
      "\n",
      "We trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\n",
      "\n",
      "Next, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\n",
      "\n",
      "# Selection of designer ChRs using trained models\n",
      "\n",
      "To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\n",
      "\n",
      "The training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\n",
      "\n",
      "From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\n",
      "\n",
      "Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\n",
      "\n",
      "# Sequence and structural determinants of ChR functional properties\n",
      "\n",
      "We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position\n",
      "\n",
      "# Machine-guided search identifies ChRs with a range of useful functional properties\n",
      "\n",
      "195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\n",
      "\n",
      "We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\n",
      "\n",
      "Designer ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\n",
      "\n",
      "Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\n",
      "\n",
      "We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs areexpressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\n",
      "\n",
      "We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\n",
      "\n",
      "# Validation of designer ChRs for neuroscience applications\n",
      "\n",
      "For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\n",
      "\n",
      "Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\n",
      "\n",
      "# Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\n",
      "\n",
      "We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\n",
      "\n",
      "We hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\n",
      "\n",
      "We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\n",
      "\n",
      "We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\n",
      "\n",
      "In order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\n",
      "\n",
      "# Discussion\n",
      "\n",
      "We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\n",
      "\n",
      "Machine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize otherneuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\n",
      "\n",
      "We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\n",
      "\n",
      "# Online methods\n",
      "\n",
      "## Construct design and cloning\n",
      "\n",
      "The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\n",
      "\n",
      "Selected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\n",
      "\n",
      "For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\n",
      "\n",
      "## HEK293T cell and primary neuronal cultures\n",
      "\n",
      "The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRsNO_CONTENT_HERE\n",
      "\n",
      "# Light delivery and imaging\n",
      "\n",
      "Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\n",
      "\n",
      "Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\n",
      "\n",
      "# Electrophysiology data analysis\n",
      "\n",
      "Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\n",
      "\n",
      "# AAV production and purification\n",
      "\n",
      "Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated inphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\n",
      "\n",
      "# Animals\n",
      "\n",
      "All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\n",
      "\n",
      "# Intravenous injections, stereotactic injections, and cannula implantation\n",
      "\n",
      "Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\n",
      "\n",
      "Analysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\n",
      "\n",
      "# Gaussian process modeling\n",
      "\n",
      "Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) usethe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\n",
      "\n",
      "For a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\n",
      "\n",
      "For classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\n",
      "\n",
      "# GP regression\n",
      "\n",
      "In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\n",
      "\n",
      "$$f* = k*^T(K + σn^2I)^{-1}y$$\n",
      "\n",
      "and variance\n",
      "\n",
      "$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\n",
      "\n",
      "Where K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:$$\\log p(y|X) = -\\frac{1}{2}y^T(K + \\sigma_n^2I)^{-1}y - \\frac{1}{2}\\log |K + \\sigma_n^2I| - \\frac{n}{2}\\log 2\\pi \\tag{3}$$\n",
      "\n",
      "where n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\n",
      "\n",
      "# GP classification\n",
      "\n",
      "In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\in \\{+1, -1\\}$, and the goal is to use the training data to make probabilistic predictions $\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\n",
      "\n",
      "# GP kernels for modeling proteins\n",
      "\n",
      "Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\n",
      "\n",
      "Three types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\n",
      "\n",
      "$$k(s, s') = (\\sigma_0^2 + \\sigma_p^2x^Tx')^d \\tag{4}$$\n",
      "\n",
      "where $\\sigma_0$ and $\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\n",
      "\n",
      "$$k(s, s') = \\sigma_p^2\\exp\\left(-\\frac{\\|x - x'\\|_2^2}{l}\\right) \\tag{5}$$\n",
      "\n",
      "where l and $\\sigma_p$ are also hyperparameters and $|\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\frac{5}{2}$ is defined as:\n",
      "\n",
      "$$k(s, s') = \\left(1 + \\frac{\\sqrt{5}\\|x - x'\\|_2}{l} + \\frac{5\\|x - x'\\|_2^2}{3l^2}\\right)\\exp\\left(-\\frac{\\sqrt{5}\\|x - x'\\|_2}{l}\\right) \\tag{6}$$Where l is once again a hyperparameter.\n",
      "\n",
      "# L1 regression feature identification and weighting\n",
      "\n",
      "L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\n",
      "\n",
      "# Statistical analysis\n",
      "\n",
      "Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\n",
      "\n",
      "# Data availability\n",
      "\n",
      "The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\n",
      "\n",
      "# Code availability\n",
      "\n",
      "Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.\n",
      "\n",
      "# Supplementary Material\n",
      "\n",
      "Refer to Web version on PubMed Central for supplementary material.\n",
      "\n",
      "# Acknowledgements\n",
      "\n",
      "We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hubgrant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\n",
      "\n",
      "# References\n",
      "\n",
      "1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\n",
      "\n",
      "2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\n",
      "\n",
      "3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\n",
      "\n",
      "4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\n",
      "\n",
      "5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\n",
      "\n",
      "6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\n",
      "\n",
      "7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\n",
      "\n",
      "8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\n",
      "\n",
      "9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\n",
      "\n",
      "10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\n",
      "\n",
      "11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\n",
      "\n",
      "12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\n",
      "\n",
      "13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\n",
      "\n",
      "14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\n",
      "\n",
      "15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\n",
      "\n",
      "16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\n",
      "\n",
      "17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\n",
      "\n",
      "18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\n",
      "\n",
      "20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\n",
      "\n",
      "21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\n",
      "\n",
      "22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\n",
      "\n",
      "23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\n",
      "\n",
      "24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\n",
      "\n",
      "25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\n",
      "\n",
      "26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\n",
      "\n",
      "27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\n",
      "\n",
      "28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\n",
      "\n",
      "29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\n",
      "\n",
      "30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\n",
      "\n",
      "31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\n",
      "\n",
      "32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\n",
      "\n",
      "33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\n",
      "\n",
      "34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\n",
      "\n",
      "35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\n",
      "\n",
      "36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\n",
      "\n",
      "37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\n",
      "\n",
      "38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\n",
      "\n",
      "40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\n",
      "\n",
      "41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\n",
      "\n",
      "42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\n",
      "\n",
      "43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\n",
      "\n",
      "44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\n",
      "\n",
      "45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\n",
      "\n",
      "46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\n",
      "\n",
      "47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).\n",
      "\n",
      "# Figure 1. Machine learning-guided optimization of ChRs.\n",
      "\n",
      "(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\n",
      "\n",
      "(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\n",
      "\n",
      "(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\n",
      "\n",
      "(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.\n",
      "\n",
      "# Figure 2\n",
      "\n",
      "The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\n",
      "\n",
      "(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\n",
      "\n",
      "(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\n",
      "\n",
      "(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\n",
      "\n",
      "(d) Peak and steady-state photocurrent strengthwith varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.\n",
      "\n",
      "# Figure 3.\n",
      "ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\n",
      "\n",
      "(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\n",
      "\n",
      "(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\n",
      "\n",
      "(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\n",
      "\n",
      "(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\n",
      "\n",
      "(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\n",
      "\n",
      "(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\n",
      "\n",
      "(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\n",
      "\n",
      "(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\n",
      "\n",
      "(i) Brain slice images showing systemic delivery (5x10^11 VG).\n",
      "\n",
      "(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\n",
      "\n",
      "(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\n",
      "\n",
      "(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\n",
      "\n",
      "### Figure Caption\n",
      "varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.\n",
      "\n",
      "# Figure 4.\n",
      "\n",
      "Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\n",
      "\n",
      "(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\n",
      "\n",
      "(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\n",
      "\n",
      "The figure contains several panels illustrating the experimental setup and results:\n",
      "\n",
      "a. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\n",
      "b. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\n",
      "c. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\n",
      "d. Systemic delivery and fiber on skull surface diagrams for WT mice.\n",
      "e. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\n",
      "f. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\n",
      "g. Time-lapse images showing mouse movement in response to optical stimulation.\n",
      "\n",
      "The figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.\n"
     ]
    }
   ],
   "source": [
    "print(f\"{full_text.strip()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "nodes = MarkdownNodeParser().get_nodes_from_node(TextNode(text=f\"{full_text.strip()}\"))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[TextNode(id_='32f72ba0-6df2-481c-9f97-dc57f3424659', embedding=None, metadata={'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\\n\\nClaire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='69c01057-03f1-4caf-9720-6bd4ed010010', embedding=None, metadata={'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics', 'Header_2': 'Abstract'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Abstract\\n\\nWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='23f6472e-6d23-49bc-a1b1-8a5cc4200972', embedding=None, metadata={'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics', 'Header_2': 'Introduction'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Introduction\\n\\nChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functionalLimitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='3e503aa7-b9b0-4287-90fc-86dc3bc7826d', embedding=None, metadata={'Header_1': 'Results'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Results', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='45417e0b-9028-45a0-b1ce-8c7c3f99e821', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Functional characterization of ChR variants for machine learning'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Functional characterization of ChR variants for machine learning\\n\\nIn previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='ee8f41b5-97a0-4889-a939-c850f504d781', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Training Gaussian process (GP) classification and regression models'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Training Gaussian process (GP) classification and regression models\\n\\nUsing the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structuralsimilarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='b4dced10-c5fb-4491-b5a5-bd02aaf18455', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Selection of designer ChRs using trained models'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Selection of designer ChRs using trained models\\n\\nTo select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant fromthose previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='7688a596-3cca-4a0f-80e5-2ecf4cac9453', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Sequence and structural determinants of ChR functional properties'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Sequence and structural determinants of ChR functional properties\\n\\nWe used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='cbb63d46-7532-419c-a7ca-650a9802be3e', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Sequence and structural determinants of ChR functional properties'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Sequence and structural determinants of ChR functional properties\\n\\n195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='f193bbd1-e31d-4ad2-897a-fc13837f437d', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Machine-guided search identifies ChRs with a range of useful functional properties'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Machine-guided search identifies ChRs with a range of useful functional properties\\n\\nWe assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='d618d791-b851-42c4-84c4-cae409c9c4e7', embedding=None, metadata={'Header_1': 'Results'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Results', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='5df9c8ba-d7db-467c-a530-3ce65a3e414a', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Machine-guided search identifies ChRs with a range of useful functional properties'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Machine-guided search identifies ChRs with a range of useful functional properties\\n\\nexpressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='6afb7638-f6f0-4a01-b84c-9b5190460733', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Validation of designer ChRs for neuroscience applications'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Validation of designer ChRs for neuroscience applications\\n\\nFor further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='4c96896b-2a64-432a-815c-6b908150a978', embedding=None, metadata={'Header_1': 'Results', 'Header_2': 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n\\nWe investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results inmany copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results inacute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='90664bca-7a99-46f4-8ec7-bd399dda51e4', embedding=None, metadata={'Header_1': 'Discussion'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Discussion\\n\\nWe demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='c6b82f67-358f-4048-a809-5f8ce4892132', embedding=None, metadata={'Header_1': 'Online methods'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Online methods', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='93c3df5b-c792-4d92-a945-b657c5ec5edc', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Construct design and cloning'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Construct design and cloning\\n\\nThe design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='9d89f35f-38ea-4427-91cd-c6ec7d799f03', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'HEK293T cell and primary neuronal cultures'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"HEK293T cell and primary neuronal cultures\\n\\nThe culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='b509b541-314a-45de-8aeb-e27bac4ec8af', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Patch-clamp electrophysiology'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Patch-clamp electrophysiology', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='df20f2ce-7e37-46a6-9693-4077f473a217', embedding=None, metadata={'Header_1': 'Online methods'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Online methods', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='4bcf85f6-09ad-46ac-8746-8299ddf475f5', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Light delivery and imaging'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Light delivery and imaging\\n\\nPatch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='a143b0b9-471f-4efc-93b8-b7127b2ed8a5', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Electrophysiology data analysis'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Electrophysiology data analysis\\n\\nElectrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='096730fb-5823-424e-9318-49a7373f238a', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'AAV production and purification'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='AAV production and purification\\n\\nProduction of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='9fb460cc-3750-403d-ba6b-fcb2254b5dbb', embedding=None, metadata={'Header_1': 'Online methods'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Online methods\\n\\nphosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='453fe94d-0f9d-4913-b543-5f7f3483d918', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Animals'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Animals\\n\\nAll procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='56b28b74-3bb8-4a7f-a721-4077e206fa04', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Intravenous injections, stereotactic injections, and cannula implantation'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Intravenous injections, stereotactic injections, and cannula implantation\\n\\nIntravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='77f8d6fc-b3d2-48d0-ba41-1ca2aaffcdb5', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Gaussian process modeling\\n\\nBoth the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='404de229-baa4-4ae9-86bd-119a7de55fb6', embedding=None, metadata={'Header_1': 'Online methods'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Online methods', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='44ec95f5-2f3b-43cd-991f-f35d5a09e787', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Gaussian process modeling\\n\\nthe C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='11be305b-c238-495d-b4b6-6800ac27da8a', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling', 'Header_3': 'GP regression'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='GP regression\\n\\nIn regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='89a874ae-8a10-4a3a-a74f-8981f2a2ff32', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling', 'Header_3': 'GP regression'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='GP regression\\n\\n$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='fee4101a-a232-4e14-acc5-1bed3f6f40c1', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling', 'Header_3': 'GP classification'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"GP classification\\n\\nIn binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='2876c1b8-a6c1-43fb-b0c4-0d23c9cfb3b8', embedding=None, metadata={'Header_1': 'Online methods', 'Header_2': 'Gaussian process modeling', 'Header_3': 'GP kernels for modeling proteins'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"GP kernels for modeling proteins\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$Where l is once again a hyperparameter.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='f151d5d6-d8fe-46ac-b134-868f9d0cc58a', embedding=None, metadata={'Header_1': 'L1 regression feature identification and weighting'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='L1 regression feature identification and weighting\\n\\nL1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='cfb40421-ba51-44d7-b649-2dea2f40a63c', embedding=None, metadata={'Header_1': 'Statistical analysis'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Statistical analysis\\n\\nPlotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='28821162-0c37-4a6c-9ead-bd2ad8f94ef2', embedding=None, metadata={'Header_1': 'Data availability'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Data availability\\n\\nThe authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='248ad4e7-779c-4321-8f84-5310c058f61a', embedding=None, metadata={'Header_1': 'Code availability'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Code availability\\n\\nCode used to train classification and regression models can be found at: https://github.com/fhalab/channels.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='799c4c38-6b49-4b17-a2da-50569db8b52d', embedding=None, metadata={'Header_1': 'Supplementary Material'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Supplementary Material\\n\\nRefer to Web version on PubMed Central for supplementary material.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='c8a1c5ef-a45c-463c-84eb-5d61439b0d9c', embedding=None, metadata={'Header_1': 'Acknowledgements'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"Acknowledgements\\n\\nWe thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hubgrant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='e57549bc-1ab9-4ba6-a88d-6379ddd8915e', embedding=None, metadata={'Header_1': 'References'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text=\"References\\n\\n1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]\", mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='44d306da-46ac-4f19-9382-a542ccc3bc61', embedding=None, metadata={'Header_1': 'References'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='References\\n\\n39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='7cf6cc03-ab73-4ea8-8315-b9225852f2b7', embedding=None, metadata={'Header_1': 'Figure 1'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 1', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='5ddc83c3-ad31-403c-8db3-a40cc432ea24', embedding=None, metadata={'Header_1': 'Figure 1', 'Header_2': 'Figure 1. Machine learning-guided optimization of ChRs.'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 1. Machine learning-guided optimization of ChRs.\\n\\n(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='8d64f145-6a62-40d4-9720-c4d9d20e208f', embedding=None, metadata={'Header_1': 'Figure 2'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 2', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='4e49b2e9-acc1-4016-a26f-abdef1450ebd', embedding=None, metadata={'Header_1': 'Figure 2', 'Header_2': 'Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 2. The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strengthwith varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='5ada6739-81a0-4771-a525-99a9f67a6545', embedding=None, metadata={'Header_1': 'Figure 3'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 3', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='3f2f178c-0e18-4e35-87c8-201fc35d88c9', embedding=None, metadata={'Header_1': 'Figure 3', 'Header_2': 'Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='dfc6118f-0062-4b9b-b7f5-9da853b694ba', embedding=None, metadata={'Header_1': 'Figure 3', 'Header_2': 'Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.', 'Header_3': 'Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 3. ChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\nvarying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='5a5d814e-feee-4a51-aa7d-c42411ffb45e', embedding=None, metadata={'Header_1': 'Figure 4'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 4', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n'),\n",
       " TextNode(id_='d234d909-a4eb-45d5-b166-4527d5cb355a', embedding=None, metadata={'Header_1': 'Figure 4', 'Header_2': 'Figure 4. Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='78f1d459-a8ce-48ab-ac36-c2a5fa1635f2', node_type=<ObjectType.TEXT: '1'>, metadata={}, hash='1898b569aeb72ef4bc06bb408514a299b6dd6dea9402713bcc50a90d16942034')}, text='Figure 4. Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.', mimetype='text/plain', start_char_idx=None, end_char_idx=None, text_template='{metadata_str}\\n\\n{content}', metadata_template='{key}: {value}', metadata_seperator='\\n')]"
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics'}\n",
      "{'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics', 'Header_2': 'Abstract'}\n",
      "{'Header_1': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics', 'Header_2': 'Introduction'}\n",
      "{'Header_1': 'Results'}\n",
      "{'Header_1': 'Results', 'Header_2': 'Functional characterization of ChR variants for machine learning'}\n",
      "{'Header_1': 'Results', 'Header_2': 'Training Gaussian process (GP) classification and regression models'}\n",
      "{'Header_1': 'Selection of designer ChRs using trained models'}\n",
      "{'Header_1': 'Sequence and structural determinants of ChR functional properties'}\n",
      "{'Header_1': 'Machine-guided search identifies ChRs with a range of useful functional properties'}\n",
      "{'Header_1': 'Validation of designer ChRs for neuroscience applications'}\n",
      "{'Header_1': 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation'}\n",
      "{'Header_1': 'Discussion'}\n",
      "{'Header_1': 'Online methods'}\n",
      "{'Header_1': 'Online methods', 'Header_2': 'Construct design and cloning'}\n",
      "{'Header_1': 'Online methods', 'Header_2': 'HEK293T cell and primary neuronal cultures'}\n",
      "{'Header_1': 'Light delivery and imaging'}\n",
      "{'Header_1': 'Electrophysiology data analysis'}\n",
      "{'Header_1': 'AAV production and purification'}\n",
      "{'Header_1': 'Animals'}\n",
      "{'Header_1': 'Intravenous injections, stereotactic injections, and cannula implantation'}\n",
      "{'Header_1': 'Gaussian process modeling'}\n",
      "{'Header_1': 'GP regression'}\n",
      "{'Header_1': 'GP classification'}\n",
      "{'Header_1': 'GP kernels for modeling proteins'}\n",
      "{'Header_1': 'L1 regression feature identification and weighting'}\n",
      "{'Header_1': 'Statistical analysis'}\n",
      "{'Header_1': 'Data availability'}\n",
      "{'Header_1': 'Code availability'}\n",
      "{'Header_1': 'Supplementary Material'}\n",
      "{'Header_1': 'Acknowledgements'}\n",
      "{'Header_1': 'References'}\n",
      "{'Header_1': 'Figure 1. Machine learning-guided optimization of ChRs.'}\n",
      "{'Header_1': 'Figure 2'}\n",
      "{'Header_1': 'Figure 3.'}\n",
      "{'Header_1': 'Figure 3.', 'Header_3': 'Figure Caption'}\n",
      "{'Header_1': 'Figure 4.'}\n"
     ]
    }
   ],
   "source": [
    "for node in nodes:\n",
    "    print(node.metadata)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [],
   "source": [
    "lines = full_text.split(\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "367"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(lines)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\n",
      "\n",
      "Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\n",
      "\n",
      "<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n",
      "<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(10):\n",
    "    print(lines[i])\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "page_dicts = json_objs[0][\"pages\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'page': 1,\n",
       "  'text': '                                     HHS Public Access\\n                                     Author manuscript\\nAuthor Manuscript                    Nat Methods. Author manuscript; available in PMC 2020 April 14.\\n                       Published in final edited form as:\\n                         Nat Methods. 2019 November ; 16(11): 1176–1184. doi:10.1038/s41592-019-0583-8.\\n\\n\\n\\n                       Machine learning-guided channelrhodopsin engineering enables\\n                       minimally-invasive optogenetics\\n                       Claire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana\\n                       Gradinaru1,*, Frances H. Arnold1,2,*\\n                       1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena;\\nAuthor ManuscriptCalifornia; USA\\n                       2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena;\\n                       California; USA\\n\\n\\n\\n                       Abstract\\n                             We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity\\n                             enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the\\n                             mammalian brain require intracranial surgery for transgene delivery and implantation of invasive\\n                             fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate\\n                             expansive optogenetics without the need for invasive implants, our engineering approach leverages\\n                             the significant literature of ChR variants to train statistical models for the design of new, high-\\nAuthor Manuscriptperformance ChRs. With Gaussian Process models trained on a limited experimental set of 102\\n                             functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light\\n                             sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via\\n                             minimally-invasive systemic transgene delivery, not possible previously due to low per-cell\\n                             transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation\\n                             without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables\\n                             minimally-invasive optogenetics.\\n\\n\\n\\n                       Introduction\\n                                           Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae.\\n                                           Transgenic expression of ChRs in the brain enables light-dependent neuronal activation1.\\nAuthor Manuscript                          These channels are widely applied as tools in neuroscience research2; however, functional\\n\\n\\n\\n                       Users may view, print, copy, and download text and data-mine the content in such documents, for the purposes of academic research,\\n                       *subject always to the full Conditions of use:http://www.nature.com/authors/editorial_policies/license.html#terms\\n                       †K.K.Y. & J.E.R. contributed equally to this work.\\n                                                                         Corresponding: viviana@caltech.edu (V.G.) & fha@cheme.caltech.edu (F.H.A).\\n                       Author Contributions\\n                       C.N.B., K.K.Y., V.G., and F.H.A. conceptualized the project. C.N.B. coordinated all experiments and data analysis. C.N.B. and K.K.Y.\\n                       built machine-learning models. C.N.B. performed construct design and cloning. C.N.B. and E.D.M. performed AAV production.\\n                       E.D.M. prepared cultured neurons. C.N.B and J.E.R. conducted electrophysiology. C.N.B. and J.E.R. performed injections. J.E.R.\\n                       performed fiber cannula implants and behavioral experiments. C.N.B. performed all data analysis. C.N.B. wrote the manuscript with\\n                       input and editing from all authors. V.G. supervised optogenetics/electrophysiology, and F.H.A. supervised the protein engineering.\\n                       Competing interests\\n                       A provisional patent application (CIT File No.: CIT-8092-P) has been filed by Caltech based on these results. C.N.B., K.K.Y., V.G.,\\n                       and F.H.A. are inventors on this provisional patent.',\n",
       "  'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\\n\\nClaire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n## Abstract\\n\\nWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\\n\\n## Introduction\\n\\nChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "  'images': [{'name': 'img_p0_1.png',\n",
       "    'height': 102,\n",
       "    'width': 102,\n",
       "    'x': 61.5,\n",
       "    'y': 22.5,\n",
       "    'original_width': 152,\n",
       "    'original_height': 152,\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p0_1.png',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 1},\n",
       "   {'name': 'page_1.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_1.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 1}],\n",
       "  'items': [{'type': 'heading',\n",
       "    'lvl': 1,\n",
       "    'value': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "    'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "    'bBox': {'x': 90, 'y': 142, 'w': 429.0159999999997, 'h': 14}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "    'md': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "    'bBox': {'x': 90, 'y': 233, 'w': 68.30000000000004, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Abstract',\n",
       "    'md': '## Abstract',\n",
       "    'bBox': {'x': 89, 'y': 295, 'w': 48.672, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "    'md': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "    'bBox': {'x': 90, 'y': 162, 'w': 394.5899999999998, 'h': 14}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Introduction',\n",
       "    'md': '## Introduction',\n",
       "    'bBox': {'x': 90, 'y': 520, 'w': 69.756, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "    'md': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "    'bBox': {'x': 90, 'y': 520, 'w': 345.41999999999985, 'h': 12}}],\n",
       "  'md_revised': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\\n\\nClaire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n## Abstract\\n\\nWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\\n\\n## Introduction\\n\\nChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional'},\n",
       " {'page': 2,\n",
       "  'text': '                   Bedbrook et al.                                                                                                   Page 2\\n\\n\\n\\nAuthor Manuscript                    limitations of available ChRs prohibit a number of optogenetic applications. These channels\\n                                    have broad activation spectra in the visible range and require high-intensity light for\\n                                    activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring\\n                                     approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to\\n                                     produce sufficient light-dependent depolarization to induce neuronal activation3. When\\n                                     applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target\\n                                     cell population to reliably activate action potentials4–6. This confines light-dependent\\n                                     activation to a small volume of brain tissue [~1 mm37. Enabling optogenetics for large brain]\\n                                     volumes without the need to implant invasive optical fibers for light delivery would be\\n                                     highly desirable for neuroscience applications.\\n\\n\\n\\n                                     Engineering ChRs to overcome limits in conductance and light sensitivity and extend the\\n                                     reach of optogenetic experiments requires overcoming three major challenges. First,\\nAuthor Manuscript                    rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the\\n                                     sequence and structural determinants of membrane protein expression and plasma-\\n                                     membrane localization are highly constrained and poorly understood8,9. Second, because\\n                                     properties of interest for neuroscience applications are assayed using low-throughput\\n                                     techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not\\n                                     feasible10. And third,in vivo applications require retention and optimization of multiple\\n                                     properties, for example, localization in mammalian cells while simultaneously tuning\\n                                     kinetics, photocurrents, and spectral properties6.\\n                                     Diverse ChRs have been published, including variants discovered from nature11,12, variants\\n                                     engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting\\n                                     from rational design16. Studies of these coupled with structural information17 and molecular\\nAuthor Manuscript                    dynamic simulations18 have established some understanding of the mechanics and sequence\\n                                     features important for specific ChR properties1,16. Despite this, it is still not possible to\\n                                     predict functional properties of new ChR sequences.\\n\\n\\n\\n                                     Our approach has been to leverage the significant literature of ChRs to train statistical\\n                                     models that enable design of highly-functional ChRs. These models take as their input the\\n                                     sequence and structural information for a given ChR variant and then predict its functional\\n                                     properties. The models use training data to learn how sequence and structural elements map\\n                                     to ChR function. Once known, that mapping can be used to predict the behavior of untested\\n                                     ChR variants.\\n\\n\\n\\n                                     We trained models in this manner and found that they accurately predict the properties of\\n                                     untested ChR sequences. We used these models to engineer 30 ‘designer’ ChR variants with\\nAuthor Manuscript                    specific combinations of desired properties, a number of which have unprecedented\\n                                     photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-\\n                                     photocurrent ChRs for applications in the mammalian brain and demonstrate their potential\\n                                     for minimally-invasive activation of populations of neurons in the brain enabled by systemic\\n                                     transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This\\n                                     work demonstrates how a machine learning-guided approach can enable engineering of\\n                                     proteins that have been challenging to engineer.\\n\\n\\n\\n                                            Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "  'images': [{'name': 'page_2.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_2.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 2}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "    'md': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\",\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 371.86999999999983, 'h': 13}}],\n",
       "  'md_revised': \"Limitations of available ChRs prohibit a number of optogenetic applications. These channels have broad activation spectra in the visible range and require high-intensity light for activation [~1 mW mm−2]. ChRs are naturally low-conductance channels requiring approximately 105–106 functional ChRs expressed in the plasma-membrane of a neuron to produce sufficient light-dependent depolarization to induce neuronal activation3. When applied to the mouse brain, ChRs require ~1–15 mW light delivered ~100 μm from the target cell population to reliably activate action potentials4–6. This confines light-dependent activation to a small volume of brain tissue [~1 mm3]7. Enabling optogenetics for large brain volumes without the need to implant invasive optical fibers for light delivery would be highly desirable for neuroscience applications.\\n\\nEngineering ChRs to overcome limits in conductance and light sensitivity and extend the reach of optogenetic experiments requires overcoming three major challenges. First, rhodopsins are trans-membrane proteins that are inherently difficult to engineer because the sequence and structural determinants of membrane protein expression and plasma-membrane localization are highly constrained and poorly understood8,9. Second, because properties of interest for neuroscience applications are assayed using low-throughput techniques, such as patch-clamp electrophysiology, engineering by directed evolution is not feasible10. And third, in vivo applications require retention and optimization of multiple properties, for example, localization in mammalian cells while simultaneously tuning kinetics, photocurrents, and spectral properties6.\\n\\nDiverse ChRs have been published, including variants discovered from nature11,12, variants engineered through recombination9,13 and mutagenesis14,15, as well as variants resulting from rational design16. Studies of these coupled with structural information17 and molecular dynamic simulations18 have established some understanding of the mechanics and sequence features important for specific ChR properties1,16. Despite this, it is still not possible to predict functional properties of new ChR sequences.\\n\\nOur approach has been to leverage the significant literature of ChRs to train statistical models that enable design of highly-functional ChRs. These models take as their input the sequence and structural information for a given ChR variant and then predict its functional properties. The models use training data to learn how sequence and structural elements map to ChR function. Once known, that mapping can be used to predict the behavior of untested ChR variants.\\n\\nWe trained models in this manner and found that they accurately predict the properties of untested ChR sequences. We used these models to engineer 30 'designer' ChR variants with specific combinations of desired properties, a number of which have unprecedented photocurrent strength and light sensitivity. We characterized these low-light sensitive, high-photocurrent ChRs for applications in the mammalian brain and demonstrate their potential for minimally-invasive activation of populations of neurons in the brain enabled by systemic transgene delivery with the engineered adeno-associated virus (AAV), rAAV-PHP.eB19. This work demonstrates how a machine learning-guided approach can enable engineering of proteins that have been challenging to engineer.\"},\n",
       " {'page': 3,\n",
       "  'text': '                   Bedbrook et al.                                                                                                          Page 3\\n\\n\\n\\nAuthor ManuscriptResults\\n                       Functional characterization of ChR variants for machine learning\\n                                          In previous work, we explored structure-guided recombination20,21 of three highly-\\n                                          functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing\\n                                          two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310)\\n                                          chimeric variants9. These recombination libraries are a rich source of functionally diverse\\n                                          sequences9. Previously, we generated 102 ChR variants selected from these recombination\\n                                          libraries8,9, 76 of these were previously characterized for photocurrent properties (with\\n                                          patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR\\n                                          recombination variants provide the primary dataset used for model training. We\\n                                          supplemented this dataset with data from other published sources including 19 ChR variants\\nAuthor Manuscript                         from nature, 14 single-mutant ChR variants, and 28 recombination variants from other\\n                                          libraries (Dataset 1). As the data produced by other labs were not collected under the same\\n                                          experimental conditions as data collected in our hands, they cannot be used for comparison\\n                                          for absolute ChR properties (i.e. photocurrent strength); however, these data do provide\\n                                          useful binary information on whether a sequence variant is functional or not. Thus, we used\\n                                          published data from other sources when training binary classification models for ChR\\n                                          function.\\n\\n\\n\\n                                          Because our goal was to optimize photocurrent strength, wavelength sensitivity, and off-\\n                                          kinetics, we used these measured properties to train machine-learning models (Figure 1a).\\n                                          Enhancing ChR photocurrent strength would enable reliable neuronal activation even under\\n                                          low-light conditions. Different off-rates can be useful for specific applications: fast off-\\n                                          kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with\\nAuthor Manuscript                         increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant\\n                                          depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it\\n                                          is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for\\n                                          ChR function8.\\n\\n\\n\\n                       Training Gaussian process (GP) classification and regression models\\n                                          Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process\\n                                          (GP) classification and regression models (Figure 1). GP models successfully predicted\\n                                          thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and,\\n                                          more recently, ChR membrane localization8. For a detailed description of the GP model\\n                                          architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive\\n                                          values for new sequences from training examples by assuming that similar inputs (ChR\\nAuthor Manuscript                         sequence variants) will have similar outputs (photocurrent properties). To quantify the\\n                                          relatedness of inputs (ChR sequence variants), we compared both sequence and structure.\\n                                          ChR sequence information is encoded in the amino acid sequence. For structural\\n                                          comparisons, we convert the 3D crystal-structural information into a ‘contact map’ that is\\n                                          convenient for modeling. Two residues are considered to be in contact and potentially\\n                                          important for structural and functional integrity if they have any non-hydrogen atoms within\\n                                          4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\\n\\n\\n\\n                                                 Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"# Results\\n\\n## Functional characterization of ChR variants for machine learning\\n\\nIn previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\\n\\n## Training Gaussian process (GP) classification and regression models\\n\\nUsing the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "  'images': [{'name': 'page_3.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_3.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 3}],\n",
       "  'items': [{'type': 'heading',\n",
       "    'lvl': 1,\n",
       "    'value': 'Results',\n",
       "    'md': '# Results',\n",
       "    'bBox': {'x': 89, 'y': 75, 'w': 43.344, 'h': 12}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Functional characterization of ChR variants for machine learning',\n",
       "    'md': '## Functional characterization of ChR variants for machine learning',\n",
       "    'bBox': {'x': 90, 'y': 95, 'w': 307.2499999999999, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.',\n",
       "    'md': 'In previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.',\n",
       "    'bBox': {'x': 150, 'y': 110, 'w': 370.22, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Training Gaussian process (GP) classification and regression models',\n",
       "    'md': '## Training Gaussian process (GP) classification and regression models',\n",
       "    'bBox': {'x': 90, 'y': 485, 'w': 329.47, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "    'md': \"Using the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\",\n",
       "    'bBox': {'x': 150, 'y': 503, 'w': 371.8799999999999, 'h': 13}}],\n",
       "  'md_revised': \"# Results\\n\\n## Functional characterization of ChR variants for machine learning\\n\\nIn previous work, we explored structure-guided recombination20,21 of three highly-functional ChR parents [CsChrimsonR (CsChrimR)11, C1C217, and CheRiff22] by designing two 10-block recombination libraries with a theoretical size of ~120,000 (i.e. 2×310) chimeric variants9. These recombination libraries are a rich source of functionally diverse sequences9. Previously, we generated 102 ChR variants selected from these recombination libraries8,9, 76 of these were previously characterized for photocurrent properties (with patch-clamp electrophysiology) and 26 we characterized here. Together, these 102 ChR recombination variants provide the primary dataset used for model training. We supplemented this dataset with data from other published sources including 19 ChR variants from nature, 14 single-mutant ChR variants, and 28 recombination variants from other libraries (Dataset 1). As the data produced by other labs were not collected under the same experimental conditions as data collected in our hands, they cannot be used for comparison for absolute ChR properties (i.e. photocurrent strength); however, these data do provide useful binary information on whether a sequence variant is functional or not. Thus, we used published data from other sources when training binary classification models for ChR function.\\n\\nBecause our goal was to optimize photocurrent strength, wavelength sensitivity, and off-kinetics, we used these measured properties to train machine-learning models (Figure 1a). Enhancing ChR photocurrent strength would enable reliable neuronal activation even under low-light conditions. Different off-rates can be useful for specific applications: fast off-kinetics enable high-frequency optical stimulation23, slow off-kinetics is correlated with increased light sensitivity3,14,15, and very slow off-kinetics can be used for constant depolarization (step-function opsins [SFOs]14). In addition to opsin functional properties, it is also necessary to optimize or maintain plasma-membrane localization, a prerequisite for ChR function8.\\n\\n## Training Gaussian process (GP) classification and regression models\\n\\nUsing the ChR sequence/structure and functional data as inputs, we trained Gaussian process (GP) classification and regression models (Figure 1). GP models successfully predicted thermostability, substrate binding affinity, and kinetics for several soluble enzymes24, and, more recently, ChR membrane localization8. For a detailed description of the GP model architecture used for protein engineering see refs 8, 24. Briefly, these models infer predictive values for new sequences from training examples by assuming that similar inputs (ChR sequence variants) will have similar outputs (photocurrent properties). To quantify the relatedness of inputs (ChR sequence variants), we compared both sequence and structure. ChR sequence information is encoded in the amino acid sequence. For structural comparisons, we convert the 3D crystal-structural information into a 'contact map' that is convenient for modeling. Two residues are considered to be in contact and potentially important for structural and functional integrity if they have any non-hydrogen atoms within 4.5 Å in the C1C2 crystal structure (3UG9.pdb)17. We defined the sequence and structural\"},\n",
       " {'page': 4,\n",
       "  'text': '                   Bedbrook et al.                                                                                                        Page 4\\n\\n\\n\\nAuthor Manuscript                          similarity between two variants by aligning them and counting the number of positions and\\n                                           contacts at which they are identical24.\\n                                          We trained a binary classification model to predict if a ChR sequence will be functional\\n                                          using all 102 training sequences from the recombination library (Dataset 2) as well as data\\n                                          from 61 variants published by others (Dataset 1). We then used this trained classification\\n                                          model to predict whether uncharacterized ChR sequence variants were functional (Figure\\n                                          1b). To test prediction accuracy, we performed 20-fold cross validation on the training data\\n                                          set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good\\n                                          predictive power (Supplemental Table 1).\\n\\n\\n\\n                                          Next, we trained three regression models, one for each of the ChR photocurrent properties of\\n                                          interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics\\nAuthor Manuscript                         (Figure 1c). Once trained, these models were used to predict photocurrent properties of new,\\n                                          untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross\\n                                          validation on the training dataset and observed high correlation between predicted and\\n                                          measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental\\n                                          Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or\\n                                          C1Chrimson crystal structure26 perform as well as models built with a contact map from the\\n                                          C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps\\n                                           share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental\\n                                           Figure 1a,b).\\n\\n\\n\\n                       Selection of designer ChRs using trained models\\n                                           To select ‘designer’ ChRs (i.e. ChRs predicted to have a useful combination of properties),\\nAuthor Manuscript                          we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to\\n                                           not localize to the plasma membrane or predicted to be non-functional. To do this, we used\\n                                           classification models of ChR localization8 and function to predict the probability of\\n                                           localization and function for each ChR sequence in the 120,000-variant recombination\\n                                           library. Not surprisingly, most ChR variants were predicted to not localize and not function.\\n                                           To focus on ChR variants predicted to localize and function, we set a threshold for the\\n                                           product of the predicted probabilities of localization and function (Figure 1b); any ChR\\n                                           sequence above that threshold would be considered for the next tier of the process. We\\n                                           selected a conservative threshold of 0.4.\\n\\n\\n\\n                                           The training data made clear that the higher the mutation distance from one of the three\\n                                           parents, the less likely it was that a sequence would be functional; however, we expect that\\nAuthor Manuscript                          more diverse sequences would also have the more diverse functional properties. To explore\\n                                           diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4\\n                                           threshold and were multi-block-swap sequences containing on average 70 mutations from\\n                                           the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their\\n                                           photocurrent properties were measured with patch-clamp electrophysiology. 59% of the\\n                                           tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap\\n                                           sequences randomly selected (i.e., not selected by the model) and having comparable\\n                                           average mutation level. This validates the classification model’s ability to make useful\\n                                           predictions about novel functional sequences, even for sequences that are very distant from\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\\n\\n## Selection of designer ChRs using trained models\\n\\nTo select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "  'images': [{'name': 'page_4.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_4.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 4}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).',\n",
       "    'md': 'similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).',\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 371.5999999999999, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Selection of designer ChRs using trained models',\n",
       "    'md': '## Selection of designer ChRs using trained models',\n",
       "    'bBox': {'x': 90, 'y': 383, 'w': 233.3900000000001, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "    'md': \"To select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\",\n",
       "    'bBox': {'x': 150, 'y': 415, 'w': 366.18999999999977, 'h': 13}}],\n",
       "  'md_revised': \"similarity between two variants by aligning them and counting the number of positions and contacts at which they are identical24.\\n\\nWe trained a binary classification model to predict if a ChR sequence will be functional using all 102 training sequences from the recombination library (Dataset 2) as well as data from 61 variants published by others (Dataset 1). We then used this trained classification model to predict whether uncharacterized ChR sequence variants were functional (Figure 1b). To test prediction accuracy, we performed 20-fold cross validation on the training data set and achieved an area under the receiver operator curve (AUC) of 0.78, indicating good predictive power (Supplemental Table 1).\\n\\nNext, we trained three regression models, one for each of the ChR photocurrent properties of interest: photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics (Figure 1c). Once trained, these models were used to predict photocurrent properties of new, untested ChRs sequence variants. To test prediction accuracy, we performed 20-fold cross validation on the training dataset and observed high correlation between predicted and measured properties for all models (Pearson correlation [R] between 0.77–0.9; Supplemental Table 1–2). Models built using contact maps from either the ChR2 crystal structure25 or C1Chrimson crystal structure26 perform as well as models built with a contact map from the C1C2 structure17 (Supplemental Table 3, Supplemental Figure 1c,d) even though these maps share only 82% and 89% of their contacts with the C1C2 map, respectively (Supplemental Figure 1a,b).\\n\\n# Selection of designer ChRs using trained models\\n\\nTo select 'designer' ChRs (i.e. ChRs predicted to have a useful combination of properties), we used a tiered approach (Figure 1d). First, we eliminated all ChR sequences predicted to not localize to the plasma membrane or predicted to be non-functional. To do this, we used classification models of ChR localization8 and function to predict the probability of localization and function for each ChR sequence in the 120,000-variant recombination library. Not surprisingly, most ChR variants were predicted to not localize and not function. To focus on ChR variants predicted to localize and function, we set a threshold for the product of the predicted probabilities of localization and function (Figure 1b); any ChR sequence above that threshold would be considered for the next tier of the process. We selected a conservative threshold of 0.4.\\n\\nThe training data made clear that the higher the mutation distance from one of the three parents, the less likely it was that a sequence would be functional; however, we expect that more diverse sequences would also have the more diverse functional properties. To explore diverse sequences predicted to function, we selected 22 ChR variants that passed the 0.4 threshold and were multi-block-swap sequences containing on average 70 mutations from the closest parent. These 22 sequences were synthesized, expressed in HEK cells, and their photocurrent properties were measured with patch-clamp electrophysiology. 59% of the tested sequences were functional (Figure 1e), compared to 38% of the multi-block swap sequences randomly selected (i.e., not selected by the model) and having comparable average mutation level. This validates the classification model's ability to make useful predictions about novel functional sequences, even for sequences that are very distant from\"},\n",
       " {'page': 5,\n",
       "  'text': '                   Bedbrook et al.                                                                                                           Page 5\\n\\n\\n\\nAuthor Manuscript                          those previously tested. We updated the models by including data from these 22 sequences\\n                                           for future rounds of predictions.\\n\\n\\n\\n                                           From the 120,000-variant recombination library, 1,161 chimeric sequence variants passed\\n                                           the conservative 0.4 predicted localization and function threshold (Figure 1). For the second\\n                                           tier of the selection process, we used the three regression models trained on all functional\\n                                           variants collected up to this point to predict the photocurrent strength, wavelength sensitivity\\n                                           of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3).\\n                                           We selected 28 designer ChRs predicted to be highly functional with different combinations\\n                                           of properties including those predicted to have the highest photocurrent strength, most red-\\n                                           shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow\\n                                           (Supplemental Figure 2–3).\\n\\n\\n\\nAuthor Manuscript                          Genes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK\\n                                           cells, and characterized for their photocurrent properties with patch-clamp\\n                                           electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected\\n                                           using the updated classification model above the 0.4 threshold both localize and function.\\n                                           For each of the designer ChR variants, the measured photocurrent properties correlated well\\n                                           with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This\\n                                           outstanding performance on a novel set of sequences demonstrates the power of this data-\\n                                           driven predictive method for engineering designer ChRs. As a negative control, we selected\\n                                           two ChR variant sequences from the recombination library that the model predicted would\\n                                           be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-\\n                                           block swap from two of the most highly functional ChR recombination variants tested. As\\n                                           predicted, these sequences were non-functional (Figure 2b), which shows that ChR\\nAuthor Manuscript                          functionality can be attenuated by incorporating even minimal diversity at certain positions.\\n                       Sequence and structural determinants of ChR functional properties\\n                                           We used L1-regularized linear regression models to identify a limited set of residues and\\n                                           structural contacts that strongly influence ChR photocurrent strength, spectral properties,\\n                                           and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of\\n                                           these sequence and structural features by weighting their contributions using L2-regularized\\n                                           linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we\\n                                           identified a set of important residues and contacts and their respective weights. A specific\\n                                           residue or contact at a given position is weighted as likely to lead to, e.g., low (negative\\n                                           weight) or high (positive weight) photocurrents. A number of residues and contacts most\\n                                           important for tuning spectral properties are proximal to the retinal-binding pocket, including\\nAuthor Manuscript                          the blue-shifting contact between A206 and F269 and the blue-shifting contact between\\n                                           F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the\\n                                           red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR\\n                                           parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-\\n                                           kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental\\n                                           Figure 4), a residue that is part of the DC-gate1. Mutation of either the aspartic acid or\\n                                           cysteine within the DC-gate has been shown to decrease off-kinetic speed14,27. While the\\n                                           cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\\n\\n## Sequence and structural determinants of ChR functional properties\\n\\nWe used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "  'images': [{'name': 'page_5.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_5.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 5}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.',\n",
       "    'md': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.',\n",
       "    'bBox': {'x': 150, 'y': 71, 'w': 370.5199999999998, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Sequence and structural determinants of ChR functional properties',\n",
       "    'md': '## Sequence and structural determinants of ChR functional properties',\n",
       "    'bBox': {'x': 90, 'y': 439, 'w': 320.06, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "    'md': 'We used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position',\n",
       "    'bBox': {'x': 150, 'y': 457, 'w': 369.2999999999998, 'h': 10}}],\n",
       "  'md_revised': 'those previously tested. We updated the models by including data from these 22 sequences for future rounds of predictions.\\n\\nFrom the 120,000-variant recombination library, 1,161 chimeric sequence variants passed the conservative 0.4 predicted localization and function threshold (Figure 1). For the second tier of the selection process, we used the three regression models trained on all functional variants collected up to this point to predict the photocurrent strength, wavelength sensitivity of photocurrents, and off-kinetics for each of these 1,161 ChR sequence variants (Dataset 3). We selected 28 designer ChRs predicted to be highly functional with different combinations of properties including those predicted to have the highest photocurrent strength, most red-shifted or blue-shifted activation wavelengths, and off-kinetics from very fast to very slow (Supplemental Figure 2–3).\\n\\nGenes encoding the 28 selected designer ChR variants were synthesized, expressed in HEK cells, and characterized for their photocurrent properties with patch-clamp electrophysiology. All 28 selected designer ChRs were functional: 100% of variants selected using the updated classification model above the 0.4 threshold both localize and function. For each of the designer ChR variants, the measured photocurrent properties correlated well with the model predictions (R>0.9 for all models) (Figure 1f, Supplemental Table 1). This outstanding performance on a novel set of sequences demonstrates the power of this data-driven predictive method for engineering designer ChRs. As a negative control, we selected two ChR variant sequences from the recombination library that the model predicted would be non-functional (ChR_29_10 and ChR_30_10). These sequences resulted from a single-block swap from two of the most highly functional ChR recombination variants tested. As predicted, these sequences were non-functional (Figure 2b), which shows that ChR functionality can be attenuated by incorporating even minimal diversity at certain positions.\\n\\n# Sequence and structural determinants of ChR functional properties\\n\\nWe used L1-regularized linear regression models to identify a limited set of residues and structural contacts that strongly influence ChR photocurrent strength, spectral properties, and off-kinetics (Supplemental Figure 4a). We can assess the relative importance of each of these sequence and structural features by weighting their contributions using L2-regularized linear regression (Dataset 4 and Supplemental Figure 4). For each functional property, we identified a set of important residues and contacts and their respective weights. A specific residue or contact at a given position is weighted as likely to lead to, e.g., low (negative weight) or high (positive weight) photocurrents. A number of residues and contacts most important for tuning spectral properties are proximal to the retinal-binding pocket, including the blue-shifting contact between A206 and F269 and the blue-shifting contact between F265 and I267 that are conserved in the blue-shifted parents C1C2 and CheRiff while the red-shifting contact between F201 and Y217 originates from the red-shifted CsChrimR parent (Supplemental Figure 4). The most heavily weighted contact contributing to off-kinetics includes the reside D195 (i.e., D156 according to ChR2 numbering) (Supplemental Figure 4), a residue that is part of the DC-gate¹. Mutation of either the aspartic acid or cysteine within the DC-gate has been shown to decrease off-kinetic speed¹⁴,²⁷. While the cysteine in the DC-gate is conserved in all three ChR parents, the aspartic acid at position'},\n",
       " {'page': 6,\n",
       "  'text': '                   Bedbrook et al.                                                                                                          Page 6\\n\\n\\n\\nAuthor Manuscript                         195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at\\n                                          that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly\\n                                          to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal\\n                                          contribute strongly to photocurrent strength. For example, the most heavily weighted contact\\n                                          includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that\\n                                          covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and\\n                                          CheRiff.\\n\\n\\n\\n                      Machine-guided search identifies ChRs with a range of useful functional properties\\n                                          We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer\\n                                          ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-\\n                                          predicted ChRs, we also assessed the two top-performing ChRs from the classification\\nAuthor Manuscript                         models’ predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-\\n                                          predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the\\n                                          30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated\\n                                          photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants\\n                                          exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants\\n                                          have larger red-light activated photocurrents when compared with the blue-light activated\\n                                          parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants\\n                                          predicted to be non-functional by the models produce <30 pA currents.\\n\\n\\n\\n                                          Designer ChRs’ off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure\\n                                          2c). This range is quite remarkable given that all designer ChRs are built from sequence\\n                                          blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5\\n                                          designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower\\nAuthor Manuscript                         off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms\\n                                          and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics\\n                                          with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5\\n                                          s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and\\n                                          ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\n\\n\\n                                          Three designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure\\n                                          5). ChR_28_10’s red-shifted spectrum matches that of CsChrimR, demonstrating that\\n                                          incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a\\n                                          red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties:\\n                                          ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar\\n                                          steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA)\\nAuthor Manuscript                         when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow\\n                                          activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\n\\n\\n                                          We assessed the light sensitivity of select designer ChRs. Compared with CsChrimR,\\n                                          CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity\\n                                          of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal\\n                                          decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm\\n                                          −2), suggesting that photocurrents were saturated at these intensities and would only\\n                                          attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\\n\\n\\n\\n                                                 Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\\n\\n## Machine-guided search identifies ChRs with a range of useful functional properties\\n\\nWe assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "  'images': [{'name': 'page_6.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_6.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 6}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': '195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.',\n",
       "    'md': '195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.',\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 370.64999999999986, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Machine-guided search identifies ChRs with a range of useful functional properties',\n",
       "    'md': '## Machine-guided search identifies ChRs with a range of useful functional properties',\n",
       "    'bBox': {'x': 89, 'y': 181, 'w': 394.2299999999999, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "    'md': \"We assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\",\n",
       "    'bBox': {'x': 149, 'y': 199, 'w': 371.80999999999995, 'h': 13}}],\n",
       "  'md_revised': \"# Machine-guided search identifies ChRs with a range of useful functional properties\\n\\n195 is only conserved in CheRiff and C1C2 but not in CsChrimR, which has a cysteine at that position. Interestingly, D195 is also part of a contact with L192 that contributes strongly to photocurrent strength (Supplemental Figure 4). A number of contacts proximal to retinal contribute strongly to photocurrent strength. For example, the most heavily weighted contact includes A295 (from CsChrimR), which is adjacent to the conserved lysine residue that covalently links retinal (Supplemental Figure 4). This position is a serine in both C1C2 and CheRiff.\\n\\nWe assessed photocurrent amplitude, wavelength sensitivity, and off-kinetics of the designer ChRs and the three parental ChRs (Figure 2). In addition to the 28 regression model-predicted ChRs, we also assessed the two top-performing ChRs from the classification models' predictions (ChR_9_4 and ChR_25_9), for a total of 30 highly-functional model-predicted ChRs as well as the two negative control ChRs (ChR_29_10, ChR_30_10). Of the 30 model-predicted ChRs, we found 12 variants with ≥2-times higher blue-light activated photocurrents than the top-performing parent (CsChrimR) (Figure 2b). Three variants exhibit ≥1.7-times higher green-light activated photocurrents than CsChrimR. Eight variants have larger red-light activated photocurrents when compared with the blue-light activated parents (CheRiff and C1C2), though none out-perform CsChrimR. Both ChR variants predicted to be non-functional by the models produce <30 pA currents.\\n\\nDesigner ChRs' off-kinetics span three orders of magnitude (τoff = 10 ms – >10 s) (Figure 2c). This range is quite remarkable given that all designer ChRs are built from sequence blocks of three parents that have similar off-kinetics (τoff = 30–50 ms). We found that 5 designer ChRs have faster off-kinetics than the fastest parent, while 16 have >5-times slower off-kinetics. The two fastest variants, ChR_3_10 and ChR_21_10 exhibit τoff = 13±0.9 ms and 12±0.4 ms, respectively (mean±SEM). Four ChRs have particularly slow off-kinetics with τoff > 1 s, including ChR_15_10, ChR_6_10, and ChR_13_10 (τoff = 4.3±0.1 s, 8.0±0.5 s, and 17±7 s, respectively). Two ChRs with very large photocurrents, ChR_25_9 and ChR_11_10, exhibit τoff = 220±10 ms and 330±30 ms, respectively.\\n\\nThree designer ChRs exhibit interesting spectral properties (Figure 2e, Supplemental Figure 5). ChR_28_10's red-shifted spectrum matches that of CsChrimR, demonstrating that incorporating sequence elements from blue-shifted ChRs into CsChrimR can still generate a red-shifted activation spectrum. Two designer ChRs exhibit novel spectral properties: ChR_11_10 has a broad activation spectrum relative to the parental spectra, with similar steady-state current strength from 400–546 nm light and strong currents (700 ± 100 pA) when activated with 567 nm light. ChR_25_9, on the other hand, exhibits a narrow activation spectrum relative to the parental spectra, with a peak at 481 nm light.\\n\\nWe assessed the light sensitivity of select designer ChRs. Compared with CsChrimR, CheRiff, and C1C2, the designer ChRs have ≥9-times larger currents at the lowest intensity of light tested (10−1 mW mm−2), larger currents at all intensities of light tested, and minimal decrease in photocurrent magnitude over the range of intensities tested (10−1–101 mW mm−2), suggesting that photocurrents were saturated at these intensities and would only attenuate at much lower light intensities (Figure 2d). These select designer ChRs are\"},\n",
       " {'page': 7,\n",
       "  'text': '                  Bedbrook et al.                                                                                                           Page 7\\n\\n\\n\\nAuthor Manuscript                        expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating\\n                                         that the improved photocurrent strength of these ChRs is not solely due to improved\\n                                         expression (Supplemental Figure 6–7).\\n\\n\\n\\n                                         We compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with\\n                                         ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used forin\\n                                         vivo optogenetics, and CoChR (fromChloromonas oogama)11, reported to be one of the\\n                                         highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–\\n                                         6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2)\\n                                         481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-\\n                                         intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR\\n                                         produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much\\n                                         lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x\\nAuthor Manuscript                        larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x\\n                                         larger steady-state photocurrents than CoChR when exposed to low-intensity light\\n                                         (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of\\n                                         these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to\\n                                         the increased accumulation of the open state under low-light conditions14.\\n\\n\\n\\n                      Validation of designer ChRs for neuroscience applications\\n                                         For further validation we selected three of the top high-conductance ChRs, ChR_9_4,\\n                                         ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3,\\n                                         respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin\\n                                         (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust\\n                                         membrane localization and expression throughout the neuron soma and neurites (Figure 3b).\\nAuthor Manuscript                        The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-\\n                                         intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of\\n                                         magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation\\n                                         (Figure 3d, Supplemental Figure 8h).\\n\\n\\n\\n                                         Next, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of\\n                                         rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter\\n                                         (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in\\n                                         ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the\\n                                         ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA\\n                                         photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits\\n                                         undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW\\nAuthor Manuscript                        mm−2 light, a four-fold improvement over ChR2(H134R)’s irradiance-matched\\n                                         photocurrents (Figure 3g). Our characterization of ChR2(H134R)’s light sensitivity and\\n                                         photocurrent strength is consistent with previous results from other labs6,22.\\n\\n\\n\\n                      Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n                                         We investigated whether these light-sensitive, high-photocurrent ChRs could provide\\n                                         optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of\\n                                         ‘non-invasive optogenetics’ relied on invasive intracranial virus delivery, which results in\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\\n\\n## Validation of designer ChRs for neuroscience applications\\n\\nFor further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\\n\\n## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n\\nWe investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "  'images': [{'name': 'page_7.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_7.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 7}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.',\n",
       "    'md': 'expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.',\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 371.9, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Validation of designer ChRs for neuroscience applications',\n",
       "    'md': '## Validation of designer ChRs for neuroscience applications',\n",
       "    'bBox': {'x': 89, 'y': 331, 'w': 276.85000000000014, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\",\n",
       "    'md': \"For further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\",\n",
       "    'bBox': {'x': 149, 'y': 137, 'w': 371.1199999999997, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation',\n",
       "    'md': '## Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation',\n",
       "    'bBox': {'x': 89, 'y': 637, 'w': 400.65000000000003, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "    'md': \"We investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\",\n",
       "    'bBox': {'x': 149, 'y': 655, 'w': 358.21999999999986, 'h': 10}}],\n",
       "  'md_revised': \"expressed at levels similar to the CsChrimR parent (the highest expressing parent) indicating that the improved photocurrent strength of these ChRs is not solely due to improved expression (Supplemental Figure 6–7).\\n\\nWe compared three of the top designer ChRs (ChR_9_4, ChR_25_9, and ChR_11_10) with ChR2(H134R)6,28, an enhanced photocurrent single mutant of ChR2 commonly used for in vivo optogenetics, and CoChR (from Chloromonas oogama)11, reported to be one of the highest conducting ChRs activated by blue light11. The selected designer ChRs produce 3–6x larger photocurrents than ChR2(H134R) when exposed to high-intensity (2.2 mW mm−2) 481 nm light and 10–18x larger photocurrents than ChR2(H134R) when exposed to low-intensity (6.5×10−2 mW mm−2) 481 nm light (Supplemental Figure 8f,g). Although CoChR produced peak currents of similar magnitude to the designer ChRs, CoChR decays to a much lower steady-state level (Supplemental Figure 8d,e) with the designer ChRs producing 2–3x larger steady-state photocurrents than CoChR when exposed to high-intensity light and 3–4x larger steady-state photocurrents than CoChR when exposed to low-intensity light (Supplemental Figure 8f,g; Supplemental Table 4). The increased low-light sensitivity of these select designer ChRs is likely due in part to their relatively slow off-kinetics leading to the increased accumulation of the open state under low-light conditions14.\\n\\n# Validation of designer ChRs for neuroscience applications\\n\\nFor further validation we selected three of the top high-conductance ChRs, ChR_9_4, ChR_25_9, and ChR_11_10, and renamed them ChRger1, ChRger2, and ChRger3, respectively, for channelrhodopsin Gaussian process-engineered recombinant opsin (Supplemental Figure 9). When expressed in cultured neurons, the ChRgers display robust membrane localization and expression throughout the neuron soma and neurites (Figure 3b). The ChRgers outperform both CoChR and ChR2(H134R) in photocurrent strength with low-intensity light in neuronal cultures (Figure 3c). The designer ChRgers require 1–2 orders of magnitude lower light intensity than CoChR and ChR2(H134R) for neuronal activation (Figure 3d, Supplemental Figure 8h).\\n\\nNext, we performed direct intracranial injections into the mouse prefrontal cortex (PFC) of rAAV-PHP.eB packaging either ChRger1–3, or ChR2(H134R) under the hSyn promoter (Supplemental Table 5). After 3–5 weeks of expression, we measured light sensitivity in ChR-expressing neurons in acute brain slices. We observed greater light sensitivity for the ChRgers compared with ChR2(H134R) (Figure 3g,h). The ChRgers exhibit >200 pA photocurrent at 10−3 mW mm−2 while at the equivalent irradiance ChR2(H134R) exhibits undetectable photocurrents. The ChRgers reach >1000 pA photocurrents with ~10−2 mW mm−2 light, a four-fold improvement over ChR2(H134R)'s irradiance-matched photocurrents (Figure 3g). Our characterization of ChR2(H134R)'s light sensitivity and photocurrent strength is consistent with previous results from other labs6,22.\\n\\n# Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation\\n\\nWe investigated whether these light-sensitive, high-photocurrent ChRs could provide optogenetic activation coupled with minimally-invasive gene delivery. Previous reports of 'non-invasive optogenetics' relied on invasive intracranial virus delivery, which results in\"},\n",
       " {'page': 8,\n",
       "  'text': '                   Bedbrook et al.                                                                                                  Page 8\\n\\n\\n\\nAuthor Manuscript                   many copies of virus per cell and thus very high expression levels of the injected\\n                                    construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad\\n                                    transduction throughout the central nervous system with a single minimally-invasive\\n                                    intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in\\n                                    brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-\\n                                    PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of\\n                                    infection with systemically delivered viral vectors resulting in insufficient opsin expression\\n                                    and light-evoked currents to control neuronal firing with commonly-used channels (e.g.\\n                                    ChR2).\\n\\n\\n\\n                                    We hypothesized that the ChRgers could overcome this limitation and allow large-volume\\n                                    optogenetic excitation following systemic transgene delivery. We systemically delivered\\n                                    rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the\\nAuthor Manuscript                   hSyn promoter and observed broad expression throughout the brain (Figure 3i). We\\n                                    measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for\\n                                    light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced\\n                                    light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing\\n                                    neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced\\n                                    activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared\\n                                    with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l).\\n                                    ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or\\n                                    ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10;\\n                                    Supplemental Table 6). These results demonstrate the need for light-sensitive and high-\\n                                    photocurrent opsins for applications where systemic delivery is desired.\\n\\n\\n\\nAuthor Manuscript                   We systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa\\n                                    promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to\\n                                    results observed after direct injection into the PFC (Figure 3g). When expressed in\\n                                    pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced\\n                                    firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency\\n                                    stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while\\n                                    ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher\\n                                    frequency stimulation (20–40 Hz) (Figure 3m).\\n\\n\\n\\n                                    We next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using\\n                                    optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral\\n                                    tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-\\nAuthor Manuscript                   floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into\\n                                    Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery\\n                                    and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an\\n                                    operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose\\n                                    poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a\\n                                    frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz)\\n                                    and higher light power (up to 10 mW) promoted greater maximum operant response rates\\n                                    (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in\\n                                    ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "  'images': [{'name': 'page_8.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_8.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 8}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "    'md': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in',\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 370.2699999999998, 'h': 13}}],\n",
       "  'md_revised': 'many copies of virus per cell and thus very high expression levels of the injected construct29. Recently, we described the AAV capsid rAAV-PHP.eB19 that produces broad transduction throughout the central nervous system with a single minimally-invasive intravenous injection in the adult mouse30,31. Systemic delivery of rAAV-PHP.eB results in brain-wide transgene delivery without invasive intracranial injections19,30,31. Use of rAAV-PHP.eB for optogenetic applications has been limited, however, by the low multiplicity of infection with systemically delivered viral vectors resulting in insufficient opsin expression and light-evoked currents to control neuronal firing with commonly-used channels (e.g. ChR2).\\n\\nWe hypothesized that the ChRgers could overcome this limitation and allow large-volume optogenetic excitation following systemic transgene delivery. We systemically delivered rAAV-PHP.eB packaging either ChRger1, ChRger2, CoChR, or ChR2(H134R) under the hSyn promoter and observed broad expression throughout the brain (Figure 3i). We measured the fraction of opsin-expressing cells with sufficient opsin-mediated currents for light-induced firing (Figure 3j). Only 4% of ChR2(H134R)-expressing neurons produced light-induced firing, while 77% of CoChR-expressing neurons, 89% of ChRger1-expressing neurons, and 100% of ChRger2- or ChRger3-expressing neurons produced light-induced activity. With systemic delivery, we observed superior light sensitivity of ChRgers compared with CoChR in both photocurrent strength (Figure 3k) and spike fidelity (Figure 3l). ChRger2-expressing neurons exhibit healthy membrane properties similar to CoChR- or ChR2(H134R)-expressing neurons both in culture and in slice (Supplemental Figure 10; Supplemental Table 6). These results demonstrate the need for light-sensitive and high-photocurrent opsins for applications where systemic delivery is desired.\\n\\nWe systemically delivered rAAV-PHP.eB packaging ChRger1–3 under the CaMKIIa promoter. With systemic delivery of ChRger2, we observed photocurrent strength similar to results observed after direct injection into the PFC (Figure 3g). When expressed in pyramidal neurons in the cortex, ChRger2 and ChRger3 enabled robust optically-induced firing at rates between 2–10 Hz, although spike fidelity was reduced at higher frequency stimulation (Figure 3m,n). ChRger2 performed best with higher frequency stimulation while ChRger1 performed worst. CoChR has better spike fidelity than the ChRgers at higher frequency stimulation (20–40 Hz) (Figure 3m).\\n\\nWe next evaluated the optogenetic efficiency of ChRger2 after systemic delivery using optogenetic intracranial self-stimulation (oICSS) of dopaminergic neurons of the ventral tegmental area (VTA)32. We systemically delivered rAAV-PHP.eB packaging a double-floxed inverted open reading frame (DIO) containing either ChRger2 or ChR2(H134R) into Dat-Cre mice (Figure 4a and Supplemental Table 5). Three weeks after systemic delivery and stereotaxic implantation of fiber-optic cannulas above the VTA, mice were placed in an operant box and were conditioned to trigger a burst of 447 nm laser stimulation via nose poke. Animals expressing ChRger2 displayed robust optogenetic self-stimulation in a frequency-dependent and laser power-dependent manner. Higher frequencies (up to 20 Hz) and higher light power (up to 10 mW) promoted greater maximum operant response rates (Figure 4a). Conversely, laser stimulation failed to reinforce operant responding in ChR2(H134R)-expressing animals (Figure 4a); these results were consistent with results in'},\n",
       " {'page': 9,\n",
       "  'text': '                   Bedbrook et al.                                                                                                  Page 9\\n\\n\\n\\nAuthor Manuscript                   acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy\\n                                    number produced by systemic delivery for robust neuronal activation.\\n\\n\\n\\n                                    In order to determine if ChRger2 would enable both minimally-invasive transgene delivery\\n                                    and minimally-invasive optical excitation, we assayed directional control of locomotion in\\n                                    freely moving animals by optogenetic stimulation of the right secondary motor cortex\\n                                    (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the\\n                                    contralateral lower extremities, causing mice to turn away from the stimulation side. We\\n                                    systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under\\n                                    a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex\\n                                    (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex\\n                                    for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We\\n                                    secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without\\nAuthor Manuscript                   puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to\\n                                    be minimally invasive. Despite the presence of the highly optically scattering calavarial\\n                                    bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals\\n                                    expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and\\n                                    Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical\\n                                    stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10\\n                                    mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by\\n                                    the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside\\n                                    the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both\\n                                    ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is\\n                                    compatible with minimally-invasive systemic gene delivery and can enable minimally-\\n                                    invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion\\nAuthor Manuscript                   nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic\\n                                    transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\\n\\n\\n\\n                      Discussion\\n                                    We demonstrated a data-driven approach to engineering ChR properties that enables efficient\\n                                    discovery of highly functional ChR variants based on data from relatively few variants. In\\n                                    this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric\\n                                    ChRs and use it to efficiently search sequence space and select top-performing variants for a\\n                                    given property10,24,34. By first eliminating the vast majority of non-functional sequences, we\\n                                    can focus on local peaks scattered throughout the landscape. Then, using regression models,\\n                                    we predict which sequences lie on the fitness peaks.\\nAuthor Manuscript                   Machine learning provides a platform for simultaneous optimization of multiple ChR\\n                                    properties that follow engineering specifications. We were able to generate variants with\\n                                    large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the\\n                                    parental or other commonly used ChRs. We also use the machine-learning models to identify\\n                                    the residues and contacts most important for ChR function. Application of this machine-\\n                                    learning pipeline (limited data collection from diverse sequences, model training and\\n                                    validation, and prediction and testing of new sequences) has great potential to optimize other\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\\n\\n## Discussion\\n\\nWe demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "  'images': [{'name': 'page_9.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_9.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 9}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.',\n",
       "    'md': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.',\n",
       "    'bBox': {'x': 150, 'y': 71, 'w': 371.3699999999998, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Discussion',\n",
       "    'md': '## Discussion',\n",
       "    'bBox': {'x': 89, 'y': 461, 'w': 64.01999999999998, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': 'We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "    'md': 'We demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other',\n",
       "    'bBox': {'x': 149, 'y': 481, 'w': 371.9299999999996, 'h': 13}}],\n",
       "  'md_revised': 'acute slice where the light-induced currents of ChR2(H134R) are too weak at the low copy number produced by systemic delivery for robust neuronal activation.\\n\\nIn order to determine if ChRger2 would enable both minimally-invasive transgene delivery and minimally-invasive optical excitation, we assayed directional control of locomotion in freely moving animals by optogenetic stimulation of the right secondary motor cortex (M2)33. In this assay, unilateral stimulation of M2 disrupts motor function in the contralateral lower extremities, causing mice to turn away from the stimulation side. We systemically administered rAAV-PHP.eB packaging either ChRger2 or ChR2(H134R) under a CaMKIIa promoter for transgene expression in excitatory pyramidal neurons in the cortex (Figure 4b, and Supplemental Table 5). We observed broad expression throughout the cortex for both ChRger2 and ChR2(H134R) injected animals (Supplemental Figure 11). We secured a fiber-optic cannula guide to the surface of the thinned skull above M2 without puncturing the dura and therefore leaving the brain intact (Figure 4b), which we consider to be minimally invasive. Despite the presence of the highly optically scattering calavarial bone, stimulation with 20 mW 447 nm light induced left-turning behavior in animals expressing ChRger2 but not in animals expressing ChR2(H134R) (Figure 4b and Supplemental Video 1–2). Left-turning behavior terminated upon conclusion of optical stimulation (Supplemental Video 1). Behavioral effects were seen at powers as low as 10 mW. To ensure that the turning behavior was not due to visual stimuli or heating caused by the stimulation laser, we repeated treadmill experiments using 671 nm light, which is outside the excitation spectrum of both opsins. 20 mW 671 nm light failed to induce turning in both ChRger2 and ChR2(H124R). Overall, these experiments demonstrate that ChRger2 is compatible with minimally-invasive systemic gene delivery and can enable minimally-invasive optogenetic excitation. Coupling ChRgers with recently reported upconversion nanoparticles may allow for non-invasive optogenetics in deep brain areas with systemic transgene delivery and tissue-penetrating near-infrared (NIR) light for neuronal excitation29.\\n\\n# Discussion\\n\\nWe demonstrated a data-driven approach to engineering ChR properties that enables efficient discovery of highly functional ChR variants based on data from relatively few variants. In this approach we approximate the ChR fitness landscape for a set of ~120,000 chimeric ChRs and use it to efficiently search sequence space and select top-performing variants for a given property10,24,34. By first eliminating the vast majority of non-functional sequences, we can focus on local peaks scattered throughout the landscape. Then, using regression models, we predict which sequences lie on the fitness peaks.\\n\\nMachine learning provides a platform for simultaneous optimization of multiple ChR properties that follow engineering specifications. We were able to generate variants with large variations in off-kinetics (10 ms to >10 s) and photocurrents that far exceed any of the parental or other commonly used ChRs. We also use the machine-learning models to identify the residues and contacts most important for ChR function. Application of this machine-learning pipeline (limited data collection from diverse sequences, model training and validation, and prediction and testing of new sequences) has great potential to optimize other'},\n",
       " {'page': 10,\n",
       "  'text': '                  Bedbrook et al.                                                                                                                 Page 10\\n\\n\\n\\nAuthor Manuscript                        neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and\\n                                         AAVs30.\\n                                         We designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and\\n                                         validated ChRger2’s application forin vivo optogenetics. The high-photocurrent properties\\n                                        of these ChRs overcome the limitation of low per-cell copy number after systemic delivery.\\n                                        ChRger2 enabled neuronal excitation with high temporal precision without invasive\\n                                        intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas,\\n                                        extending what is currently possible for optogenetics experiments.\\n\\n\\n\\n                      Online methods\\n                      Construct design and cloning\\nAuthor Manuscript                       The design, construction, and characterization of the recombination library of chimeras is\\n                                         described in detail in Bedbrooket al.9. The 10-block contiguous and 10-block\\n                                        noncontiguous recombination libraries were designed and built using SCHEMA\\n                                        recombination9. Software packages for calculating SCHEMA energies are openly available\\n                                        at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these\\n                                        libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217,\\n                                        and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9\\n                                        blocks of one parent and a single block from one of the other two parents) and multi-block-\\n                                        swap chimera sequences.\\n\\n\\n\\n                                        Selected ChR variant genes were inserted into a constant vector backbone [pFCK from\\n                                        Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS)\\nAuthor Manuscript                       sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR\\n                                        variants contain the SpyTag sequence following the N-terminal signal peptide for the\\n                                        SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36.\\n                                        The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR\\n                                        variant sequences used in this study are documented in Dataset 2. All selected ChR genes\\n                                        were synthesized and cloned in the pFCK mammalian expression vector by Twist\\n                                        Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were\\n                                        created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n                                        For characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3,\\n                                        CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene\\n                                        plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a\\n                                        pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR\\nAuthor Manuscript                       was inserted with a TS sequence5 and fluorescent protein (eYFP).\\n\\n\\n\\n                      HEK293T cell and primary neuronal cultures\\n                                        The culturing and characterization ChRs in HEK cells is described in Bedbrook                     et al.9,36.\\n                                        Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with\\n                                        10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were\\n                                        transfected with purified ChR variant DNA using FuGENE®6 reagent according to the\\n                                        manufacturer’s (Promega) recommendations. Cells were given 48 hours to express the ChRs\\n\\n\\n\\n                                               Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\\n\\n## Online methods\\n\\n### Construct design and cloning\\n\\nThe design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\\n\\n### HEK293T cell and primary neuronal cultures\\n\\nThe culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "  'images': [{'name': 'page_10.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_10.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 10}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\",\n",
       "    'md': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\",\n",
       "    'bBox': {'x': 149, 'y': 68, 'w': 368.50999999999976, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Online methods',\n",
       "    'md': '## Online methods',\n",
       "    'bBox': {'x': 89, 'y': 209, 'w': 90.68399999999997, 'h': 12}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 3,\n",
       "    'value': 'Construct design and cloning',\n",
       "    'md': '### Construct design and cloning',\n",
       "    'bBox': {'x': 89, 'y': 229, 'w': 140.93, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).',\n",
       "    'md': 'The design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).',\n",
       "    'bBox': {'x': 78, 'y': 45, 'w': 370.03999999999996, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 3,\n",
       "    'value': 'HEK293T cell and primary neuronal cultures',\n",
       "    'md': '### HEK293T cell and primary neuronal cultures',\n",
       "    'bBox': {'x': 89, 'y': 615, 'w': 209.97000000000014, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "    'md': \"The culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\",\n",
       "    'bBox': {'x': 78, 'y': 45, 'w': 365.77, 'h': 13.514623181074967}}],\n",
       "  'md_revised': \"neuroscience tools, e.g., anion-conducting ChRs12, calcium sensors, voltage sensors35, and AAVs30.\\n\\nWe designed high-performance ChRs (ChRger1–3) with unprecedented light sensitivity and validated ChRger2's application for in vivo optogenetics. The high-photocurrent properties of these ChRs overcome the limitation of low per-cell copy number after systemic delivery. ChRger2 enabled neuronal excitation with high temporal precision without invasive intracranial surgery for virus delivery or fiber optic implantation for superficial brain areas, extending what is currently possible for optogenetics experiments.\\n\\n# Online methods\\n\\n## Construct design and cloning\\n\\nThe design, construction, and characterization of the recombination library of chimeras is described in detail in Bedbrook et al.9. The 10-block contiguous and 10-block noncontiguous recombination libraries were designed and built using SCHEMA recombination9. Software packages for calculating SCHEMA energies are openly available at cheme.che.caltech.edu/groups/fha/Software.htm. Each chimeric ChR variant in these libraries is composed of blocks of sequence from the parental ChR (CsChrimR11, C1C217, and CheRiff22), including chimeras with single-block swaps (chimeras consisting of 9 blocks of one parent and a single block from one of the other two parents) and multi-block-swap chimera sequences.\\n\\nSelected ChR variant genes were inserted into a constant vector backbone [pFCK from Addgene plasmid #5169322] with a CMV promoter, Golgi export trafficking signal (TS) sequence (KSRITSEGEYIPLDQIDINV)5, and fluorescent protein (mKate). All ChR variants contain the SpyTag sequence following the N-terminal signal peptide for the SpyTag/SpyCatcher labeling assays used to characterize ChR membrane localization9,36. The C1C2 parent for the recombination libraries is mammalian codon-optimized. ChR variant sequences used in this study are documented in Dataset 2. All selected ChR genes were synthesized and cloned in the pFCK mammalian expression vector by Twist Bioscience. For visualization, sequence alignment between C1C2 and designer ChRs were created using ClustalΩ and visualized using ENDscript37 (Supplemental Figure 1c,d).\\n\\nFor characterization in neurons, selected ChR variants [ChRger1, ChRger2, ChRger3, CoChR11, and hChR2(H134R)] were inserted into a pAAV-hSyn vector backbone [Addgene plasmid #26973], a pAAV-CamKIIa vector backbone [Addgene plasmid #51087], and a pAAV-CAG-DIO vector backbone [Addgene plasmid #104052]. In all backbones, each ChR was inserted with a TS sequence5 and fluorescent protein (eYFP).\\n\\n## HEK293T cell and primary neuronal cultures\\n\\nThe culturing and characterization ChRs in HEK cells is described in Bedbrook et al.9,36. Briefly, HEK cells were cultured at 37 °C and 5% CO2 in D10 [DMEM supplemented with 10% (vol/vol) FBS, 1% sodium bicarbonate, and 1% sodium pyruvate]. HEK cells were transfected with purified ChR variant DNA using FuGENE®6 reagent according to the manufacturer's (Promega) recommendations. Cells were given 48 hours to express the ChRs\"},\n",
       " {'page': 11,\n",
       "  'text': '                  Bedbrook et al.                                                                                                      Page 11\\n\\n\\n\\nAuthor Manuscript                        before photocurrent measurements. Primary hippocampal neuronal cultures were prepped\\n                                         from C57BL/6N mouse embryos 16–18 days post-fertilization (E16–E18 Charles-River\\n                                         Labs) and cultured at 37 °C in the presence of 5% CO2 in Neurobasal media supplemented\\n                                         with glutamine and B27. Cells were transduced 3–4 days after plating with rAAV-PHP.eB\\n                                         packaging ChR2(H134R), CoChR, ChRger1, ChRger2, or ChRger3. Whole-cell recordings\\n                                         were performed 5–10 days after transduction.\\n\\n\\n\\n                      Patch-clamp electrophysiology\\n                                         Whole-cell patch-clamp and cell-attached recordings were performed in transfected HEK\\n                                         cells, transduced cultured neurons, and acute brain slices to measure light-activated inward\\n                                         currents or neuronal firing. For electrophysiological recordings, cultured cells were\\n                                         continuously perfused with extracellular solution at room temperature (in mM: 140 NaCl, 5\\nAuthor Manuscript                        KCl, 10 HEPES, 2 MgCl2, 2 CaCl2, 10 glucose; pH 7.35) while mounted on the microscope\\n                                         stage. For slice recordings, 32 °C artificial cerebrospinal fluid (ACSF) was continuously\\n                                         perfused over slices. ACSF contained 127 mM NaCl, 2.5 mM KCl, 25 mM NaHCO3, 1.25\\n                                         mM NaH2PO4, 12 mMd-glucose, 0.4 mM sodium ascorbate, 2 mM CaCl2, and 1 mM\\n                                         MgCl2 and was bubbled continuously with 95% oxygen / 5% CO2. Firing and photocurrent\\n                                         measurements were performed in the presence of 3 mM kynurenic acid and 100 μM\\n                                         picrotoxin to block optically evoked ionotropic glutamatergic and GABAergic currents,\\n                                         respectively.\\n\\n\\n\\n                                         Patch pipettes were fabricated from borosilicate capillary glass tubing (1B150–4; World\\n                                         Precision Instruments) using a model P-2000 laser puller (Sutter Instruments) to resistances\\n                                         of 3–6 MΩ. Pipettes were filled with K-gluconate intracellular solution containing the\\n                                         following (in mM): 134 K gluconate, 5 EGTA, 10 HEPES, 2 MgCl2, 0.5 CaCl2, 3 ATP, and\\nAuthor Manuscript                        0.2 GTP. Whole-cell patch-clamp and cell-attached recordings were made using a\\n                                         Multiclamp 700B amplifier (Molecular Devices), a Digidata 1440 digitizer (Molecular\\n                                         Devices), and a PC running pClamp (version 10.4) software (Molecular Devices) to generate\\n                                         current injection waveforms and to record voltage and current traces.\\n\\n\\n\\n                                         Photocurrents were recorded from cells in voltage clamp held at −60 mV. Neuronal firing\\n                                         was measured in current clamp mode with current injection for a −60 mV holding potential.\\n                                         Access resistance (Ra) and membrane resistance (Rm) were monitored throughout recording,\\n                                         and cells were discarded if Ra or Rm changed more than 15%. During ChR variant\\n                                         functional screening in HEK cells, photocurrents were only recorded from cells that passed\\n                                         our recording criteria: Rm > 200 MΩ and holding current >−100 pA. Our measured\\n                                         membrane properties of ChR expressing neurons were consistent with previous literature of\\nAuthor Manuscript                        opsin-expressing cells11 and are also consistent with previous reports of properties of\\n                                         cultured hippocampal neurons38,39 and PFC neurons in slice40,41 (Supplemental Figures 10).\\n                                         For cell culture experiments, the experimenter was blinded to the identity of the ChR being\\n                                         patched but not to the fluorescence level of the cells. For acute slice recordings, the\\n                                         experimenter was not blinded to the identity of the ChR.\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'NO_CONTENT_HERE',\n",
       "  'images': [{'name': 'page_11.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_11.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 11}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'NO_CONTENT_HERE',\n",
       "    'md': 'NO_CONTENT_HERE',\n",
       "    'bBox': {'x': 0, 'y': 0, 'w': 612}}],\n",
       "  'md_revised': 'NO_CONTENT_HERE'},\n",
       " {'page': 12,\n",
       "  'text': '                  Bedbrook et al.                                                                                                      Page 12\\n\\n\\n\\nAuthor ManuscriptLight delivery and imaging\\n                                         Patch-clamp recordings were done with short light pulses to measure photocurrents. Light\\n                                         pulse duration, wavelength, and power were varied depending on the experiment (as\\n                                         described in the text). Light pulses were generated using a Lumencor SPECTRAX light\\n                                         engine. The illumination/output spectra for each color were measured (Supplemental Figure\\n                                         5). To evaluate normalized green photocurrent, we measured photocurrent strength at three\\n                                         wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm,\\n                                         and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these\\n                                         measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640\\n                                         nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we\\n                                         measured photocurrents at seven different wavelengths (peak ± half width half maximum):\\n                                         (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481\\nAuthor Manuscript                        ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for\\n                                         each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n                                         Imaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5\\n                                         sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR\\n                                         expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880\\n                                         confocal microscope and Zen software.\\n\\n\\n\\n                      Electrophysiology data analysis\\n                                         Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and\\n                                         custom data-processing scripts written using open-source packages in the Python\\n                                         programming language to perform baseline adjustments, find the peak and steady state\\nAuthor Manuscript                        inward currents, perform monoexponential fits of photocurrent decay for off-kinetic\\n                                         properties, and quantify spike fidelity. Only neurons with an uncompensated series\\n                                         resistance between 5 and 25 MΩ,Rm > 90 MΩ, and holding current >−150 pA (holding at\\n                                         −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent\\n                                         amplitude was not adjusted for expression level since both expression and conductance\\n                                         contribute to thein vivo utility of the tool. However, comparisons of expression with\\n                                         photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7.\\n                                         As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure\\n                                         1a). As a metric for the ChR activation spectrum, the normalized current strength induced by\\n                                         exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to\\n                                         characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the\\n                                         photocurrent decay rate, τoff (Figure 1a).\\nAuthor ManuscriptAAV production and purification\\n                                         Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE,\\n                                         pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X =\\n                                         ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods\\n                                         described in Devermanet al.42 and Challiset al.31. Briefly, triple transfection of HEK293T\\n                                         cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested\\n                                         from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556)\\n                                         step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': '## Light delivery and imaging\\n\\nPatch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\\n\\n## Electrophysiology data analysis\\n\\nElectrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\\n\\n## AAV production and purification\\n\\nProduction of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "  'images': [{'name': 'page_12.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_12.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 12}],\n",
       "  'items': [{'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Light delivery and imaging',\n",
       "    'md': '## Light delivery and imaging',\n",
       "    'bBox': {'x': 90, 'y': 71, 'w': 126.70000000000002, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.',\n",
       "    'md': 'Patch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.',\n",
       "    'bBox': {'x': 149, 'y': 89, 'w': 369.5999999999999, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Electrophysiology data analysis',\n",
       "    'md': '## Electrophysiology data analysis',\n",
       "    'bBox': {'x': 89, 'y': 349, 'w': 151.73, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).',\n",
       "    'md': 'Electrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).',\n",
       "    'bBox': {'x': 149, 'y': 367, 'w': 371.0999999999998, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'AAV production and purification',\n",
       "    'md': '## AAV production and purification',\n",
       "    'bBox': {'x': 89, 'y': 589, 'w': 152.90000000000003, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "    'md': 'Production of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in',\n",
       "    'bBox': {'x': 149, 'y': 607, 'w': 369.63999999999976, 'h': 10}}],\n",
       "  'md_revised': '# Light delivery and imaging\\n\\nPatch-clamp recordings were done with short light pulses to measure photocurrents. Light pulse duration, wavelength, and power were varied depending on the experiment (as described in the text). Light pulses were generated using a Lumencor SPECTRAX light engine. The illumination/output spectra for each color were measured (Supplemental Figure 5). To evaluate normalized green photocurrent, we measured photocurrent strength at three wavelengths (peak ± half width at half maximum): (red) 640 ± 3 nm, (green) 546 ± 16 nm, and (cyan) 481 ± 3 nm with a 0.5 s light pulse. Light intensity was matched for these measurements, with 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. For full spectra measurements depicted in Figure 2e, we measured photocurrents at seven different wavelengths (peak ± half width half maximum): (red) 640 ± 3 nm, (yellow) 567 ± 13 nm, (green) 546 ± 16 nm, (teal) 523 ± 6 nm, (cyan) 481 ± 3 nm LED, (blue) 439 ± 8 nm LED, and (violet) 397 ± 3 nm with a 0.5 s light pulse for each color. Light intensity is matched across wavelengths at 1.3 mW mm−2.\\n\\nImaging of ChR variants expression in HEK cells was performed using an Andor Neo 5.5 sCMOS camera and Micro-Manager Open Source Microscopy Software. Imaging of ChR expression in neuronal cultures and in brain slices was performed using a Zeiss LSM 880 confocal microscope and Zen software.\\n\\n# Electrophysiology data analysis\\n\\nElectrophysiology data were analyzed using Clampfit 10.7 (Molecular Devices, LLC) and custom data-processing scripts written using open-source packages in the Python programming language to perform baseline adjustments, find the peak and steady state inward currents, perform monoexponential fits of photocurrent decay for off-kinetic properties, and quantify spike fidelity. Only neurons with an uncompensated series resistance between 5 and 25 MΩ, Rm > 90 MΩ, and holding current >−150 pA (holding at −60 mV) were included in data analysis (Supplemental Figures 10). The photocurrent amplitude was not adjusted for expression level since both expression and conductance contribute to the in vivo utility of the tool. However, comparisons of expression with photocurrent strength for all ChR variants tested are included in Supplemental Figures 6–7. As metrics of photocurrent strength, peak and steady-state photocurrent were used (Figure 1a). As a metric for the ChR activation spectrum, the normalized current strength induced by exposure to green light (546 nm) was used (Figure 1a). Two parameters were used to characterize ChR off-kinetics: the time to reach 50% of the light-activated current and the photocurrent decay rate, τoff (Figure 1a).\\n\\n# AAV production and purification\\n\\nProduction of recombinant AAV-PHP.eB packaging pAAV-hSyn-X-TS-eYFP-WPRE, pAAV-CAG-DIO[X-TS-eYFP]-WPRE, and pAAV-CaMKIIa-X-TS-eYFP-WPRE (X = ChR2(H134R), CoChR, ChRger1, ChRger2, and ChRger3) was done following the methods described in Deverman et al.42 and Challis et al.31. Briefly, triple transfection of HEK293T cells (ATCC) was performed using polyethylenimine (PEI). Viral particles were harvested from the media and cells. Virus was then purified over iodixanol (Optiprep, Sigma; D1556) step gradients (15%, 25%, 40% and 60%). Viruses were concentrated and formulated in'},\n",
       " {'page': 13,\n",
       "  'text': '                  Bedbrook et al.                                                                                                          Page 13\\n\\n\\n\\nAuthor Manuscript                        phosphate buffered saline (PBS). Virus titers were determined by measuring the number of\\n                                         DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\\n\\n\\n\\n                      Animals\\n                                         All procedures were approved by the California Institute of Technology Institutional Animal\\n                                         Care and Use Committee (IACUC).Dat-Cre mice (006660) and C57Bl/6J mice (000664)\\n                                         were purchased from Jackson Laboratory.\\n\\n\\n\\n                      Intravenous injections, stereotactic injections, and cannula implantation\\n                                         Intravenous administration of rAAV vectors was performed by injecting the virus into the\\n                                         retro-orbital sinus at viral titers indicated in the text. There were no observed health issues\\n                                         with animals after systemic injection of virus at the titers presented in the paper. Mice\\nAuthor Manuscript                        remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice\\n                                         electrophysiology, there was no observed indication of poor cell health due to viral-mediated\\n                                         expression, which was quantified by measuring the membrane resistance [Rm], leak current\\n                                         [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local\\n                                         expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl\\n                                         of purified AAV vectors at 5×1012 vg ml−1 targeting the following coordinates: anterior-\\n                                         posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For\\n                                         stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses,\\n                                         MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the\\n                                         VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV,\\n                                         4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm\\n                                         mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were\\nAuthor Manuscript                        surgically secured to the surface of the skull above M2 (unilaterally) targeted to the\\n                                         following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a\\n                                         standard drill to create a level surface for the fiber-skull interface. Light was delivered from\\n                                         either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-\\n                                         LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-\\n                                         ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull\\n                                         with Metabond (Parkel, SKU S396) and dental cement.\\n\\n\\n\\n                                         Analysis of behavioral experiments was performed using the open-source MATLAB\\n                                         program OptiMouse43 to track mouse nose, body, and tail position while the mouse was\\n                                         running on the treadmill. Optogenetic intracranial self-stimulation was performed using a\\n                                         mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR\\n                                         nose port (Model 80116TM).\\nAuthor Manuscript\\n                      Gaussian process modeling\\n                                         Both the GP regression and classification modeling methods applied in this paper are based\\n                                         on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple\\n                                         Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/\\n                                         muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2\\n                                         sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high\\n                                         sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use\\n\\n\\n\\n                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\\n\\n## Animals\\n\\nAll procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\\n\\n## Intravenous injections, stereotactic injections, and cannula implantation\\n\\nIntravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\\n\\n## Gaussian process modeling\\n\\nBoth the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "  'images': [{'name': 'page_13.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_13.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 13}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.',\n",
       "    'md': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.',\n",
       "    'bBox': {'x': 150, 'y': 71, 'w': 368.00999999999976, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Animals',\n",
       "    'md': '## Animals',\n",
       "    'bBox': {'x': 90, 'y': 111, 'w': 38.900000000000006, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.',\n",
       "    'md': 'All procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.',\n",
       "    'bBox': {'x': 150, 'y': 129, 'w': 369.77999999999975, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Intravenous injections, stereotactic injections, and cannula implantation',\n",
       "    'md': '## Intravenous injections, stereotactic injections, and cannula implantation',\n",
       "    'bBox': {'x': 90, 'y': 183, 'w': 343.18, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).',\n",
       "    'md': 'Intravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).',\n",
       "    'bBox': {'x': 90,\n",
       "     'y': 111,\n",
       "     'w': 371.3499999999996,\n",
       "     'h': 10.514623181075024}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Gaussian process modeling',\n",
       "    'md': '## Gaussian process modeling',\n",
       "    'bBox': {'x': 89, 'y': 601, 'w': 133.18000000000006, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "    'md': 'Both the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use',\n",
       "    'bBox': {'x': 149, 'y': 619, 'w': 370.3499999999998, 'h': 10}}],\n",
       "  'md_revised': 'phosphate buffered saline (PBS). Virus titers were determined by measuring the number of DNase I–resistant viral genomes using qPCR with linearized genome plasmid as a standard.\\n\\n# Animals\\n\\nAll procedures were approved by the California Institute of Technology Institutional Animal Care and Use Committee (IACUC). Dat-Cre mice (006660) and C57Bl/6J mice (000664) were purchased from Jackson Laboratory.\\n\\n# Intravenous injections, stereotactic injections, and cannula implantation\\n\\nIntravenous administration of rAAV vectors was performed by injecting the virus into the retro-orbital sinus at viral titers indicated in the text. There were no observed health issues with animals after systemic injection of virus at the titers presented in the paper. Mice remain healthy >6 months after systemic delivery of ChR2 and ChRgers. With slice electrophysiology, there was no observed indication of poor cell health due to viral-mediated expression, which was quantified by measuring the membrane resistance [Rm], leak current [holding at −60 mV], and resting membrane potential (Supplemental Figures 10). Local expression in the prefrontal cortex (PFC) was achieved by direct stereotactic injection of 1 μl of purified AAV vectors at 5×10¹² vg ml⁻¹ targeting the following coordinates: anterior-posterior (AP), −1.7; media-lateral (ML), +/− 0.5; and dorsal-ventral (DV), −2.2. For stimulation of the VTA, 300 μm outer diameter mono fiber-optic cannulae (Doric Lenses, MFC_300/330–0.37_6mm_ZF1.25_FLT) were stereotaxically implanted 200 μm above the VTA bilaterally targeted to the following coordinates: AP, −3.44 mm; ML, +/−0.48 mm; DV, 4.4 mm. For stimulation of the right secondary motor cortex (M2), 3 mm long, 400 μm mono fiber-optic cannulae (Doric Lenses, MFC_400/430–0.48_3mm_ZF1.25_FLT) were surgically secured to the surface of the skull above M2 (unilaterally) targeted to the following coordinates: AP, 1 mm; ML, 0.5 mm. The skull was thinned ~40–50% with a standard drill to create a level surface for the fiber-skull interface. Light was delivered from either a 447 nm or 671 nm laser (Changchun New Industries [CNI] Model with PSU-H-LED) via mono fiber-optic patch cable(s) (Doric Lenses, MFP_400/430/1100–0.48_2m_FC-ZF1.25) coupled to the fiber-optic cannula(e). Fiber-optic cannulae were secured to the skull with Metabond (Parkel, SKU S396) and dental cement.\\n\\nAnalysis of behavioral experiments was performed using the open-source MATLAB program OptiMouse⁴³ to track mouse nose, body, and tail position while the mouse was running on the treadmill. Optogenetic intracranial self-stimulation was performed using a mouse modular test chamber (Lafayette Instruments, Model 80015NS) outfitted with an IR nose port (Model 80116TM).\\n\\n# Gaussian process modeling\\n\\nBoth the GP regression and classification modeling methods applied in this paper are based on work detailed in ref 8 and 23. For modeling, all sequences were aligned using MUltiple Sequence Comparison by Log-Expectation (MUSCLE) (https://www.ebi.ac.uk/Tools/msa/muscle/). For modeling, aligned sequences were truncated to match the length of the C1C2 sequence, eliminating N- and C-terminal fragments with poor alignment quality due to high sequence diversity (Dataset 1 and Dataset 2). Structural encodings (i.e., the contact map) use'},\n",
       " {'page': 14,\n",
       "  'text': '                   Bedbrook et al.                                                                                                     Page 14\\n\\n\\n\\nAuthor Manuscript                   the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact\\n                                    architecture observed in the C1C2 crystal structure. Models built using structural encodings\\n                                    built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb)\\n                                    performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The\\n                                    models are robust to differences in contact maps because they use both sequence and\\n                                    structural information, which is somewhat redundant.\\n\\n\\n\\n                                    For a given ChR, the contact map is simply a list of contacting amino acids with their\\n                                    positions. For example, a contact between alanine at position 134 and methionine at position\\n                                    1 of the amino acid sequence would be encoded by [(‘A134’), (‘M1’)]. Both sequence and\\n                                    structural information were one-hot encoded. Regression models for ChR properties were\\n                                    trained to predict the logarithm of the measured properties. All training data was normalized\\n                                    to have mean zero and standard deviation one.\\nAuthor Manuscript\\n                                    Gaussian process regression and classification models require kernel functions that measure\\n                                    the similarity between protein sequences. Learning involves optimizing the form of the\\n                                    kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be\\n                                    optimal for all ChR properties (Supplemental Table 1).\\n\\n\\n\\n                                    For classification model training, all 102 functionally characterized ChR variants from our\\n                                    recombination libraries (Dataset 2) were used as well as data from 61 sequence variants\\n                                    published by others (Dataset 1). The model was then updated with data collected from the\\n                                    22 additional ChR recombination variants with high sequence diversity (~70 mutations from\\n                                    the closest parent) and predicted to be functional (Figure 1d). For training the regression\\n                                    models, all 102 functionally characterized training sequences (Dataset 2) were initially used\\nAuthor Manuscript                   and then the models were updated with data collected from the 22 additional ChR variants\\n                                    (Figure 1d).\\n\\n\\n\\n                                    GP regression—In regression, the goal is to infer the value of an unknown functionf(x) at\\n                                    a novel point x*given observationsy at inputsX. Assuming that the observations are subject\\n                                    to independent and identically distributed Gaussian noise with variance σn2, the posterior\\n                                    distribution of f * = f x* for Gaussian process regression is Gaussian with mean\\n\\n\\n\\n                                                                                   f * = k*T K + σn2I −1y                                    (1)\\n\\n\\n\\n                                    and variance\\n\\n\\n\\nAuthor Manuscript                                                         v* = k x*, x* − k*T K + σn2I −1k*                                  (2)\\n\\n\\n\\n                                    WhereK is the symmetric, square covariance matrix for the training set:Kij =k(x,x) forxii        j\\n                                    andxj in the training set. k*is the vector of covariances between the novel input and each\\n                                    input in the training set, and k*i = k x*, xi . The hyperparameters in the kernel functions and\\n                                    the noise hyperparameterσn were determined by maximizing the log marginal likelihood:\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\\n\\n## GP regression\\n\\nIn regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\",\n",
       "  'images': [{'name': 'page_14.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_14.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 14}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\",\n",
       "    'md': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\",\n",
       "    'bBox': {'x': 149, 'y': 71, 'w': 369.9499999999997, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'GP regression',\n",
       "    'md': '## GP regression',\n",
       "    'bBox': {'x': 477, 'y': 619, 'w': 2.224, 'h': 8.411698544860009}},\n",
       "   {'type': 'text',\n",
       "    'value': 'In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:',\n",
       "    'md': 'In regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:',\n",
       "    'bBox': {'x': 149, 'y': 479, 'w': 352.27, 'h': 14}}],\n",
       "  'md_revised': \"the C1C2 crystal structure (3UG9.pdb) and assume that ChR chimeras share the contact architecture observed in the C1C2 crystal structure. Models built using structural encodings built from the ChR2 structure (6EID.pdb) and the C1Chrimson structure (5ZIH.pdb) performed as well as models using the C1C2 structure (Supplemental Figure 1c,d). The models are robust to differences in contact maps because they use both sequence and structural information, which is somewhat redundant.\\n\\nFor a given ChR, the contact map is simply a list of contacting amino acids with their positions. For example, a contact between alanine at position 134 and methionine at position 1 of the amino acid sequence would be encoded by [('A134'), ('M1')]. Both sequence and structural information were one-hot encoded. Regression models for ChR properties were trained to predict the logarithm of the measured properties. All training data was normalized to have mean zero and standard deviation one.\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. Learning involves optimizing the form of the kernel and its hyperparameters (Supplemental Table 2). The Matérn kernel was found to be optimal for all ChR properties (Supplemental Table 1).\\n\\nFor classification model training, all 102 functionally characterized ChR variants from our recombination libraries (Dataset 2) were used as well as data from 61 sequence variants published by others (Dataset 1). The model was then updated with data collected from the 22 additional ChR recombination variants with high sequence diversity (~70 mutations from the closest parent) and predicted to be functional (Figure 1d). For training the regression models, all 102 functionally characterized training sequences (Dataset 2) were initially used and then the models were updated with data collected from the 22 additional ChR variants (Figure 1d).\\n\\n# GP regression\\n\\nIn regression, the goal is to infer the value of an unknown function f(x) at a novel point x* given observations y at inputs X. Assuming that the observations are subject to independent and identically distributed Gaussian noise with variance σn2, the posterior distribution of f* = f(x*) for Gaussian process regression is Gaussian with mean\\n\\n$$f* = k*^T(K + σn^2I)^{-1}y$$\\n\\nand variance\\n\\n$$v* = k(x*, x*) - k*^T(K + σn^2I)^{-1}k*$$\\n\\nWhere K is the symmetric, square covariance matrix for the training set: Kij = k(xi, xj) for xi and xj in the training set. k* is the vector of covariances between the novel input and each input in the training set, and k*i = k(x*, xi). The hyperparameters in the kernel functions and the noise hyperparameter σn were determined by maximizing the log marginal likelihood:\"},\n",
       " {'page': 15,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                                                                                                                                                                                                 Page 15\\n\\n\\n\\nAuthor Manuscript                                                                                                                                                                             1yT K + σn2I −1y − 2 1log K + σn2I − 2 nlog 2π\\n                                                                                                                    log p y|X = − 2                                                                                                                                                                                                         (3)\\n                                      where nis the dimensionality of the inputs. Regression was implemented using open-source\\n                                      packages in the SciPy ecosystem44–46.\\n\\n\\n\\n                                    GP classification—In binary classification, instead of continuous outputsy, the outputs\\n                                    are class labelsyi ∈ {+1, −1}, and the goal is to use the training data to make probabilistic\\n                                      predictions π x* = p y* = + 1 x* . We use Laplace’s method to approximate the posterior\\n                                    distribution. Hyperparameters in the kernels are found by maximizing the marginal\\n                                    likelihood. Classification was implemented using open-source packages in the SciPy\\n                                    ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is\\nAuthor Manuscript                   or is not functional. A ChR sequence was considered to be functional if its photocurrents\\n                                    were >100 pA upon light exposure, a threshold set as an approximate lower bound for\\n                                    current necessary for neuronal activation.\\n\\n\\n\\n                                    GP kernels for modeling proteins—Gaussian process regression and classification\\n                                    models require kernel functions that measure the similarity between protein sequences. A\\n                                    protein sequences of lengthL is defined by the amino acid present at each location. This can\\n                                    be encoded as a binary feature vector                                                                                                                                                xse that indicates the presence or absence of each\\n                                      amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids).\\n                                      Likewise, the protein’s structure can be represented as a residue-residue contact map. The\\n                                      contact map can be encoded as a binary feature vectorxst that indicates the presence or\\n                                    absence of each possible contacting pair. Both the sequence and structure feature vectors\\nAuthor Manuscript                   were used by concatenating them to form a sequence-structure feature vector.\\n                                    Three types of kernel functions                                                                                                                        k(s,s) were considered: polynomial kernels, squaredij\\n                                    exponential kernels, and Matérn kernels. These different forms represent possible functions\\n                                    for the protein’s fitness landscape. The polynomial kernel is defined as:\\n\\n\\n\\n                                                                                                                                                                                                   k s, s′ =                                                       σ02 + σp2 xT x′ d                                                        (4)\\n\\n\\n\\n                                      whereσ0 andσp are hyperparameters. We considered polynomial kernels with                                                                                                                                                                                                         d = 3. The\\n                                      squared exponential kernel is defined as:\\n\\n\\n\\n                                                                                                                                                                                                                                              σ2 exp − x − x′ 22\\n                                                                                                                                                                                    k s, s′ =                                                         p                                    l                                                (5)\\nAuthor Manuscript\\n                                      wherel and                                                   σp are also hyperparameters and |⋅|2 is the L2 norm. Finally, the Matérn kernel\\n                                      with v=                                     5is defined as:\\n                                                                                  2\\n\\n\\n\\n                                                                                                                                                                                         5 x − x′ 2+ 5 x − x′ 22                                                                              2  exp − 5 x − x′ 22\\n                                                                                                    k s, s′ =                                                       1 +                                                                                                               3l2                                                   (6)\\n                                                                                                                                                                                                                l                                                                                                   l\\n\\n\\n\\n                                                                Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\\n\\n## GP classification\\n\\nIn binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\\n\\n## GP kernels for modeling proteins\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "  'images': [{'name': 'page_15.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_15.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 15}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': '$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.',\n",
       "    'md': '$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.',\n",
       "    'bBox': {'x': 150, 'y': 128, 'w': 153.5, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'GP classification',\n",
       "    'md': '## GP classification',\n",
       "    'bBox': {'x': 289, 'y': 438, 'w': 4.032, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\",\n",
       "    'md': \"In binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\",\n",
       "    'bBox': {'x': 149, 'y': 128, 'w': 366.64999999999986, 'h': 14}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'GP kernels for modeling proteins',\n",
       "    'md': '## GP kernels for modeling proteins',\n",
       "    'bBox': {'x': 289, 'y': 438, 'w': 4.032, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "    'md': \"Gaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\",\n",
       "    'bBox': {'x': 149,\n",
       "     'y': 314,\n",
       "     'w': 369.75998000000004,\n",
       "     'h': 10.514623181075024}}],\n",
       "  'md_revised': \"$$\\\\log p(y|X) = -\\\\frac{1}{2}y^T(K + \\\\sigma_n^2I)^{-1}y - \\\\frac{1}{2}\\\\log |K + \\\\sigma_n^2I| - \\\\frac{n}{2}\\\\log 2\\\\pi \\\\tag{3}$$\\n\\nwhere n is the dimensionality of the inputs. Regression was implemented using open-source packages in the SciPy ecosystem44–46.\\n\\n# GP classification\\n\\nIn binary classification, instead of continuous outputs y, the outputs are class labels $y_i \\\\in \\\\{+1, -1\\\\}$, and the goal is to use the training data to make probabilistic predictions $\\\\pi(x_*) = p(y_* = +1|x_*)$. We use Laplace's method to approximate the posterior distribution. Hyperparameters in the kernels are found by maximizing the marginal likelihood. Classification was implemented using open-source packages in the SciPy ecosystem44–46. The binary classification model was trained to predict if a ChR sequence is or is not functional. A ChR sequence was considered to be functional if its photocurrents were >100 pA upon light exposure, a threshold set as an approximate lower bound for current necessary for neuronal activation.\\n\\n# GP kernels for modeling proteins\\n\\nGaussian process regression and classification models require kernel functions that measure the similarity between protein sequences. A protein sequence s of length L is defined by the amino acid present at each location. This can be encoded as a binary feature vector $x_{se}$ that indicates the presence or absence of each amino acid at each position resulting in a vector of length 20L (for 20 possible amino acids). Likewise, the protein's structure can be represented as a residue-residue contact map. The contact map can be encoded as a binary feature vector $x_{st}$ that indicates the presence or absence of each possible contacting pair. Both the sequence and structure feature vectors were used by concatenating them to form a sequence-structure feature vector.\\n\\nThree types of kernel functions $k(s_i, s_j)$ were considered: polynomial kernels, squared exponential kernels, and Matérn kernels. These different forms represent possible functions for the protein's fitness landscape. The polynomial kernel is defined as:\\n\\n$$k(s, s') = (\\\\sigma_0^2 + \\\\sigma_p^2x^Tx')^d \\\\tag{4}$$\\n\\nwhere $\\\\sigma_0$ and $\\\\sigma_p$ are hyperparameters. We considered polynomial kernels with d = 3. The squared exponential kernel is defined as:\\n\\n$$k(s, s') = \\\\sigma_p^2\\\\exp\\\\left(-\\\\frac{\\\\|x - x'\\\\|_2^2}{l}\\\\right) \\\\tag{5}$$\\n\\nwhere l and $\\\\sigma_p$ are also hyperparameters and $|\\\\cdot|_2$ is the L2 norm. Finally, the Matérn kernel with $v = \\\\frac{5}{2}$ is defined as:\\n\\n$$k(s, s') = \\\\left(1 + \\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l} + \\\\frac{5\\\\|x - x'\\\\|_2^2}{3l^2}\\\\right)\\\\exp\\\\left(-\\\\frac{\\\\sqrt{5}\\\\|x - x'\\\\|_2}{l}\\\\right) \\\\tag{6}$$\"},\n",
       " {'page': 16,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                              Page 16\\n\\n\\n\\nAuthor Manuscript                          Wherel is once again a hyperparameter.\\n                                           L1 regression feature identification and weighting—L1 regression was used to\\n                                           identify residues and contacts in the ChR structure most important for each ChR functional\\n                                           property of interest. First, residues and contacts that covary were identified using the\\n                                           concatenated sequence and structure binary feature vector for each of the training set ChR\\n                                           variants. Each set of covarying residues and contacts was combined into a single feature. L1\\n                                           linear regression was used to select the features that contribute most to each ChR functional\\n                                           property of interest. The level of regularization was chosen by maximizing the log marginal\\n                                           likelihood of the Gaussian process regression model trained on the features selected at that\\n                                           level of regularization. We then performed Bayesian ridge regression on the selected features\\n                                           using the default settings in scikit-learn47. Residues and contacts with the largest absolute\\nAuthor Manuscript                          Bayesian ridge linear regression weights were plotted onto the C1C2 structure\\n                                           (Supplemental Figure 4). For feature identification and weighting, models were trained on\\n                                           both the training set and also the test set of 28 ChR sequences predicted to have useful\\n                                           combinations of diverse properties.\\n\\n\\n\\n                       Statistical analysis\\n                                           Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01.\\n                                           For statistical comparisons, we first performed a D’Agostino & Pearson normality test. If the\\n                                           p-value of a D’Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-\\n                                           Wallis test with Dunn’s multiple comparisonspost hoc test was used. If the data passed the\\n                                           normality test, a one-way ANOVA was used.\\n\\n\\n\\n                       Data availability\\nAuthor Manuscript                          The authors declare that data supporting the findings of this study are available within the\\n                                           paper and its supplementary information files. Source data for classification model training\\n                                           are provided in Dataset 1 and Dataset 2. Source data for regression model training are\\n                                           provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution\\n                                           at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\\n\\n\\n\\n                       Code availability\\n                                           Code used to train classification and regression models can be found at: https://github.com/\\n                                           fhalab/channels.\\n\\n\\n\\n                       Supplementary Material\\nAuthor Manuscript                          Refer to Web version on PubMed Central for supplementary material.\\n\\n\\n\\n                       Acknowledgements\\n                                           We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech\\n                                           Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer\\n                                           measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for\\n                                           helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative\\n                                           Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN\\n                                           R01MH117069, NIH Director’s Pioneer Award DP1OD025535, NIH Director’s New Innovator Award\\n                                           DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"Where l is once again a hyperparameter.\\n\\n## L1 regression feature identification and weighting\\n\\nL1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\\n\\n## Statistical analysis\\n\\nPlotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\\n\\n## Data availability\\n\\nThe authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\\n\\n## Code availability\\n\\nCode used to train classification and regression models can be found at: https://github.com/fhalab/channels.\\n\\n## Supplementary Material\\n\\nRefer to Web version on PubMed Central for supplementary material.\\n\\n## Acknowledgements\\n\\nWe thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "  'images': [{'name': 'page_16.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_16.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 16}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'Where l is once again a hyperparameter.',\n",
       "    'md': 'Where l is once again a hyperparameter.',\n",
       "    'bBox': {'x': 0, 'y': 0, 'w': 612}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'L1 regression feature identification and weighting',\n",
       "    'md': '## L1 regression feature identification and weighting',\n",
       "    'bBox': {'x': 0, 'y': 0, 'w': 612}},\n",
       "   {'type': 'text',\n",
       "    'value': 'L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.',\n",
       "    'md': 'L1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.',\n",
       "    'bBox': {'x': 149, 'y': 111, 'w': 371.4499999999998, 'h': 13}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Statistical analysis',\n",
       "    'md': '## Statistical analysis',\n",
       "    'bBox': {'x': 90, 'y': 305, 'w': 89.33999999999997, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': \"Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\",\n",
       "    'md': \"Plotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\",\n",
       "    'bBox': {'x': 90, 'y': 305, 'w': 363.16999999999996, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Data availability',\n",
       "    'md': '## Data availability',\n",
       "    'bBox': {'x': 90, 'y': 405, 'w': 75.24000000000001, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).',\n",
       "    'md': 'The authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).',\n",
       "    'bBox': {'x': 150, 'y': 423, 'w': 369.5499999999997, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Code availability',\n",
       "    'md': '## Code availability',\n",
       "    'bBox': {'x': 90, 'y': 505, 'w': 78.57000000000002, 'h': 10}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.',\n",
       "    'md': 'Code used to train classification and regression models can be found at: https://github.com/fhalab/channels.',\n",
       "    'bBox': {'x': 150, 'y': 523, 'w': 364.45000000000005, 'h': 10}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Supplementary Material',\n",
       "    'md': '## Supplementary Material',\n",
       "    'bBox': {'x': 90, 'y': 567, 'w': 135.49200000000002, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Refer to Web version on PubMed Central for supplementary material.',\n",
       "    'md': 'Refer to Web version on PubMed Central for supplementary material.',\n",
       "    'bBox': {'x': 90, 'y': 567, 'w': 251.10000000000025, 'h': 12}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Acknowledgements',\n",
       "    'md': '## Acknowledgements',\n",
       "    'bBox': {'x': 90, 'y': 618, 'w': 112.392, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': \"We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "    'md': \"We thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\",\n",
       "    'bBox': {'x': 149, 'y': 636, 'w': 368.2239999999998, 'h': 8}}],\n",
       "  'md_revised': \"Where l is once again a hyperparameter.\\n\\n# L1 regression feature identification and weighting\\n\\nL1 regression was used to identify residues and contacts in the ChR structure most important for each ChR functional property of interest. First, residues and contacts that covary were identified using the concatenated sequence and structure binary feature vector for each of the training set ChR variants. Each set of covarying residues and contacts was combined into a single feature. L1 linear regression was used to select the features that contribute most to each ChR functional property of interest. The level of regularization was chosen by maximizing the log marginal likelihood of the Gaussian process regression model trained on the features selected at that level of regularization. We then performed Bayesian ridge regression on the selected features using the default settings in scikit-learn47. Residues and contacts with the largest absolute Bayesian ridge linear regression weights were plotted onto the C1C2 structure (Supplemental Figure 4). For feature identification and weighting, models were trained on both the training set and also the test set of 28 ChR sequences predicted to have useful combinations of diverse properties.\\n\\n# Statistical analysis\\n\\nPlotting and statistical analysis were done in Python 2.7 and 3.6 and GraphPad Prism 7.01. For statistical comparisons, we first performed a D'Agostino & Pearson normality test. If the p-value of a D'Agostino & Pearson normality test was < 0.05, the non-parametric Kruskal-Wallis test with Dunn's multiple comparisons post hoc test was used. If the data passed the normality test, a one-way ANOVA was used.\\n\\n# Data availability\\n\\nThe authors declare that data supporting the findings of this study are available within the paper and its supplementary information files. Source data for classification model training are provided in Dataset 1 and Dataset 2. Source data for regression model training are provided in Dataset 2. DNA constructs for the ChRger variants are deposited for distribution at Addgene (http://www.addgene.org, plasmid numbers 127237–44).\\n\\n# Code availability\\n\\nCode used to train classification and regression models can be found at: https://github.com/fhalab/channels.\\n\\n# Supplementary Material\\n\\nRefer to Web version on PubMed Central for supplementary material.\\n\\n# Acknowledgements\\n\\nWe thank Twist Bioscience for synthesizing and cloning ChR sequences, Dr. Daniel Wagenaar and the Caltech Neurotechnology Center for building the mouse treadmill, Joshua Brake for performing spectrometer measurements, Dr. John Bedbrook for critical reading of the manuscript, and the Gradinaru and Arnold labs for helpful discussions. This work was funded by the National Institute of Health (V.G.), the Institute for Collaborative Biotechnologies grant W911NF-09–0001 from the U.S. Army Research Office (F.H.A), NIH BRAIN R01MH117069, NIH Director's Pioneer Award DP1OD025535, NIH Director's New Innovator Award DP2NS087949, and SPARC OT2OD023848. Additional funding includes the NSF NeuroNex Technology Hub\"},\n",
       " {'page': 17,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                   Page 17\\n\\n\\n\\n                                    grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the\\nAuthor Manuscript                   Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector\\n                                    Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content\\n                                    of the information does not necessarily reflect the position or the policy of the Government, and no official\\n                                    endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards\\n                                    F31MH102913. J.E.R. is supported by the Children’s Tumor Foundation (Young Investigator Award 2016-01-006).\\n\\n\\n\\n                        References\\n                                    1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/\\n                                       science.aan5544 (2017).\\n                                    2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems.\\n                                       Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n                                    3. Lin JY A user’s guide to channelrhodopsin variants: features, limitations and future developments.\\n                                       Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed:\\n                                       20621963]\\nAuthor Manuscript                   4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain\\n                                       structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n                                    5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics.\\n                                       Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n                                    6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of\\n                                       microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed:\\n                                       22179551]\\n                                    7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature\\n                                       neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n                                    8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral\\n                                       membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane\\n                                       localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017).\\n                                       [PubMed: 29059183]\\n                                    9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric\\nAuthor Manuscript                      channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of\\n                                       America (2017).\\n                                    10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature\\n                                         reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n                                    11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods\\n                                         11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n                                    12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion\\n                                         channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi:\\n                                         10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n                                    13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of\\n                                         channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–\\n                                         1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n                                    14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches.\\n                                         Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n                                    15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin\\nAuthor Manuscript                        variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/\\n                                         j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n                                    16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion\\n                                         selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016).\\n                                         [PubMed: 26699459]\\n                                    17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482,\\n                                         369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n                                    18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344,\\n                                         409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\\n\\n## References\\n\\n1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "  'images': [{'name': 'page_17.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_17.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 17}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\",\n",
       "    'md': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\",\n",
       "    'bBox': {'x': 150, 'y': 68, 'w': 367.84000000000003, 'h': 8}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'References',\n",
       "    'md': '## References',\n",
       "    'bBox': {'x': 90, 'y': 144, 'w': 64.572, 'h': 12}},\n",
       "   {'type': 'text',\n",
       "    'value': \"1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "    'md': \"1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\",\n",
       "    'bBox': {'x': 150, 'y': 162, 'w': 371.8170000000001, 'h': 9}}],\n",
       "  'md_revised': \"grant 1707316 (V.G.), the CZI Neurodegeneration Challenge Network (V.G.), the Vallee Foundation (V.G.), the Heritage Medical Research Institute (V.G.), and the Beckman Institute for CLARITY, Optogenetics and Vector Engineering Research for technology development and broad dissemination: clover.caltech.edu (V.G.). The content of the information does not necessarily reflect the position or the policy of the Government, and no official endorsement should be inferred. C.N.B. is funded by Ruth L. Kirschstein National Research Service Awards F31MH102913. J.E.R. is supported by the Children's Tumor Foundation (Young Investigator Award 2016-01-006).\\n\\n# References\\n\\n1. Deisseroth K & Hegemann P The form and function of channelrhodopsin. Science 357, doi:10.1126/science.aan5544 (2017).\\n\\n2. Yizhar O, Fenno LE, Davidson TJ, Mogri M & Deisseroth K Optogenetics in neural systems. Neuron 71, 9–34, doi:10.1016/j.neuron.2011.06.004 (2011). [PubMed: 21745635]\\n\\n3. Lin JY A user's guide to channelrhodopsin variants: features, limitations and future developments. Experimental physiology 96, 19–25, doi:10.1113/expphysiol.2009.051961 (2011). [PubMed: 20621963]\\n\\n4. Zhang F et al. Optogenetic interrogation of neural circuits: technology for probing mammalian brain structures. Nature protocols 5, 439–456, doi:10.1038/nprot.2009.226 (2010). [PubMed: 20203662]\\n\\n5. Gradinaru V et al. Molecular and cellular approaches for diversifying and extending optogenetics. Cell 141, 154–165, doi:10.1016/j.cell.2010.02.037 (2010). [PubMed: 20303157]\\n\\n6. Mattis J et al. Principles for applying optogenetic tools derived from direct comparative analysis of microbial opsins. Nature methods 9, 159–172, doi:10.1038/nmeth.1808 (2011). [PubMed: 22179551]\\n\\n7. Chuong AS et al. Noninvasive optical inhibition with a red-shifted microbial rhodopsin. Nature neuroscience 17, 1123–1129, doi:10.1038/nn.3752 (2014). [PubMed: 24997763]\\n\\n8. Bedbrook CN, Yang KK, Rice AJ, Gradinaru V & Arnold FH Machine learning to design integral membrane channelrhodopsins for efficient eukaryotic expression and plasma membrane localization. PLoS computational biology 13, e1005786, doi:10.1371/journal.pcbi.1005786 (2017). [PubMed: 29059183]\\n\\n9. Bedbrook CN et al. Structure-guided SCHEMA recombination generates diverse chimeric channelrhodopsins. Proceedings of the National Academy of Sciences of the United States of America (2017).\\n\\n10. Romero PA & Arnold FH Exploring protein fitness landscapes by directed evolution. Nature reviews. Molecular cell biology 10, 866–876, doi:10.1038/nrm2805 (2009). [PubMed: 19935669]\\n\\n11. Klapoetke NC et al. Independent optical excitation of distinct neural populations. Nature methods 11, 338–346, doi:10.1038/nmeth.2836 (2014). [PubMed: 24509633]\\n\\n12. Govorunova EG, Sineshchekov OA, Janz R, Liu X & Spudich JL Natural light-gated anion channels: A family of microbial rhodopsins for advanced optogenetics. Science 349, 647–650, doi: 10.1126/science.aaa7484 (2015). [PubMed: 26113638]\\n\\n13. Lin JY, Knutsen PM, Muller A, Kleinfeld D & Tsien RY ReaChR: a red-shifted variant of channelrhodopsin enables deep transcranial optogenetic excitation. Nature neuroscience 16, 1499–1508, doi:10.1038/nn.3502 (2013). [PubMed: 23995068]\\n\\n14. Berndt A, Yizhar O, Gunaydin LA, Hegemann P & Deisseroth K Bi-stable neural state switches. Nature neuroscience 12, 229–234, doi:10.1038/nn.2247 (2009). [PubMed: 19079251]\\n\\n15. Lin JY, Lin MZ, Steinbach P & Tsien RY Characterization of engineered channelrhodopsin variants with improved properties and kinetics. Biophysical journal 96, 1803–1814, doi:10.1016/j.bpj.2008.11.034 (2009). [PubMed: 19254539]\\n\\n16. Berndt A et al. Structural foundations of optogenetics: Determinants of channelrhodopsin ion selectivity. Proc Natl Acad Sci U S A 113, 822–829, doi:10.1073/pnas.1523341113 (2016). [PubMed: 26699459]\\n\\n17. Kato HE et al. Crystal structure of the channelrhodopsin light-gated cation channel. Nature 482, 369–374, doi:10.1038/nature10870 (2012). [PubMed: 22266941]\\n\\n18. Wietek J et al. Conversion of channelrhodopsin into a light-gated chloride channel. Science 344, 409–412, doi:10.1126/science.1249375 (2014). [PubMed: 24674867]\"},\n",
       " {'page': 18,\n",
       "  'text': '                   Bedbrook et al.                                                                                                            Page 18\\n\\n\\n\\nAuthor Manuscript                   19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and\\n                                         peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017).\\n                                         [PubMed: 28671695]\\n                                    20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related\\n                                         proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202\\n                                         (2013). [PubMed: 23225662]\\n                                    21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by\\n                                         recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed:\\n                                         12042875]\\n                                    22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered\\n                                         microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed:\\n                                         24952910]\\n                                    23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi:\\n                                         10.1038/nn.2495 (2010). [PubMed: 20081849]\\n                                    24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian\\nAuthor Manuscript                        processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013).\\n                                         [PubMed: 23277561]\\n                                    25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi:\\n                                         10.1126/science.aan8862 (2017).\\n                                    26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature\\n                                         communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n                                    27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of\\n                                         channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/\\n                                         bi901634p (2010). [PubMed: 20000562]\\n                                    28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans\\n                                         triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.\\n                                         2005.11.032 (2005). [PubMed: 16360690]\\n                                    29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated\\n                                         optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\nAuthor Manuscript                   30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and\\n                                         Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-\\n                                         neuro-080317-062048 (2018).\\n                                    31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents.\\n                                         Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n                                    32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron\\n                                         Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.\\n                                         2015.10.017 (2015). [PubMed: 26586182]\\n                                    33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in\\n                                         vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–\\n                                         14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n                                    34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering.\\n                                         Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n                                    35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in\\n                                         mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/\\nAuthor Manuscript                        ncomms5894 (2014).\\n                                    36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-\\n                                         Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.\\n                                         2015.06.020 (2015). [PubMed: 26211362]\\n                                    37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server.\\n                                         Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n                                    38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic\\n                                         Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular\\n                                         neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "  'images': [{'name': 'page_18.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_18.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 18}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "    'md': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]',\n",
       "    'bBox': {'x': 150, 'y': 69, 'w': 371.17800000000034, 'h': 9}}],\n",
       "  'md_revised': '19. Chan KY et al. Engineered AAVs for efficient noninvasive gene delivery to the central and peripheral nervous systems. Nature neuroscience 20, 1172–1179, doi:10.1038/nn.4593 (2017). [PubMed: 28671695]\\n\\n20. Smith MA, Romero PA, Wu T, Brustad EM & Arnold FH Chimeragenesis of distantly-related proteins by noncontiguous recombination. Protein science 22, 231–238, doi:10.1002/pro.2202 (2013). [PubMed: 23225662]\\n\\n21. Voigt CA, Martinez C, Wang ZG, Mayo SL & Arnold FH Protein building blocks preserved by recombination. Nature structural biology 9, 553–558, doi:10.1038/nsb805 (2002). [PubMed: 12042875]\\n\\n22. Hochbaum DR et al. All-optical electrophysiology in mammalian neurons using engineered microbial rhodopsins. Nature methods 11, 825–833, doi:10.1038/nmeth.3000 (2014). [PubMed: 24952910]\\n\\n23. Gunaydin LA et al. Ultrafast optogenetic control. Nature neuroscience 13, 387–392, doi: 10.1038/nn.2495 (2010). [PubMed: 20081849]\\n\\n24. Romero PA, Krause A & Arnold FH Navigating the protein fitness landscape with Gaussian processes. Proc Natl Acad Sci U S A 110, E193–201, doi:10.1073/pnas.1215251110 (2013). [PubMed: 23277561]\\n\\n25. Volkov O et al. Structural insights into ion conduction by channelrhodopsin 2. Science 358, doi: 10.1126/science.aan8862 (2017).\\n\\n26. Oda K et al. Crystal structure of the red light-activated channelrhodopsin Chrimson. Nature communications 9, 3949, doi:10.1038/s41467-018-06421-9 (2018).\\n\\n27. Bamann C, Gueta R, Kleinlogel S, Nagel G & Bamberg E Structural guidance of the photocycle of channelrhodopsin-2 by an interhelical hydrogen bond. Biochemistry 49, 267–278, doi:10.1021/bi901634p (2010). [PubMed: 20000562]\\n\\n28. Nagel G et al. Light activation of channelrhodopsin-2 in excitable cells of Caenorhabditis elegans triggers rapid behavioral responses. Current biology : CB 15, 2279–2284, doi:10.1016/j.cub.2005.11.032 (2005). [PubMed: 16360690]\\n\\n29. Chen S et al. Near-infrared deep brain stimulation via upconversion nanoparticle-mediated optogenetics. Science 359, 679–684, doi:10.1126/science.aaq1144 (2018). [PubMed: 29439241]\\n\\n30. Bedbrook CN, Deverman BE & Gradinaru V Viral Strategies for Targeting the Central and Peripheral Nervous Systems. Annual review of neuroscience 41, 323–348, doi:10.1146/annurev-neuro-080317-062048 (2018).\\n\\n31. Challis RC et al. Systemic AAV vectors for widespread and targeted gene delivery in rodents. Nature protocols, doi:10.1038/s41596-018-0097-3 (2019).\\n\\n32. Pascoli V, Terrier J, Hiver A & Luscher C Sufficiency of Mesolimbic Dopamine Neuron Stimulation for the Progression to Addiction. Neuron 88, 1054–1066, doi:10.1016/j.neuron.2015.10.017 (2015). [PubMed: 26586182]\\n\\n33. Gradinaru V et al. Targeting and readout strategies for fast optical neural control in vitro and in vivo. The Journal of neuroscience : the official journal of the Society for Neuroscience 27, 14231–14238, doi:10.1523/JNEUROSCI.3578-07.2007 (2007). [PubMed: 18160630]\\n\\n34. Yang KK, Wu Z & Arnold FH Machine-learning-guided directed evolution for protein engineering. Nature methods, doi:10.1038/s41592-019-0496-6 (2019).\\n\\n35. Flytzanis NC et al. Archaerhodopsin variants with enhanced voltage-sensitive fluorescence in mammalian and Caenorhabditis elegans neurons. Nature communications 5, 4894, doi:10.1038/ncomms5894 (2014).\\n\\n36. Bedbrook CN et al. Genetically Encoded Spy Peptide Fusion System to Detect Plasma Membrane-Localized Proteins In Vivo. Chemistry & biology 22, 1108–1121, doi:10.1016/j.chembiol.2015.06.020 (2015). [PubMed: 26211362]\\n\\n37. Robert X & Gouet P Deciphering key features in protein structures with the new ENDscript server. Nucleic acids research 42, W320–324, doi:10.1093/nar/gku316 (2014). [PubMed: 24753421]\\n\\n38. Fan J et al. Reduced Hyperpolarization-Activated Current Contributes to Enhanced Intrinsic Excitability in Cultured Hippocampal Neurons from PrP(−/−) Mice. Frontiers in cellular neuroscience 10, 74, doi:10.3389/fncel.2016.00074 (2016). [PubMed: 27047338]'},\n",
       " {'page': 19,\n",
       "  'text': '                   Bedbrook et al.                                                                                                          Page 19\\n\\n\\n\\nAuthor Manuscript                   39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in\\n                                         hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n                                    40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development\\n                                         of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9,\\n                                         5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n                                    41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal\\n                                         neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/\\n                                         bht278 (2015). [PubMed: 24108807]\\n                                    42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to\\n                                         the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n                                    43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis\\n                                         of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017).\\n                                         [PubMed: 28506280]\\n                                    44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical\\n                                         computation. Computing in Science and Engineering 13, 22–30 (2011).\\nAuthor Manuscript                   45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–\\n                                         95 (2007).\\n                                    46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20\\n                                         (2007).\\n                                    47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning\\n                                         Research 12, 2825–2830 (2011).\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "  'images': [{'name': 'page_19.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_19.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 19}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "    'md': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).',\n",
       "    'bBox': {'x': 150, 'y': 69, 'w': 370.071, 'h': 9}}],\n",
       "  'md_revised': '39. Slomowitz E et al. Interplay between population firing stability and single neuron dynamics in hippocampal networks. eLife 4, doi:10.7554/eLife.04378 (2015).\\n\\n40. Kroon T, van Hugte E, van Linge L, Mansvelder HD & Meredith RM Early postnatal development of pyramidal neurons across layers of the mouse medial prefrontal cortex. Scientific reports 9, 5037, doi:10.1038/s41598-019-41661-9 (2019). [PubMed: 30911152]\\n\\n41. van Aerde KI & Feldmeyer D Morphological and physiological characterization of pyramidal neuron subtypes in rat medial prefrontal cortex. Cerebral cortex 25, 788–805, doi:10.1093/cercor/bht278 (2015). [PubMed: 24108807]\\n\\n42. Deverman BE et al. Cre-dependent selection yields AAV variants for widespread gene transfer to the adult brain. Nature biotechnology 34, 204–209, doi:10.1038/nbt.3440 (2016).\\n\\n43. Ben-Shaul Y OptiMouse: a comprehensive open source program for reliable detection and analysis of mouse body and nose positions. BMC biology 15, 41, doi:10.1186/s12915-017-0377-3 (2017). [PubMed: 28506280]\\n\\n44. Walt S, Colbert SC & Varoquaux G The NumPy array: a structure for efficient numerical computation. Computing in Science and Engineering 13, 22–30 (2011).\\n\\n45. Hunter JD Matplotlib: A 2D Graphics Environment. Computing in Science and Engineering 9, 90–95 (2007).\\n\\n46. Oliphant TE Python for Scientific Computing. Computing in Science and Engineering 9, 10–20 (2007).\\n\\n47. Pedregosa F et al. Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research 12, 2825–2830 (2011).'},\n",
       " {'page': 20,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                                                                  Page 20\\n                                       Photocurrent properties                                Irain Gassilicalion mnoclols                                                Irain rogression mocels\\nAuthor Manuscript                            Current sirengh                  Ihreshold out nn-{unclianai ChRs Irom Ihe @brary                      approximate Ihe litness lantscape {ordifferent properties\\n                                                                             H                                               1,161 ChRs            Current strength             Olf-klnetlcs          Spectral properlies\\n                                                      Sleajy                                                                     ahove\\n                                                                                                                               Uhroslold     1\\n                                              \\'crk-kinebcs                                                                     28 top\\n                                                                             L                                                 VatiaIs\\n                                                                                                                             pp > 0.4        1\\n                                          Wavelengih sensilivity                    0,0                            0,8                                                                                  \"25\\n                                                  acilyaticn                                                                                           mea sured                 mea sured                 ICasurod\\n                                                                                      Preclcted probabililty of locallzing\\n                                                                                Traning dala                                 Traning cata\\n                                                                                 (102 variants -                               124 vaiants}\\n                                                                                      Kanons\\n                                    !                                                                 Cla ssilicalion22 varantsvarials                Regrossion                                Chractorizalion\\n                                         400 Wavclengh [nm]500600                                       moclels                                       morlels                                        valiclalion\\n                                                                             Training set                                                                                              Z00 R = 0.92\\nAuthor Manuscript                   2L z0o0                                                                                                                                2           TE00\\n                                                                                                                                                                                       I0o\\n                                          IOOO                                                                                                                                          F00\\n                                                                                                                                                                                                 Tooo Cu\\n                                       2                                                                                       1                                                              R = 0.97\\n                                    9                                                                                                                                      8             20\\n                                       I                                                                                                                                      I     L 075110olR = 0.96\\n                                       {                                                                                        :                                              1       0 501\\n                                       1                                                                                                                                      1\\nAuthor Manuscript                                       parents            single-block             IuIli-block           classilicalion        regression\\n                                                                               suap                    seap                                                                                         Measured\\n                                            mutalions:\\n                                            Figure 1.\\n                                           Machine learning-guided optimization of ChRs. (a) Upon light exposure, ChRs open and\\n                                           reach a        peak inward current and then desensitize reaching a lowersteady-state current. We\\n                                           use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR\\n                                           off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the\\n                                           time to reach 50% of the light-exposed current after light removal. As a metric for\\n                                           wavelength sensitivity of activation, we used the normalized photocurrent with green (546\\n                                           nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and\\n                                           red-shifted ChRs (peak activation: ~520–650 nm). (b) We trained classification models to\\nAuthor Manuscript                          predict whether ChRs would localize correctly to the plasma membrane and function (i.e.,\\n                                           ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of\\n                                           localization and function). (c) We then trained regression models to approximate the fitness\\n                                           landscape for each property of interest for the recombination library (inset show hypothetical\\n                                           fitness landscapes). (b–c) Models were trained with photocurrent properties for each ChR in\\n                                           the training set (plots show 20-fold cross validation on the training set). Sequences predicted\\n                                           to localize and function by the classification models and predicted to have an optimized set\\n                                           of functional properties by the regression models were selected for further characterization,\\n\\n\\n\\n                                                     Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'Figure 1\\n\\n## Figure 1. Machine learning-guided optimization of ChRs.\\n\\n(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "  'images': [{'name': 'img_p19_1.png',\n",
       "    'height': 766,\n",
       "    'width': 873,\n",
       "    'x': 120.13676,\n",
       "    'y': 61.999998000000005,\n",
       "    'original_width': 1800,\n",
       "    'original_height': 1579,\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p19_1.png',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 20},\n",
       "   {'name': 'page_20.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_20.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 20}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'Figure 1',\n",
       "    'md': 'Figure 1',\n",
       "    'bBox': {'x': 140.41257364375716,\n",
       "     'y': 102.04741652741515,\n",
       "     'w': 237.37537924398623,\n",
       "     'h': 32.63122991122715}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Figure 1. Machine learning-guided optimization of ChRs.',\n",
       "    'md': '## Figure 1. Machine learning-guided optimization of ChRs.',\n",
       "    'bBox': {'x': 140.41257364375716,\n",
       "     'y': 89.68710216710183,\n",
       "     'w': 97.4228118980527,\n",
       "     'h': 48.4524322924282}},\n",
       "   {'type': 'text',\n",
       "    'value': '(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "    'md': '(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.\\n\\nThe figure contains several parts:\\n\\na. Photocurrent properties diagram showing current strength, off-kinetics, and wavelength sensitivity of activation.\\n\\nb. Train classification models graph showing the predicted probability of localizing for 1,161 ChRs from the library.\\n\\nc. Train regression models graphs for current strength, off-kinetics, and spectral properties.\\n\\nd. A flowchart showing the process from training data to classification and regression models, and finally to characterization & validation.\\n\\ne. Training set data for peak photocurrent, log10 off-kinetics, and norm. green current across different mutation types (parents, single-block swap, multi-block swap, classification, regression).\\n\\nf. Characterization & validation graphs showing correlations between predicted and measured values for peak photocurrent, log10 off-kinetics, and norm. green current.\\n\\nThe image demonstrates the machine learning approach used to optimize channelrhodopsins (ChRs) by predicting their localization, function, and various properties using classification and regression models trained on experimental data.',\n",
       "    'bBox': {'x': 133.48912508247423,\n",
       "     'y': 67.93294889295039,\n",
       "     'w': 371.8399999999998,\n",
       "     'h': 48.4524322924282}}],\n",
       "  'md_revised': '# Figure 1. Machine learning-guided optimization of ChRs.\\n\\n(a) Upon light exposure, ChRs open and reach a peak inward current and then desensitize reaching a lower steady-state current. We use both peak and steady-state current as metrics for photocurrent strength. To evaluate ChR off-kinetics we used the current decay rate (τoff) after a 1 ms light exposure and also the time to reach 50% of the light-exposed current after light removal. As a metric for wavelength sensitivity of activation, we used the normalized photocurrent with green (546 nm) light, which easily differentiates blue-shifted ChRs (peak activation: ~450–480 nm) and red-shifted ChRs (peak activation: ~520–650 nm).\\n\\n(b) We trained classification models to predict whether ChRs would localize correctly to the plasma membrane and function (i.e., ChRs above the 0.4 threshold for the product of the predicted probabilities (pp) of localization and function).\\n\\n(c) We then trained regression models to approximate the fitness landscape for each property of interest for the recombination library (inset show hypothetical fitness landscapes).\\n\\n(b–c) Models were trained with photocurrent properties for each ChR in the training set (plots show 20-fold cross validation on the training set). Sequences predicted to localize and function by the classification models and predicted to have an optimized set of functional properties by the regression models were selected for further characterization.'},\n",
       " {'page': 21,\n",
       "  'text': '                   Bedbrook et al.                                                                                                Page 21\\n\\n\\n\\nAuthor Manuscript                   e.g., the 28 top variants. (d) The classification model was trained with 102 recombination\\n                                    variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression\\n                                    models were trained with 124 recombination variants. (e) Measurements of training set ChR\\n                                    and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n\\n                                    = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set\\n                                    data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model\\n                                    predictions vs measured photocurrent property for each of the 28 designer ChRs. R\\n                                    represents the Pearson correlation coefficient.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "  'images': [{'name': 'page_21.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_21.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 21}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "    'md': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.',\n",
       "    'bBox': {'x': 149,\n",
       "     'y': 71,\n",
       "     'w': 370.94000000000005,\n",
       "     'h': 10.51462318107501}}],\n",
       "  'md_revised': 'e.g., the 28 top variants. (d) The classification model was trained with 102 recombination variants (Dataset 2) and 61 previously-published ChRs (Dataset 1) and the regression models were trained with 124 recombination variants. (e) Measurements of training set ChR and model-predicted ChR, peak photocurrent, off-kinetics, and normalized green current (n = 3–8 cells per variant; Dataset 2). Each gray-colored point is a ChR variant. Training set data are shaded in blue. Mean number of mutations for each set is below the plots. (f) Model predictions vs measured photocurrent property for each of the 28 designer ChRs. R represents the Pearson correlation coefficient.'},\n",
       " {'page': 22,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                             Page 22\\n                                        CheRiff     CsChrimR       C1C2        11_10                                  10 10        15_10       28_10        21_10\\nAuthor Manuscript\\n                                                                                                                                                                            ZumM\\n                                                              Photocurrent strength with different wavelength excitation                                 Tos (ms]\\n                                                                                                                                                       102   103  104\\n                                    89 000500         Peak\\n                                                      Steady state                                                                                            Parenis\\n                                                                                                                                                                      21_10\\n                                                                                                                                         CsChiimR                        10Q ms\\n                                      3 2000                                                                                               CneRill\\nAuthor Manuscript                   20 1000\\n                                                                                                                                                                      Cheril\\n                                        4000\\n                                    8 2000\\n                                    ?1\\n                                   # 20001000\\n                                                                                3    8e9 83 e9\\n                                     [ 0.75                                               2500                                          2500\\nAuthor Manuscript                                                                      1 2000                                       0 2000\\n                                                                                     2    1500                                       8 1500\\n                                                                                          1000                                      82  1000\\n                                        0.25                                               500                                       2   500\\n                                        0.00\\n                                              400     Wavelength [nm] 600500                      10-1 Intensity [mW mm?]                             Intensity [mW mm?]\\n                                           Figure 2.\\n                                           The model-predicted ChRs exhibit a large range of functional properties often far exceeding\\n                                           the parents. (a) Representative current traces after 0.5 s light exposure for select designer\\n                                           ChR variants with corresponding expression and localization in HEK cells. Vertical colored\\n                                           scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents\\nAuthor Manuscript                          250 ms. The variant color presented in (a) is constant throughout panels. (b) Measured peak\\n                                           and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8\\n                                           cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm\\n                                           light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2. (c) Off-kinetics decay rate (τoff)\\n                                           following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2).\\n                                           Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1\\n                                           ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly,\\n                                           ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in\\n                                           photocurrent 0.5 s after the light exposure. (d) Peak and steady-state photocurrent strength\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'Figure 2\\n\\n## Figure 2.\\nThe model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "  'images': [{'name': 'img_p21_1.png',\n",
       "    'height': 859,\n",
       "    'width': 873,\n",
       "    'x': 120,\n",
       "    'y': 62.000001999999995,\n",
       "    'original_width': 1800,\n",
       "    'original_height': 1770,\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p21_1.png',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 22},\n",
       "   {'name': 'page_22.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_22.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 22}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'Figure 2',\n",
       "    'md': 'Figure 2',\n",
       "    'bBox': {'x': 126.9278350515464,\n",
       "     'y': 240.0302772968568,\n",
       "     'w': 11.876288659793813,\n",
       "     'h': 17.803027529685682}},\n",
       "   {'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Figure 2.',\n",
       "    'md': '## Figure 2.',\n",
       "    'bBox': {'x': 126.9278350515464,\n",
       "     'y': 240.0302772968568,\n",
       "     'w': 34.335,\n",
       "     'h': 17.803027529685682}},\n",
       "   {'type': 'text',\n",
       "    'value': 'The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "    'md': 'The model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength\\n\\nThe figure contains several panels:\\n\\na) A series of current traces and corresponding cell images for different ChR variants.\\n\\nb) Bar graphs showing photocurrent strength at different wavelengths for peak and steady-state measurements.\\n\\nc) A bar graph showing off-kinetics decay rates for different ChR variants, with an inset of representative current traces.\\n\\nd) Line graphs showing normalized photocurrent vs wavelength.\\n\\ne) Line graphs showing peak and steady-state photocurrent vs light intensity.\\n\\nEach panel contains detailed data comparing various ChR variants across different properties and conditions.',\n",
       "    'bBox': {'x': 126.9278350515464,\n",
       "     'y': 69.41793013736903,\n",
       "     'w': 369.28,\n",
       "     'h': 25.715484209545984}}],\n",
       "  'md_revised': '# Figure 2\\n\\nThe model-predicted ChRs exhibit a large range of functional properties often far exceeding the parents.\\n\\n(a) Representative current traces after 0.5 s light exposure for select designer ChR variants with corresponding expression and localization in HEK cells. Vertical colored scale bar for each ChR current trace represents 500 pA, and horizontal scale bar represents 250 ms. The variant color presented in (a) is constant throughout panels.\\n\\n(b) Measured peak and steady-state photocurrent with different wavelengths of light in HEK cells (n = 4–8 cells, see Dataset 2). 397 nm light at 1.5 mW mm−2, 481 nm light at 2.3 mW mm−2, 546 nm light at 2.8 mW mm−2, and 640 nm light at 2.2 mW mm−2.\\n\\n(c) Off-kinetics decay rate (τoff) following a 1 ms exposure to 481 nm light at 2.3 mW mm−2 (n = 4–8 cells, see Dataset 2). Parent ChRs are highlighted in light gray. Inset shows representative current traces with 1 ms light exposure for select ChRs revealing distinct profiles: ChR_21_10 turns off rapidly, ChR_25_9 and ChR_11_10 turn off more slowly, and ChR_15_10 exhibits little decrease in photocurrent 0.5 s after the light exposure.\\n\\n(d) Peak and steady-state photocurrent strength'},\n",
       " {'page': 23,\n",
       "  'text': '                  Bedbrook et al.                                                                                                Page 23\\n\\n\\n\\nAuthor Manuscript                  with varying light irradiances compared with parental ChRs (CheRiff,n = 5 cells;\\n                                   CsChrimR,n = 5 cells; C1C2,n = 4 cells; 28_10,n = 5 cells; 11_10,n = 5 cells; 25_9,n = 5\\n                                   cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs\\n                                   (CheRiff,n = 6 cells; CsChrimR,n = 5 cells; C1C2,n = 4 cells; 11_10,n = 6 cells; 12_10,n\\n                                   = 7 cells; 25_9,n = 5 cells; 10_10,n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and\\n                                   ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data\\n                                   are mean±SEM.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                          Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "  'images': [{'name': 'page_23.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_23.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 23}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "    'md': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.',\n",
       "    'bBox': {'x': 150, 'y': 99, 'w': 369.76999999999975, 'h': 10}}],\n",
       "  'md_revised': 'with varying light irradiances compared with parental ChRs (CheRiff, n = 5 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 28_10, n = 5 cells; 11_10, n = 5 cells; 25_9, n = 5 cells). (e) Wavelength sensitivity of activation for select ChRs compared with parental ChRs (CheRiff, n = 6 cells; CsChrimR, n = 5 cells; C1C2, n = 4 cells; 11_10, n = 6 cells; 12_10, n = 7 cells; 25_9, n = 5 cells; 10_10, n = 4 cells). Top variants, ChR_9_4, ChR_25_9, and ChR_11_10 are named ChRger1, ChRger2, and ChRger3 in subsequent figures. Plotted data are mean±SEM.'},\n",
       " {'page': 24,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                                Page 24\\n                                        Camkil                          [PCPCE                                                                                         Light intensity:\\n                                                ChR     BYFP WPREOGHpa                            peak    steady slale                                               3x102 (Mw mm?]\\nAuthor Manuscript                             88683                                         2500  8x10 J mW mm?                   Chageri\\n                                                                                                                                  ChAgerz\\n                                               CoCnr                                    3 2o0]                              {     ChRger3                            50 mv ChR2\\n                                                ChAz                                     L 1500                                   CoChR\\n                                                                                                                                  Chaz\\n                                        ChR?                 ChRgert                        1000                            8                                             CoChR\\n                                                                                                                                   Light intensity (mw mm]                ChRgeri\\n                                                                                           4ooo\\n                                                                                         1 3000             Mm\\n                                                             IMum                                                                                       ChR2              ChRgerz\\n                                        ChRger?              ChRger3                                                                                 ChRgerz]\\n                                                                                                                                                              ChRgeri\\n                                                                                            2000                            {\\n                                                                                            1000-                                                                         ChRger3\\n                                                                                                                            3\\n                                                                                                 CoChR ChRgerzChRgert ChRgera           Frequency (Hz]\\n                                               Direci intracranial injeclion Systemic delivery (Sx10\" VC)          Systemic delivery (Ixio\" VG)   m        Systemic delivery (Ix1O\" VG)\\nAuthor Manuscript                              Systemlc delivery Sx10\" VG)                                         CnRgerz\\n                                   L 1500 chRgerz\\n                                                 2 2000 ChAgori                                             1 1000 CaCha                             1\\n                                            ChRgor3\\n                                      1000 ChRZ                                                                500                                          ChAigerz\\n                                                                                                                                                                    CoChR\\n                                       500                                                                                                           8      ChAger3\\n                                           10\"Light intensity [mW mm*7]                                              Light intensity [mW mm?]                     Frequency [Hz]\\n                                                                                                                   Systemic delivery (Ix10\" VG)         ChRger1   ChRger2    ChRger3\\n                                    0.5 5 VYV I &r10 \\'                                                      2       ChAgerz\\n                                                             Light intensity[IW mm ?]                               CoChR                           2 Hz\\n                                                                             Systemic delivery (1x10\" VG)   1\\n                                                                                 Light excitable cells (%0)\\n                                                                                                            3                                       5 Hz                           50 mV\\n                                                                Ixio\\'      ChR2\\n                                  1000 PA                                 CoChR                                                                         L\\n                                                                         ChRagi                                                                    20 Hz\\n                                                                 2x0\"    ChAgerz\\n                                                                         ChRger3                                    10 *Light intensity [mW mm ]\\n                                            Figure 3.\\nAuthor Manuscript                          ChRger variants in cultured neurons and in acute brain slices outperform the commonly used\\n                                           ChR2(H134R) and the high performance CoChR. (a) ChRs were cloned into an AAV vector\\n                                           with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression\\n                                           in culture and      in vivo. (b) Cultured neurons expressing ChRgers and ChR2(H134R) under\\n                                           the hSyn promoter (repeated independently six times per construct with similar results). (c)\\n                                           Peak and steady-state photocurrent with low-intensity (8×10−3 mW mm−2) and moderate-\\n                                            intensity (0.8 mW mm−2) light in cultured neurons (ChR2,n = 16 cells; CoChR,n = 17\\n                                            cells; ChRger1,n = 9 cells; ChRger2,n = 24 cells; ChRger3,n = 9 cells). (d) Spike fidelity\\n                                            with varying intensity light for 5 ms light-pulse width at 2 Hz stimulation (ChRger1,n = 6\\n                                            cells; ChRger2,n = 6 cells; ChRger3,n = 6 cells; CoChR,n = 7 cells; ChR2,n = 7 cells). (e)\\n                                            Spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured\\n                                            neurons (ChRger1,n = 9 cells; ChRger2,n = 12 cells; ChRger3,n = 7 cells; ChR2,n = 8\\nAuthor Manuscript                           cells). (f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms\\n                                            pulsed low-intensity blue light stimulation (3×10−2 mW mm−2) shows robust neuronal firing\\n                                            for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced\\n                                            depolarization. (g) Photocurrent strength with varying light irradiances in acute brain slice\\n                                            after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC\\n                                            (ChRger1,n = 11 cells; ChRger2,n = 11 cells; ChRger3,n = 11 cells; ChR2,n = 9 cells) or\\n                                            after systemic delivery of CamKIIa-ChRger2 (ChRger2,n = 6 cells; 5×1011 vg/animal). (h)\\n                                            Representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse at\\n\\n\\n\\n                                                   Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "  'images': [{'name': 'img_p23_1.png',\n",
       "    'height': 646,\n",
       "    'width': 873,\n",
       "    'x': 120.16205,\n",
       "    'y': 61.99999699999995,\n",
       "    'original_width': 1800,\n",
       "    'original_height': 1333,\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p23_1.png',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 24},\n",
       "   {'name': 'page_24.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_24.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 24}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "    'md': 'Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).\\n\\nThe figure caption provides detailed information about the experimental setup, measurements, and results for each panel of the figure.',\n",
       "    'bBox': {'x': 149,\n",
       "     'y': 70.4126283947368,\n",
       "     'w': 371.63999999999976,\n",
       "     'h': 20.289287481424154}}],\n",
       "  'md_revised': '# Figure 3.\\nChRger variants in cultured neurons and in acute brain slices outperform the commonly used ChR2(H134R) and the high performance CoChR.\\n\\n(a) Diagram showing ChRs cloned into an AAV vector with either the hSyn or CamKIIa promoter and packaged into rAAV-PHP.eB for expression in culture and in vivo.\\n\\n(b) Microscope images of cultured neurons expressing ChRgers and ChR2(H134R) under the hSyn promoter. Four images are shown, each depicting neurons with different ChR variants.\\n\\n(c) Two bar graphs showing peak and steady-state photocurrent with low-intensity (8×10^-3 mW mm^-2) and moderate-intensity (0.8 mW mm^-2) light in cultured neurons. Data is shown for ChR2, CoChR, ChRger1, ChRger2, and ChRger3.\\n\\n(d) Line graph showing spike fidelity with varying light intensity for 5 ms light-pulse width at 2 Hz stimulation. Lines are shown for ChRger1, ChRger2, ChRger3, CoChR, and ChR2.\\n\\n(e) Line graph depicting spike fidelity with varying stimulation frequency with 2 ms light-pulse width in cultured neurons. Lines are shown for ChR2, ChRger1, ChRger2, ChRger3, and CoChR.\\n\\n(f) Representative voltage traces of ChRgers and ChR2(H134R) at 2 Hz with 5 ms pulsed low-intensity blue light stimulation (3×10^-2 mW mm^-2). Traces show robust neuronal firing for ChRgers while ChR2(H134R) and CoChR exhibit only sub-threshold light-induced depolarization.\\n\\n(g) Line graph showing photocurrent strength with varying light irradiances in acute brain slice after direct injection of rAAV-PHP.eB packaged hSyn-ChR constructs into the PFC or after systemic delivery of CamKIIa-ChRger2.\\n\\n(h) Diagram and traces showing light intensity and representative current traces of ChRgers and ChR2(H134R) with a 300 ms light pulse.\\n\\n(i) Brain slice images showing systemic delivery (5x10^11 VG).\\n\\n(k) Line graph showing photocurrent for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(l) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying light intensity.\\n\\n(m) Line graph showing spike fidelity for systemic delivery (1x10^11 VG) with varying frequency.\\n\\n(n) Representative voltage traces for ChRger1, ChRger2, and ChRger3 at different stimulation frequencies (2 Hz, 5 Hz, 20 Hz).'},\n",
       " {'page': 25,\n",
       "  'text': '                   Bedbrook et al.                                                                                                    Page 25\\n\\n\\n\\nAuthor Manuscript                   varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of\\n                                    rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression\\n                                    throughout the cortex (5×1011 vg/animal; repeated independently five times per construct\\n                                    with similar results). (j) The fraction of light excitable neurons in the PFC after systemic\\n                                    delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only\\n                                    neurons expressing the eYFP marker (1×1011 vg/animal). Peak (solid line) and steady-state\\n                                    (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in\\n                                    acute brain slice after systemic delivery (1×1011 vg/animal) of hSyn-ChRger2 (n = 13 cells)\\n                                    and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying\\n                                    stimulation frequency in acute brain slice after systemic delivery (1×1011 vg/animal)\\n                                    (CoChR,      n = 15 cells; ChRger1,n = 9 cells; ChRger2,n = 5 cells; ChRger3,n = 8 cells) with\\n                                    1 mW mm−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW\\nAuthor Manuscript                   mm−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\nAuthor Manuscript\\n\\n\\n\\n                                           Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "  'images': [{'name': 'page_25.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_25.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 25}],\n",
       "  'items': [{'type': 'text',\n",
       "    'value': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "    'md': 'varying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.',\n",
       "    'bBox': {'x': 149,\n",
       "     'y': 71,\n",
       "     'w': 370.5099999999998,\n",
       "     'h': 13.514623181075024}}],\n",
       "  'md_revised': '### Figure Caption\\nvarying light irradiances in acute brain slice after direct injection. (i) Systemic delivery of rAAV-PHP.eB packaged hSyn-ChRger2 or hSyn-ChR2(H134R) resulted in broad expression throughout the cortex (5×10^11 vg/animal; repeated independently five times per construct with similar results). (j) The fraction of light excitable neurons in the PFC after systemic delivery of hSyn-ChRs measured by cell-attached recording in acute slice targeting only neurons expressing the eYFP marker (1×10^11 vg/animal). Peak (solid line) and steady-state (dashed line) photocurrent strength (k) and spike fidelity (l) with varying light irradiances in acute brain slice after systemic delivery (1×10^11 vg/animal) of hSyn-ChRger2 (n = 13 cells) and hSyn-CoChR (n = 14 cells) (recorded in PFC neurons). (m) Spike fidelity with varying stimulation frequency in acute brain slice after systemic delivery (1×10^11 vg/animal) (CoChR, n = 15 cells; ChRger1, n = 9 cells; ChRger2, n = 5 cells; ChRger3, n = 8 cells) with 1 mW mm^−2 intensity light. (n) Representative voltage traces with blue light–driven (1 mW mm^−2) spiking at the indicated frequencies. vg, viral genomes. Plotted data are mean±SEM.'},\n",
       " {'page': 26,\n",
       "  'text': '                   Bedbrook et al.                                                                                                                             Page 26\\n                                               Systemic delivery               Fiber implantation       0 Fiber    anti-eYFP                 Intracrania self-stimulation\\nAuthor Manuscript                                      843    WPAE ngHpa                                                               10 Hz stimulation         mW power\\n                                                     ChRgerz                                                      VTA          1 400 ChR2ChRger2\\n                                                          TAAV PHPeB                                   ChRz-TS-eYFP            1 300\\n                                                                                                            Fiber                200\\n                                               Dat-Cre                                                                         2 100\\n                                                                                                        ChRgerz TS- eYFP300 Hm           Power [mW] 10 20  1 3  Simulation\\n                                                                                                                                                               irequency [Hz]\\n                                                Systemic delivery               Fiber on skull surface                     Minimally invasive optical control of locomotion\\n                                       Camkila           #YFP  WPRE hGHpa                                   8                 ChBger?                          ChR?\\n                                               ChAgerz          TAAV-PHPeB                       skull                  671 nm\\n                                                                                                                              447 Dmt               O00 447 nin\\n                                                                                                                                                        671 nm\\n                                                                                                           2 I\\n                                                                                                           8\\nAuthor Manuscript\\n                                                                                                                              Frames                          Frames\\n                                                                                                            ChRgerz\\n                                                                                                           1 Frame                   Frame                 Frame\\n                                                                                                               Frame                 Frame                 Frame\\n                                  anti-eYFP         50 Mum          ChRqerz TS eYFP 50 m                    8\\n                                  ChR2-IS eYFP\\n                                           Figure 4.\\n                                           Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral\\n                                           modulation. (a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP\\n                                           or ChR2(H134R)-TS-eYFP (3×1011 vg/mouse) intoDat-Cre animals coupled with fiber\\n                                           optic implantation above the VTA enabled blue light-induced intracranial self-stimulation\\nAuthor Manuscript                          (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light\\n                                           power and varying stimulation frequencies. ChRger2,n = 4 animals; ChR2(H134R),                                    n = 4\\n                                           animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and\\n                                           ChRger2 (bottom). (b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged\\n                                           CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×1011 vg/mouse) into wild type\\n                                           (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the\\n                                           skull above the right M2 that had been thinned to create a level surface for the fiber-skull\\n                                           interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed\\n                                           velocity. Coronal slices show expression throughout cortex with higher magnification image\\n                                           of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation\\n                                           of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz\\n                                           stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2,                          n = 5 animals;\\nAuthor Manuscript                          ChR2(H134R),n = 5 animals. No turning behavior was observed in any animal with 10 Hz\\n                                           stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral\\n                                           genomes.\\n\\n\\n\\n                                                  Nat Methods. Author manuscript; available in PMC 2020 April 14.',\n",
       "  'md': '## Figure 4.\\n\\nValidation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "  'images': [{'name': 'img_p25_1.png',\n",
       "    'height': 543,\n",
       "    'width': 873,\n",
       "    'x': 120,\n",
       "    'y': 62.00000300000005,\n",
       "    'original_width': 1800,\n",
       "    'original_height': 1118,\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p25_1.png',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 26},\n",
       "   {'name': 'page_26.jpg',\n",
       "    'height': 792,\n",
       "    'width': 612,\n",
       "    'x': 0,\n",
       "    'y': 0,\n",
       "    'type': 'full_page_screenshot',\n",
       "    'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_26.jpg',\n",
       "    'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "    'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "    'page_number': 26}],\n",
       "  'items': [{'type': 'heading',\n",
       "    'lvl': 2,\n",
       "    'value': 'Figure 4.',\n",
       "    'md': '## Figure 4.',\n",
       "    'bBox': {'x': 150, 'y': 351, 'w': 34.335, 'h': 9}},\n",
       "   {'type': 'text',\n",
       "    'value': 'Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "    'md': 'Validation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.',\n",
       "    'bBox': {'x': 121.48453608247422,\n",
       "     'y': 63.4824339779006,\n",
       "     'w': 369.5099999999998,\n",
       "     'h': 27.672044920810322}}],\n",
       "  'md_revised': '# Figure 4.\\n\\nValidation of high-performance ChRger2 for minimally-invasive optogenetic behavioral modulation.\\n\\n(a) Systemic delivery of rAAV-PHP.eB packaged CAG-DIO ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (3×10^11 vg/mouse) into Dat-Cre animals coupled with fiber optic implantation above the VTA enabled blue light-induced intracranial self-stimulation (ten 5 ms laser pulses) exclusively with ChRger2 and not ChR2(H134R) with varying light power and varying stimulation frequencies. ChRger2, n = 4 animals; ChR2(H134R), n = 4 animals. Images show fiber placement and opsin expression for ChR2(H134R) (top) and ChRger2 (bottom).\\n\\n(b) Minimally-invasive, systemic delivery of rAAV-PHP.eB packaged CaMKIIa ChRger2-TS-eYFP or ChR2(H134R)-TS-eYFP (5×10^11 vg/mouse) into wild type (WT) animals coupled with surgically secured fiber-optic cannula guide to the surface of the skull above the right M2 that had been thinned to create a level surface for the fiber-skull interface. Three weeks later, mice were trained to walk on a linear-track treadmill at fixed velocity. Coronal slices show expression throughout cortex with higher magnification image of M2 (inset) for ChR2(H134R) (left) and ChRger2 (right). Unilateral blue light stimulation of M2 induced turning behavior exclusively with ChRger2 and not ChR2(H134R) (10 Hz stimulation with 5 ms 447 nm light pulses at 20 mW). ChRger2, n = 5 animals; ChR2(H134R), n = 5 animals. No turning behavior was observed in any animal with 10 Hz stimulation with 5 ms 671 nm light pulses (20 mW). Plotted data are mean±SEM. vg, viral genomes.\\n\\nThe figure contains several panels illustrating the experimental setup and results:\\n\\na. Systemic delivery and fiber implantation diagrams for Dat-Cre mice.\\nb. Fluorescence image showing ChR2-TS-eYFP expression in the VTA.\\nc. Graphs showing intracranial self-stimulation results for ChRger2 and ChR2 at different stimulation powers and frequencies.\\nd. Systemic delivery and fiber on skull surface diagrams for WT mice.\\ne. Fluorescence images comparing anti-eYFP and ChR2-TS-eYFP expression.\\nf. Graphs showing minimally invasive optical control of locomotion for ChRger2 and ChR2 at different wavelengths.\\ng. Time-lapse images showing mouse movement in response to optical stimulation.\\n\\nThe figure demonstrates the superior performance of ChRger2 over ChR2(H134R) in both intracranial self-stimulation and minimally-invasive optogenetic control of locomotion experiments.'}]"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page_dicts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[PageMetadata(page_number=1, figure_count=0, figure_names=[], section_headers=['Abstract', 'Introduction'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=2, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=3, figure_count=0, figure_names=[], section_headers=['Results', 'Functional characterization of ChR variants for machine learning', 'Training Gaussian process (GP) classification and regression models'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=4, figure_count=0, figure_names=[], section_headers=['Selection of designer ChRs using trained models'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=5, figure_count=0, figure_names=[], section_headers=['Sequence and structural determinants of ChR functional properties'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=6, figure_count=0, figure_names=[], section_headers=['Machine-guided search identifies ChRs with a range of useful functional properties'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=7, figure_count=0, figure_names=[], section_headers=['Validation of designer ChRs for neuroscience applications', 'Designer ChRs and systemic AAVs enable minimally-invasive optogenetic excitation'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=8, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=9, figure_count=0, figure_names=[], section_headers=['Discussion'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=10, figure_count=0, figure_names=[], section_headers=['Online methods', 'Construct design and cloning', 'HEK293T cell and primary neuronal cultures'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=11, figure_count=0, figure_names=[], section_headers=['Patch-clamp electrophysiology'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=12, figure_count=0, figure_names=[], section_headers=['Light delivery and imaging', 'Electrophysiology data analysis', 'AAV production and purification'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=13, figure_count=0, figure_names=[], section_headers=['Animals', 'Intravenous injections, stereotactic injections, and cannula implantation', 'Gaussian process modeling'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=14, figure_count=0, figure_names=[], section_headers=['GP regression'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=15, figure_count=0, figure_names=[], section_headers=['GP classification', 'GP kernels for modeling proteins'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=16, figure_count=0, figure_names=[], section_headers=['L1 regression feature identification and weighting', 'Statistical analysis', 'Data availability', 'Code availability', 'Supplementary Material', 'Acknowledgements'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=17, figure_count=0, figure_names=[], section_headers=['References'], contains_fig_caption=False),\n",
       " PageMetadata(page_number=18, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=19, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=20, figure_count=1, figure_names=['Figure 1.'], section_headers=[], contains_fig_caption=True),\n",
       " PageMetadata(page_number=21, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=22, figure_count=1, figure_names=['Figure 2.'], section_headers=[], contains_fig_caption=True),\n",
       " PageMetadata(page_number=23, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=False),\n",
       " PageMetadata(page_number=24, figure_count=1, figure_names=['Figure 3.'], section_headers=[], contains_fig_caption=True),\n",
       " PageMetadata(page_number=25, figure_count=0, figure_names=[], section_headers=[], contains_fig_caption=True),\n",
       " PageMetadata(page_number=26, figure_count=1, figure_names=['Figure 4.'], section_headers=[], contains_fig_caption=True)]"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page_metadata_responses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'page': 1,\n",
       " 'text': '                                     HHS Public Access\\n                                     Author manuscript\\nAuthor Manuscript                    Nat Methods. Author manuscript; available in PMC 2020 April 14.\\n                       Published in final edited form as:\\n                         Nat Methods. 2019 November ; 16(11): 1176–1184. doi:10.1038/s41592-019-0583-8.\\n\\n\\n\\n                       Machine learning-guided channelrhodopsin engineering enables\\n                       minimally-invasive optogenetics\\n                       Claire N. Bedbrook1, Kevin K. Yang2,†, J. Elliott Robinson1,†, Elisha D. Mackey1, Viviana\\n                       Gradinaru1,*, Frances H. Arnold1,2,*\\n                       1Division of Biology and Biological Engineering; California Institute of Technology; Pasadena;\\nAuthor ManuscriptCalifornia; USA\\n                       2Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena;\\n                       California; USA\\n\\n\\n\\n                       Abstract\\n                             We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity\\n                             enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the\\n                             mammalian brain require intracranial surgery for transgene delivery and implantation of invasive\\n                             fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate\\n                             expansive optogenetics without the need for invasive implants, our engineering approach leverages\\n                             the significant literature of ChR variants to train statistical models for the design of new, high-\\nAuthor Manuscriptperformance ChRs. With Gaussian Process models trained on a limited experimental set of 102\\n                             functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light\\n                             sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via\\n                             minimally-invasive systemic transgene delivery, not possible previously due to low per-cell\\n                             transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation\\n                             without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables\\n                             minimally-invasive optogenetics.\\n\\n\\n\\n                       Introduction\\n                                           Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae.\\n                                           Transgenic expression of ChRs in the brain enables light-dependent neuronal activation1.\\nAuthor Manuscript                          These channels are widely applied as tools in neuroscience research2; however, functional\\n\\n\\n\\n                       Users may view, print, copy, and download text and data-mine the content in such documents, for the purposes of academic research,\\n                       *subject always to the full Conditions of use:http://www.nature.com/authors/editorial_policies/license.html#terms\\n                       †K.K.Y. & J.E.R. contributed equally to this work.\\n                                                                         Corresponding: viviana@caltech.edu (V.G.) & fha@cheme.caltech.edu (F.H.A).\\n                       Author Contributions\\n                       C.N.B., K.K.Y., V.G., and F.H.A. conceptualized the project. C.N.B. coordinated all experiments and data analysis. C.N.B. and K.K.Y.\\n                       built machine-learning models. C.N.B. performed construct design and cloning. C.N.B. and E.D.M. performed AAV production.\\n                       E.D.M. prepared cultured neurons. C.N.B and J.E.R. conducted electrophysiology. C.N.B. and J.E.R. performed injections. J.E.R.\\n                       performed fiber cannula implants and behavioral experiments. C.N.B. performed all data analysis. C.N.B. wrote the manuscript with\\n                       input and editing from all authors. V.G. supervised optogenetics/electrophysiology, and F.H.A. supervised the protein engineering.\\n                       Competing interests\\n                       A provisional patent application (CIT File No.: CIT-8092-P) has been filed by Caltech based on these results. C.N.B., K.K.Y., V.G.,\\n                       and F.H.A. are inventors on this provisional patent.',\n",
       " 'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics\\n\\nClaire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n## Abstract\\n\\nWe engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.\\n\\n## Introduction\\n\\nChannelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       " 'images': [{'name': 'img_p0_1.png',\n",
       "   'height': 102,\n",
       "   'width': 102,\n",
       "   'x': 61.5,\n",
       "   'y': 22.5,\n",
       "   'original_width': 152,\n",
       "   'original_height': 152,\n",
       "   'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-img_p0_1.png',\n",
       "   'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "   'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "   'page_number': 1},\n",
       "  {'name': 'page_1.jpg',\n",
       "   'height': 792,\n",
       "   'width': 612,\n",
       "   'x': 0,\n",
       "   'y': 0,\n",
       "   'type': 'full_page_screenshot',\n",
       "   'path': './storage/tmp/data_images/6f087e76-e0dd-470f-a0aa-d16fed3c55f2-page_1.jpg',\n",
       "   'job_id': '6f087e76-e0dd-470f-a0aa-d16fed3c55f2',\n",
       "   'original_file_path': '/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039.pdf',\n",
       "   'page_number': 1}],\n",
       " 'items': [{'type': 'heading',\n",
       "   'lvl': 1,\n",
       "   'value': 'Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "   'md': '# Machine learning-guided channelrhodopsin engineering enables minimally-invasive optogenetics',\n",
       "   'bBox': {'x': 90, 'y': 142, 'w': 429.0159999999997, 'h': 14}},\n",
       "  {'type': 'text',\n",
       "   'value': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "   'md': 'Claire N. Bedbrook<sup>1</sup>, Kevin K. Yang<sup>2,†</sup>, J. Elliott Robinson<sup>1,†</sup>, Elisha D. Mackey<sup>1</sup>, Viviana Gradinaru<sup>1,*</sup>, Frances H. Arnold<sup>1,2,*</sup>\\n\\n<sup>1</sup>Division of Biology and Biological Engineering; California Institute of Technology; Pasadena; California; USA\\n\\n<sup>2</sup>Division of Chemistry and Chemical Engineering; California Institute of Technology; Pasadena; California; USA',\n",
       "   'bBox': {'x': 90, 'y': 233, 'w': 68.30000000000004, 'h': 10}},\n",
       "  {'type': 'heading',\n",
       "   'lvl': 2,\n",
       "   'value': 'Abstract',\n",
       "   'md': '## Abstract',\n",
       "   'bBox': {'x': 89, 'y': 295, 'w': 48.672, 'h': 12}},\n",
       "  {'type': 'text',\n",
       "   'value': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "   'md': 'We engineered light-gated channelrhodopsins (ChRs) whose current strength and light sensitivity enable minimally-invasive neuronal circuit interrogation. Current ChR tools applied to the mammalian brain require intracranial surgery for transgene delivery and implantation of invasive fiber-optic cables to produce light-dependent activation of a small volume of tissue. To facilitate expansive optogenetics without the need for invasive implants, our engineering approach leverages the significant literature of ChR variants to train statistical models for the design of new, high-performance ChRs. With Gaussian Process models trained on a limited experimental set of 102 functionally characterized ChRs, we designed high-photocurrent ChRs with unprecedented light sensitivity; three of these, ChRger1–3, enable optogenetic activation of the nervous system via minimally-invasive systemic transgene delivery, not possible previously due to low per-cell transgene copy produced by systemic delivery. ChRger2 enables light-induced neuronal excitation without invasive intracranial surgery for virus delivery or fiber optic implantation, i.e. enables minimally-invasive optogenetics.',\n",
       "   'bBox': {'x': 90, 'y': 162, 'w': 394.5899999999998, 'h': 14}},\n",
       "  {'type': 'heading',\n",
       "   'lvl': 2,\n",
       "   'value': 'Introduction',\n",
       "   'md': '## Introduction',\n",
       "   'bBox': {'x': 90, 'y': 520, 'w': 69.756, 'h': 12}},\n",
       "  {'type': 'text',\n",
       "   'value': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "   'md': 'Channelrhodopsins (ChRs) are light-gated ion channels found in photosynthetic algae. Transgenic expression of ChRs in the brain enables light-dependent neuronal activation<sup>1</sup>. These channels are widely applied as tools in neuroscience research<sup>2</sup>; however, functional\\n\\n[The rest of the introduction is cut off in the image]',\n",
       "   'bBox': {'x': 90, 'y': 520, 'w': 345.41999999999985, 'h': 12}}]}"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "json_objs[0][\"pages\"][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "page_1_img = PIL.Image.open(\"/Users/loyalshababo/dev/plotreader/sandbox/storage/tmp/nihms-1538039_0.png\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
